---
title: "HIV Microbiome Analyses - Patel et al"
author: "Sweta Patel"
date: "10/4/2023"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

**PULL IN ALL DEMOGRAPHICS CODE, ETC HERE**

We need to re-do all of our relative abundance plots with the same dataframe as the source. So we will create melted_df here (skipping relative_df) and recreate our relative abundance plot from new figure 1 to make sure that looks consistent. Next, we will filter our melted_df dataframe to create the following relative abundance plots:

1. CD4 class normal vs low (Fig 1D)
2. any breastfeeding vs not (Fig S3)
3. sibling pairs (Fig S4)
4. HEU, HUU, and immunocompetent vs immunocompromised CLWH (Fig S6)

###All relative abundance plots

####Creating original melted_df_all dataframe
```{r}
chisib_prune = subset_samples(pruned, subject != "mom")

phy.otus <- transform_sample_counts(chisib_prune, function(Abundance) Abundance/sum(Abundance))
otu_df <- psmelt(phy.otus)
otu_df$OTU <- as.character(otu_df$OTU)
otu_abundances <- aggregate(otu_df$Abundance, by=list(OTU=otu_df$OTU), FUN=sum)
otu_abundances$x <- (otu_abundances$x)/(nsamples(chisib_prune))
names(otu_abundances)[names(otu_abundances) == 'x'] <- 'otu_Ab'
sum(otu_abundances$otu_Ab)  #sums to 1
nrow(otu_abundances)        #6709 species
remove(phy.otus, otu_df)

# Agglomerate taxa into species --> already done, so no need to do again
# Transform to relative abundances
chisib.rel <- transform_sample_counts(chisib_prune, function(Abundance) Abundance/sum(Abundance))
sample_sums(chisib.rel)  # This is a sanity check to make sure that relative abundance was calculated for each sample prior to pooling data
#All samples add to 1
melted_df_all <- psmelt(chisib.rel)

#Now create a dataframe of species abundances that we can then arrange in order. Will rename everything that is not the top 10 as "other"
melted_df_all$Species <- as.character(melted_df_all$Species)
species_abundances <- aggregate(melted_df_all$Abundance, by=list(Species=melted_df_all$Species,
                                                           OTU=melted_df_all$OTU), FUN=mean)
species_abundances <- aggregate(species_abundances$x, by=list(Species=species_abundances$Species), FUN=sum)
names(species_abundances)[names(species_abundances) == 'x'] <- 'species_Ab'
sum(species_abundances$species_Ab)    # Should sum to 1 (sum of relative abundances of species)
nrow(species_abundances)            # Corresponds to # of unique species: 6706 ***WHY is this different from 6709?###

#create rule for top 10 species
species_abundances <- arrange(species_abundances, desc(species_Ab))  
TOPSpecies <- unique(species_abundances$Species[1:10])
species_df <- species_abundances[species_abundances$Species %in% TOPSpecies,]
species_df$Species <- factor(species_df$Species, levels = species_df$Species[order(-species_df$species_Ab)])

#rename all non-top 10 species in melted_df as "other" and see if this works
melted_df_all$Species[!(melted_df_all$Species %in% TOPSpecies)] <- "Other"
sum(melted_df_all$Abundance) #sums to 153 exactly; n=153 so tracks?
table(melted_df_all$Species)

#so far so good! all of our top 10 species are consistent
```

Next, we will subset out n=143 kids (melted_df) create the color palette and variables needed for Figure 1B

```{r}
#Subset out melted_df from melted_df_all for most of our analyses and sub-analyses: 
melted_df <- subset(melted_df_all, melted_df_all$subject == "child")

#To generate our standard color palette for this manuscript: 
library(RColorBrewer)
library(scales)
chisp2 <- c("#293352", "#800000FF", "#767676FF", "#FFA319FF", "#8A9045FF", "#155F83FF", "#C16622FF", "#4D004B", "#8F3931FF", "#58593FFF", "#350E20FF")
show_col(chisp2)

#source for rotating x labels: https://stackoverflow.com/questions/1330989/rotating-and-spacing-axis-labels-in-ggplot2
melted_df$subject_abundances <- (melted_df$Abundance)/(length(unique(melted_df$Sample))) 
#abbreviate HIV status
melted_df$hiv2 <- NA
melted_df$hiv2 [melted_df$hiv == "Infected"] <- "CLWH"
melted_df$hiv2 [melted_df$hiv == "Exposed Uninfected"] <- "HEU children"
melted_df$hiv2 [melted_df$hiv == "Unexposed"] <- "HUU children"
table(melted_df$hiv2)
table(melted_df$hiv)
#matches up with table(melted_df$hiv)
melted_df$hiv2 <- as.factor(melted_df$hiv2)
melted_df$hiv2 <- reorder(melted_df$hiv2, new.order=c("CLWH", "HEU children", "HUU children"))
melted_df$Species <- factor(melted_df$Species)
###FIRST, create figure without re-ordering anything, screenshot it, and then create a new version that is re-ordered and compare the 2. The point of this is to make sure we're not causing any errors in the renaming process

##############
#NO REORDERING
##############
all_raw <- ggplot(melted_df[order(melted_df$Species, decreasing = TRUE),], aes(x=hiv2, y=subject_abundances, fill=Species)) +
  geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2)

#save for comparison as png
png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig1b_raw.png", width = 9, height = 5, units = 'in', res = 600)
all_raw
dev.off()
```

- Now will create our barplot for figure 1B with the species reordered

```{r}
#reorder species names
melted_df$Species <- reorder(melted_df$Species, new.order=c("Corynebacterium accolens", "Corynebacterium propinquum", "Corynebacterium pseudodiphtheriticum", "Dolosigranulum pigrum", "Haemophilus influenzae", "Moraxella catarrhalis", "Moraxella lincolnii", "Moraxella nonliquefaciens", "Staphylococcus aureus", "Streptococcus pneumoniae", "Other"))

library(ggtext)
p2 <- ggplot(melted_df[order(melted_df$Species, decreasing = TRUE),], aes(x=hiv2, y=subject_abundances, fill=Species)) +
  geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  xlab("HIV Status") + ylab("Mean relative abundance") 

p2
#this looks right when compared to prior figure 1b and species_df abundances (overall stacked rows eyeball as expected based on overall abundances)

#can we create a variable with abundance by HIV status?
melted_df$hiv_abundances <- (melted_df$Abundance)/(length(unique(melted_df$hiv2))) 
table(melted_df$hiv_abundances) #nope
#what if we aggregate species abundances within our melted_df dataframe?
melted_df$species_abundances <- aggregate(melted_df$Abundance, by=list(OTU=melted_df$OTU), FUN=mean) #mismatch in rows

```

#####How can we check our numbers? 

There is probably a more efficient way to do this, but for now will create individual dataframes for our top 10 species and compare abundances by our variable of interest (hiv and cd4 status for fig 1, etc) to make sure what we are visualizing in the plot makes sense.

Creating species-specific dataframes:

```{r}
#Corynebacterium accolens
cor_acc <- filter(melted_df, Species == "Corynebacterium accolens")
summary(cor_acc$Abundance)
tapply(cor_acc$Abundance, cor_acc$hiv2, summary)

#Corynebacterium propinquum
cor_p <- filter(melted_df, Species == "Corynebacterium propinquum")
summary(cor_p$Abundance) #matches up with species_df abundance
tapply(cor_p$Abundance, cor_p$hiv2, summary)

#Corynebacterium pseudodiphtheriticum
cor_ps <- filter(melted_df, Species == "Corynebacterium pseudodiphtheriticum")
summary(cor_ps$Abundance)
tapply(cor_ps$Abundance, cor_ps$hiv2, summary)

#Dolosigranulum pigrum
dpig <- filter(melted_df, Species == "Dolosigranulum pigrum")
summary(dpig$Abundance)
tapply(dpig$Abundance, dpig$hiv2, summary)

#Haemophilus influenzae
hflu <- filter(melted_df, Species == "Haemophilus influenzae")
summary(hflu$Abundance)
tapply(hflu$Abundance, hflu$hiv2, summary)

#Moraxella cattarhalis
mcat <- filter(melted_df, Species == "Moraxella catarrhalis")
summary(mcat$Abundance)
tapply(mcat$Abundance, mcat$hiv2, summary) 

#Moraxella lincolnii
mlin <- filter(melted_df, Species == "Moraxella lincolnii")
summary(mlin$Abundance)
tapply(mlin$Abundance, mlin$hiv2, summary) 

#Moraxella nonliquefaciens
mnon <- filter(melted_df, Species == "Moraxella nonliquefaciens")
summary(mnon$Abundance)
tapply(mnon$Abundance, mnon$hiv2, summary)

#Staphylococcus aureus
sa <- filter(melted_df, Species == "Staphylococcus aureus")
summary(sa$Abundance)
tapply(sa$Abundance, sa$hiv2, mean)

#Streptococcus pneumoniae
spneum <- filter(melted_df, Species == "Streptococcus pneumoniae")
summary(spneum$Abundance)
tapply(spneum$Abundance, spneum$hiv2, mean)

#Other
oth <- filter(melted_df, Species == "Other")
summary(oth$Abundance)
aggregate(Abundance ~ Species, data = oth, sum)
#sum of all Other abundances = 25.47148; if we divide this by our total # children (n=143), does this give us an average of all the other species present? --> comes out to 0.1781222; if we add this to the abundances in our species_df dataframe, it adds up to 1!!!
aggregate(Abundance ~ Species + hiv2, data = oth, sum)
#CLWH: 7.805837/44 = 0.1774054
#HEU:  9.548323/49 = 0.1948637
#HUU:  8.117322/50 = 0.1623464

#Created table in excel with this data; roughly looks to match up with our plot
```


####CD4 classification: normal vs low
```{r}
#create a subsetted df of melted_df that only has CLWH
melt_cd4 <- subset(melted_df, melted_df$hiv == "Infected")
table(melt_cd4$Species)
sum(melt_cd4$Abundance) #confirmed n=44

#create our CD4 variables in that dataframe
melt_cd4$newest_cd4 <- NA
melt_cd4$newest_cd4 [melt_cd4$chi_cd4_weeks1 < melt_cd4$chi_cd4_weeks2] <- 1
melt_cd4$newest_cd4 [melt_cd4$chi_cd4_weeks2 < melt_cd4$chi_cd4_weeks1] <- 2
melt_cd4$newest_cd4 [melt_cd4$chi_cd4_weeks2 < 0] <- 1  #we want to include the cd4 collected BEFORE enrollment where possible
melt_cd4$newest_cd4 [is.na(melt_cd4$chi_cd4_perc_2)] <- 1    #if we are missing cd4_2, we will use cd4_1
melt_cd4$newest_cd4 [is.na(melt_cd4$chi_cd4_perc_1)] <- 2    #one child is missing cd4_1 but has cd4_2
melt_cd4$newest_cd4 [is.na(melt_cd4$chi_cd4_perc_1) & is.na(melt_cd4$chi_cd4_perc_2)] <- NA
table(melt_cd4$newest_cd4)

#create new dataframe without B-45 and B-52 (aka new df with children missing CD4 data removed)
melt_cd4 <- subset(melt_cd4, !is.na(newest_cd4))
#confirm sum
sum(melt_cd4$Abundance)
melt_cd4$Var1 <- melt_cd4$chi_cd4_perc_1
melt_cd4$Var2 <- melt_cd4$chi_cd4_perc_2

melt_cd4 <- melt_cd4 %>% rowwise() %>%
  mutate(newest_cd4perc = get(paste0('Var', newest_cd4)))

summary(melt_cd4$newest_cd4perc)
table(melt_cd4$newest_cd4perc)

#make new variable for immune competent CD4 or not, with CD4 percentage >=25% = immune competent
melt_cd4$cd4nL <- NA
melt_cd4$cd4nL [melt_cd4$newest_cd4perc >=25] <- "1"
melt_cd4$cd4nL [melt_cd4$newest_cd4perc < 25] <- "0"
table(melt_cd4$cd4nL)  #11 kids with CD4% < 25 (when you divide sum by 6709 rows per participant)

#New variable for CD4 cutoff
melt_cd4$cd4cat <- NA
melt_cd4$cd4cat [melt_cd4$cd4nL==1] <- "Normal CD4"
melt_cd4$cd4cat [melt_cd4$cd4nL==0] <- "Low CD4"
table(melt_cd4$cd4cat)
class(melt_cd4$cd4cat)
melt_cd4$cd4cat <- as.factor(melt_cd4$cd4cat)
melt_cd4$cd4cat <- reorder(melt_cd4$cd4cat, new.order=c("Low CD4", "Normal CD4"))

#create plot without adjusting labels (though NOTE: we already re-ordered the species variable in our OG melted_df dataframe)
cd4_raw <- ggplot(melt_cd4[order(melt_cd4$Species, decreasing = TRUE),], aes(x=cd4cat, y=subject_abundances, fill=Species)) +
  geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2)
#continues to look like V2 of our figure 1D (which looks different from V1 of figure 1D but also has different species)

png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig1d_raw.png", width = 9, height = 5, units = 'in', res = 600)
cd4_raw
dev.off()
```

- Now recreate Figure 1D with the reformatted labels
```{r}
p4 <- ggplot(melt_cd4[order(melt_cd4$Species, decreasing = TRUE),], aes(x=cd4cat, y=subject_abundances, fill=Species)) +
  geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  xlab("HIV Status") + ylab("Mean relative abundance")

p4
```

Will check our figure against the mean relative abundances of each species

```{r}
cor_acc <- filter(melt_cd4, Species == "Corynebacterium accolens")
summary(cor_acc$Abundance)
tapply(cor_acc$Abundance, cor_acc$cd4cat, mean)

#Corynebacterium propinquum
cor_p <- filter(melt_cd4, Species == "Corynebacterium propinquum")
tapply(cor_p$Abundance, cor_p$cd4cat, mean)

#Corynebacterium pseudodiphtheriticum
cor_ps <- filter(melt_cd4, Species == "Corynebacterium pseudodiphtheriticum")
tapply(cor_ps$Abundance, cor_ps$cd4cat, mean)

#Dolosigranulum pigrum
dpig <- filter(melt_cd4, Species == "Dolosigranulum pigrum")
tapply(dpig$Abundance, dpig$cd4cat, mean)

#Haemophilus influenzae
hflu <- filter(melt_cd4, Species == "Haemophilus influenzae")
tapply(hflu$Abundance, hflu$cd4cat, mean)

#Moraxella cattarhalis
mcat <- filter(melt_cd4, Species == "Moraxella catarrhalis")
tapply(mcat$Abundance, mcat$cd4cat, mean) 

#Moraxella lincolnii
mlin <- filter(melt_cd4, Species == "Moraxella lincolnii")
tapply(mlin$Abundance, mlin$cd4cat, mean) 

#Moraxella nonliquefaciens
mnon <- filter(melt_cd4, Species == "Moraxella nonliquefaciens")
tapply(mnon$Abundance, mnon$cd4cat, mean)

#Staphylococcus aureus
sa <- filter(melt_cd4, Species == "Staphylococcus aureus")
tapply(sa$Abundance, sa$cd4cat, mean)

#Streptococcus pneumoniae
spneum <- filter(melt_cd4, Species == "Streptococcus pneumoniae")
tapply(spneum$Abundance, spneum$cd4cat, mean)

#Other
oth <- filter(melt_cd4, Species == "Other")
summary(oth$Abundance)
aggregate(Abundance ~ Species + cd4cat, data = oth, sum)
#Low CD4: 2.730539/11 = 0.2482308
#Normal CD4: 4.779737/31 = 0.1541851

```

Our numbers match well to our updated figure 1D, so I am confident about the validity of the figure.

**Create updated figure 1**
```{r}
fig1 <- plot_grid(p1, p2, p3, p4, labels = "AUTO", nrow = 2, ncol = 2, align = "h", axis = "b", rel_widths = c(1,1.3))

fig1

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig1NEW_V3.pdf", plot = fig1, dpi = 300, height = 9, width = 11)

png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig1NEW_V3.png",
    width = 11, height = 9, units = 'in', res = 600)
fig1
dev.off()
```

####Breastfeeding sensitivity analysis

```{r}
#create new variable in melted_df for any breastfeeding
#should have 84 children with any breastfeeding and 59 children who were only formula fed
melted_df$bfeed_any <- NA
melted_df$bfeed_any [melted_df$bfeed_1 == 1 | melted_df$bfeed_1 == 3] <- "Yes"
melted_df$bfeed_any [melted_df$bfeed_1 == 2] <- "No"
table(melted_df$bfeed_any) #when you divide by 6709 rows per child, consistent (59 No, 84 Yes)

table(child$bfeed_1)
child$bfeed_any <- NA
child$bfeed_any [child$bfeed_1 == 1 | child$bfeed_1 == 3] <- "Yes"
child$bfeed_any [child$bfeed_1 == 2] <- "No"
table(child$bfeed_any)
table(child$bfeed_any, child$hiv) #should have 21 CLWH, 16 HEU, and 47 HUU kids

#now subset out by breastfeeding status 
melt_bf <- subset(melted_df, melted_df$bfeed_any == "Yes")
table(melt_bf$Species)
sum(melt_bf$Abundance) #84; matches up
```

- create figures now with this dataframe

```{r}
class(melt_bf$bfeed_any)

#create plot and then compare to actual relative abundances
spec_bf <- ggplot(melt_bf[order(melt_bf$Species, decreasing = TRUE),], aes(x=hiv2, y=subject_abundances, fill=Species)) +
  geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  xlab("HIV Status") + ylab("Mean relative abundance")

spec_bf
```

Will check our figure against the mean relative abundances of each species

```{r}
#Corynebacterium accolens
cor_acc <- filter(melt_bf, Species == "Corynebacterium accolens")
tapply(cor_acc$Abundance, cor_acc$hiv2, mean)

#Corynebacterium propinquum
cor_p <- filter(melt_bf, Species == "Corynebacterium propinquum")
tapply(cor_p$Abundance, cor_p$hiv2, mean)

#Corynebacterium pseudodiphtheriticum
cor_ps <- filter(melt_bf, Species == "Corynebacterium pseudodiphtheriticum")
tapply(cor_ps$Abundance, cor_ps$hiv2, mean)

#Dolosigranulum pigrum
dpig <- filter(melt_bf, Species == "Dolosigranulum pigrum")
tapply(dpig$Abundance, dpig$hiv2, mean)

#Haemophilus influenzae
hflu <- filter(melt_bf, Species == "Haemophilus influenzae")
tapply(hflu$Abundance, hflu$hiv2, mean)

#Moraxella cattarhalis
mcat <- filter(melt_bf, Species == "Moraxella catarrhalis")
tapply(mcat$Abundance, mcat$hiv2, mean) 

#Moraxella lincolnii
mlin <- filter(melt_bf, Species == "Moraxella lincolnii")
tapply(mlin$Abundance, mlin$hiv2, mean) 

#Moraxella nonliquefaciens
mnon <- filter(melt_bf, Species == "Moraxella nonliquefaciens")
tapply(mnon$Abundance, mnon$hiv2, mean)

#Staphylococcus aureus
sa <- filter(melt_bf, Species == "Staphylococcus aureus")
tapply(sa$Abundance, sa$hiv2, mean)

#Streptococcus pneumoniae
spneum <- filter(melt_bf, Species == "Streptococcus pneumoniae")
tapply(spneum$Abundance, spneum$hiv2, mean)

#Other
oth <- filter(melt_bf, Species == "Other") #562716/6709 = 83.8748
summary(oth$Abundance)
aggregate(Abundance ~ Species + hiv2, data = oth, sum)
#have 21 CLWH, 16 HEU, and 47 HUU kids
#CLWH: 3.047206/21 = 0.145105
#HEU:  3.721417/16 = 0.2325886
#HUU: 7.243559/47 = 0.1541183
```

Our numbers match well to our updated figure, so I am confident about the validity of the figure.

####Sibling pairs (Fig S4)

```{r}
melt_sib <- subset(melted_df_all, melted_df_all$sample_id == "B.09.CHI" | melted_df_all$sample_id == "B.09.SIB" | melted_df_all$sample_id == "B.16.CHI" | melted_df_all$sample_id == "B.16.SIB" | melted_df_all$sample_id == "B.17.CHI" | melted_df_all$sample_id == "B.17.SIB" | melted_df_all$sample_id == "B.20.CHI" | melted_df_all$sample_id == "B.20.SIB" | melted_df_all$sample_id == "B.23.CHI" | melted_df_all$sample_id == "B.23.SIB" | melted_df_all$sample_id == "B.25.CHI" |
melted_df_all$sample_id == "B.25.SIB" | melted_df_all$sample_id == "B.32.CHI" | melted_df_all$sample_id == "B.32.SIB" | melted_df_all$sample_id == "B.35.CHI" | melted_df_all$sample_id == "B.35.SIB" | melted_df_all$sample_id == "B.48.CHI" | melted_df_all$sample_id == "B.48.SIB")


melt_sib$subject_abundances <- (melt_sib$Abundance)/(length(unique(melt_sib$Sample))) 

#new sibling label variable for figure
melt_sib$subject2 <- NA
melt_sib$subject2 [melt_sib$subject == "child"] <- "CLWH"
melt_sib$subject2 [melt_sib$subject == "sib"] <- "HEU siblings"
table(melt_sib$subject2)
melt_sib$subject2 <- as.factor(melt_sib$subject2)
melt_sib$subject2 <- reorder(melt_sib$subject2, new.order=c("CLWH", "HEU siblings"))

table(melt_sib$Species)

#reorder species names
class(melt_sib$Species)
melt_sib$Species <- as.factor(melt_sib$Species)
melt_sib$Species <- reorder(melt_sib$Species, new.order=c("Corynebacterium accolens", "Corynebacterium propinquum", "Corynebacterium pseudodiphtheriticum", "Dolosigranulum pigrum", "Haemophilus influenzae", "Moraxella catarrhalis", "Moraxella lincolnii", "Moraxella nonliquefaciens", "Staphylococcus aureus", "Streptococcus pneumoniae", "Other"))
```

- Now will create our barplot for figure 1B with the species reordered

```{r}
spec_sib <- ggplot(melt_sib[order(melt_sib$Species, decreasing = TRUE),], aes(x=subject2, y=subject_abundances, fill=Species)) +
  geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  xlab("HIV Status") + ylab("Mean relative abundance") 

spec_sib

#save as .png and ?.pdf for supplemental file
ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig_sib_NEW.pdf", plot = spec_sib, dpi = 300, width = 9, height = 5)

png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig sib NEW.png",
    width = 9, height = 5, units = 'in', res = 600)
spec_sib
dev.off()
```

Make sure the figures correspond to the relative abundances

```{r}
#Corynebacterium accolens
cor_acc <- filter(melt_sib, Species == "Corynebacterium accolens")
tapply(cor_acc$Abundance, cor_acc$subject2, mean)

#Corynebacterium propinquum
cor_p <- filter(melt_sib, Species == "Corynebacterium propinquum")
tapply(cor_p$Abundance, cor_p$subject2, mean)

#Corynebacterium pseudodiphtheriticum
cor_ps <- filter(melt_sib, Species == "Corynebacterium pseudodiphtheriticum")
tapply(cor_ps$Abundance, cor_ps$subject2, mean)

#Dolosigranulum pigrum
dpig <- filter(melt_sib, Species == "Dolosigranulum pigrum")
tapply(dpig$Abundance, dpig$subject2, mean)

#Haemophilus influenzae
hflu <- filter(melt_sib, Species == "Haemophilus influenzae")
tapply(hflu$Abundance, hflu$subject2, mean)

#Moraxella cattarhalis
mcat <- filter(melt_sib, Species == "Moraxella catarrhalis")
tapply(mcat$Abundance, mcat$subject2, mean) 

#Moraxella lincolnii
mlin <- filter(melt_sib, Species == "Moraxella lincolnii")
tapply(mlin$Abundance, mlin$subject2, mean) 

#Moraxella nonliquefaciens
mnon <- filter(melt_sib, Species == "Moraxella nonliquefaciens")
tapply(mnon$Abundance, mnon$subject2, mean)

#Staphylococcus aureus
sa <- filter(melt_sib, Species == "Staphylococcus aureus")
tapply(sa$Abundance, sa$subject2, mean)

#Streptococcus pneumoniae
spneum <- filter(melt_sib, Species == "Streptococcus pneumoniae")
tapply(spneum$Abundance, spneum$subject2, mean)

#Other
oth <- filter(melt_sib, Species == "Other") #120582/6709 = 17.97
aggregate(Abundance ~ Species + subject2, data = oth, sum)
#CLWH: 1.751258/9 = 0.1945842
#HEU sibs: 1.313867/9 = 0.1459852
```

Our numbers match well to our updated figure, so I am confident about the validity of the figure.

####HEU, HUU, and immunocompetent vs immunocompromised CLWH (Fig S6)

**Immunocompromised CLWH compared to HEU and HUU children**
```{r}
melt_hlowcd4 <- subset(melted_df, melted_df$sample_id != "B.06.CHI" & melted_df$sample_id != "B.07.CHI" & melted_df$sample_id != "B.12.CHI" & melted_df$sample_id != "B.13.CHI" & melted_df$sample_id != "B.14.CHI" & melted_df$sample_id != "B.15.CHI" &
melted_df$sample_id != "B.16.CHI" & melted_df$sample_id != "B.17.CHI" & melted_df$sample_id != "B.19.CHI" & melted_df$sample_id != "B.20.CHI" & melted_df$sample_id != "B.22.CHI" & melted_df$sample_id != "B.23.CHI" & melted_df$sample_id != "B.26.CHI" & melted_df$sample_id != "B.28.CHI" & melted_df$sample_id != "B.29.CHI" & melted_df$sample_id != "B.30.CHI" & melted_df$sample_id != "B.31.CHI" & melted_df$sample_id != "B.33.CHI" & melted_df$sample_id != "B.34.CHI" & melted_df$sample_id != "B.35.CHI" & melted_df$sample_id != "B.36.CHI" &  melted_df$sample_id != "B.37.CHI" & melted_df$sample_id != "B.38.CHI" & melted_df$sample_id != "B.39.CHI" & melted_df$sample_id != "B.40.CHI" & melted_df$sample_id != "B.41.CHI" & melted_df$sample_id != "B.44.CHI" & melted_df$sample_id != "B.45.CHI" & melted_df$sample_id != "B.48.CHI" & melted_df$sample_id != "B.49.CHI" & melted_df$sample_id != "B.50.CHI" & melted_df$sample_id != "B.51.CHI" & melted_df$sample_id != "B.52.CHI")

#Should end up with 11 CLWH, 49 HEU, 50 HUU --> n=110
table(melt_hlowcd4$hiv2)

#New HIV category variable for figure
melt_hlowcd4$hiv2 <- NA
melt_hlowcd4$hiv2 [melt_hlowcd4$hiv == "Infected"] <- "Immunocompromised CLWH"
melt_hlowcd4$hiv2 [melt_hlowcd4$hiv == "Exposed Uninfected"] <- "HEU children"
melt_hlowcd4$hiv2 [melt_hlowcd4$hiv == "Unexposed"] <- "HUU children"
table(melt_hlowcd4$hiv2)
#matches up with table(relative_df$hiv)
melt_hlowcd4$hiv2 <- as.factor(melt_hlowcd4$hiv2)
melt_hlowcd4$hiv2 <- reorder(melt_hlowcd4$hiv2, new.order=c("Immunocompromised CLWH", "HEU children", "HUU children"))

spec_hlow <- ggplot(melt_hlowcd4[order(melt_hlowcd4$Species, decreasing = TRUE),], aes(x=hiv2, y=subject_abundances, fill=Species)) + geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  xlab("HIV Status") + ylab("Mean relative abundance") 

spec_hlow
```

Make sure the figures correspond to the relative abundances

```{r}
#Corynebacterium accolens
cor_acc <- filter(melt_hlowcd4, Species == "Corynebacterium accolens")
tapply(cor_acc$Abundance, cor_acc$hiv2, mean)

#Corynebacterium propinquum
cor_p <- filter(melt_hlowcd4, Species == "Corynebacterium propinquum")
tapply(cor_p$Abundance, cor_p$hiv2, mean)

#Corynebacterium pseudodiphtheriticum
cor_ps <- filter(melt_hlowcd4, Species == "Corynebacterium pseudodiphtheriticum")
tapply(cor_ps$Abundance, cor_ps$hiv2, mean)

#Dolosigranulum pigrum
dpig <- filter(melt_hlowcd4, Species == "Dolosigranulum pigrum")
tapply(dpig$Abundance, dpig$hiv2, mean)

#Haemophilus influenzae
hflu <- filter(melt_hlowcd4, Species == "Haemophilus influenzae")
tapply(hflu$Abundance, hflu$hiv2, mean)

#Moraxella cattarhalis
mcat <- filter(melt_hlowcd4, Species == "Moraxella catarrhalis")
tapply(mcat$Abundance, mcat$hiv2, mean) 

#Moraxella lincolnii
mlin <- filter(melt_hlowcd4, Species == "Moraxella lincolnii")
tapply(mlin$Abundance, mlin$hiv2, mean) 

#Moraxella nonliquefaciens
mnon <- filter(melt_hlowcd4, Species == "Moraxella nonliquefaciens")
tapply(mnon$Abundance, mnon$hiv2, mean)

#Staphylococcus aureus
sa <- filter(melt_hlowcd4, Species == "Staphylococcus aureus")
tapply(sa$Abundance, sa$hiv2, mean)

#Streptococcus pneumoniae
spneum <- filter(melt_hlowcd4, Species == "Streptococcus pneumoniae")
tapply(spneum$Abundance, spneum$hiv2, mean)

#Other
oth <- filter(melt_hlowcd4, Species == "Other") #736890/6709 = 109.836
aggregate(Abundance ~ Species + hiv2, data = oth, sum)
#Immunocompromised CLWH: 2.730539/11 = 0.2482308
#HEU: 9.548323/49 = 0.1948637
#HUU: 8.117322/50 = 0.1623464
```

Our numbers match well to our updated figure, so I am confident about the validity of the figure.
```

**Immunocompetent CLWH compared to HEU and HUU children**
```{r}
melt_hucd4 = subset(melted_df, sample_id != "B.01.CHI" & sample_id != "B.03.CHI" & sample_id != "B.04.CHI" &
                         sample_id != "B.05.CHI" & sample_id != "B.09.CHI" & sample_id != "B.11.CHI" &
                         sample_id != "B.18.CHI" & sample_id != "B.25.CHI" & sample_id != "B.27.CHI" &
                         sample_id != "B.32.CHI" & sample_id != "B.45.CHI" & sample_id != "B.46.CHI" &
                         sample_id != "B.52.CHI")
                         
#872170/6709 = 130

table(melt_hucd4$hiv2)  #31 CLWH; matches up 
                     
#New HIV category variable for figure
melt_hucd4$hiv2 <- NA
melt_hucd4$hiv2 [melt_hucd4$hiv == "Infected"] <- "Immunocompetent CLWH"
melt_hucd4$hiv2 [melt_hucd4$hiv == "Exposed Uninfected"] <- "HEU children"
melt_hucd4$hiv2 [melt_hucd4$hiv == "Unexposed"] <- "HUU children"
table(melt_hucd4$hiv2)
#matches up with table(relative_df$hiv)
melt_hucd4$hiv2 <- as.factor(melt_hucd4$hiv2)
melt_hucd4$hiv2 <- reorder(melt_hucd4$hiv2, new.order=c("Immunocompetent CLWH", "HEU children", "HUU children"))

spec_hu <- ggplot(melt_hucd4[order(melt_hucd4$Species, decreasing = TRUE),], aes(x=hiv2, y=subject_abundances, fill=Species)) + geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  xlab("HIV Status") + ylab("Mean relative abundance") 

spec_hu
```

Make sure the figures correspond to the relative abundances

```{r}
#Corynebacterium accolens
cor_acc <- filter(melt_hucd4, Species == "Corynebacterium accolens")
tapply(cor_acc$Abundance, cor_acc$hiv2, mean)

#Corynebacterium propinquum
cor_p <- filter(melt_hucd4, Species == "Corynebacterium propinquum")
tapply(cor_p$Abundance, cor_p$hiv2, mean)

#Corynebacterium pseudodiphtheriticum
cor_ps <- filter(melt_hucd4, Species == "Corynebacterium pseudodiphtheriticum")
tapply(cor_ps$Abundance, cor_ps$hiv2, mean)

#Dolosigranulum pigrum
dpig <- filter(melt_hucd4, Species == "Dolosigranulum pigrum")
tapply(dpig$Abundance, dpig$hiv2, mean)

#Haemophilus influenzae
hflu <- filter(melt_hucd4, Species == "Haemophilus influenzae")
tapply(hflu$Abundance, hflu$hiv2, mean)

#Moraxella cattarhalis
mcat <- filter(melt_hucd4, Species == "Moraxella catarrhalis")
tapply(mcat$Abundance, mcat$hiv2, mean) 

#Moraxella lincolnii
mlin <- filter(melt_hucd4, Species == "Moraxella lincolnii")
tapply(mlin$Abundance, mlin$hiv2, mean) 

#Moraxella nonliquefaciens
mnon <- filter(melt_hucd4, Species == "Moraxella nonliquefaciens")
tapply(mnon$Abundance, mnon$hiv2, mean)

#Staphylococcus aureus
sa <- filter(melt_hucd4, Species == "Staphylococcus aureus")
tapply(sa$Abundance, sa$hiv2, mean)

#Streptococcus pneumoniae
spneum <- filter(melt_hucd4, Species == "Streptococcus pneumoniae")
tapply(spneum$Abundance, spneum$hiv2, mean)

#Other
oth <- filter(melt_hucd4, Species == "Other") # 870870/6709 = 129.8062
aggregate(Abundance ~ Species + hiv2, data = oth, sum)
#Immunocompetent CLWH: 4.779737/31 = 0.1541851
#HEU: 9.548323/49 = 0.1948637
#HUU: 8.117322/50 = 0.1623464
```

Our numbers match well to our updated figure, so I am confident about the validity of the figure.
```
#### Sample sums (For Lancet Microbe appeal)
We have our total number of paired reads that remained after processing via Kaiju and kneaddata, but what is left after we prune and use decontam for the children and siblings in our study? 

```{r}
sums <- sample_sums(chisib_prune)
sums <- as.data.frame(sums)
#save sample_ID as variable
sums$sample_id <- rownames(sums)

sorted_sums <- sums[order(sums$sums),]

summary(sums$sums)

#Now we will create a sample_sums variable just for the children and merge this with our metadata file to look for differences by HIV status:
chi_prune = subset_samples(pruned, subject == "child")
chisums <- sample_sums(chi_prune)
chisums <- as.data.frame(chisums)
#save sample_ID as variable
chisums$sample_id <- rownames(chisums)
summary(chisums$chisums)

child2 <- merge(child, chisums, by="sample_id")
tapply(child2$chisums, child$hiv, summary)
kruskal.test(child2$chisums, child$hiv) 
```

#### Rarefaction curves (for LM appeal)
- Ranacapa package in R --> has 'ggrare' function that M. Corder has used.
- tutorial for ggrare: https://github.com/gauravsk/ranacapa/blob/master/R/ggrare.R
- step parameter (ChatGPT explanation): The "step" parameter determines how much the rarefied sample size increases at each step of the rarefaction process. For example, if you set the step to 100, the rarefaction curve will be calculated for sample sizes increasing by increments of 100 reads.

Choosing an appropriate step size depends on the distribution of reads across your samples and the desired resolution of your rarefaction curve. Smaller step sizes will result in more points on the rarefaction curve and finer resolution, but may also increase computation time, especially for large datasets. Conversely, larger step sizes will result in fewer points and coarser resolution but faster computation.

Initially set step to 10 because I wasn't thinking of the number of reads - oops. Since I have a median of 77K reads per sample, should probably set to 1000 and go down from there depending on what the curves look like
```{r}
library(ranacapa)
ggrare(chisib_prune, step = 1000, label = NULL, color = NULL,
    plot = TRUE, parallel = FALSE, se = TRUE)

p <- ggrare(chisib_prune, step = 1500, label = NULL, color = NULL,
    plot = TRUE, parallel = FALSE, se = TRUE)

#zooming in: start with sequence sample size (x axis) of 250K
p1 <- p + coord_cartesian(xlim = c(0, 250000)) #visually appears most samples flatten by <25K reads, so zoom more:
p2 <- p + coord_cartesian(xlim = c(0, 100000))
p3 <- p + coord_cartesian(xlim = c(0, 25000))
#Using prior plots of paired_reads (should have been sample_sums) vs #Observed species, we set a cutoff of 2500 reads; zoom in further to see what this looks like
p4 <- p + coord_cartesian(xlim = c(0, 10000))
#set Y lim to 750 to allow for more zooming
p5 <- p + coord_cartesian(xlim = c(0, 10000), ylim = c(0,600))
p6 <- p + coord_cartesian(xlim = c(0, 5000), ylim = c(0,600))

rarefaction <- plot_grid(p, p1, p2, p3, p4, labels = "AUTO", nrow = 2, ncol = 3, align = "h", axis = "b")
rarefaction  

rarefaction2 <- plot_grid(p, p5, p6, labels = "AUTO", nrow = 1, ncol = 3, align = "h", axis = "b")
rarefaction2

#save panel images for d/w MK:
ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/rarefaction_zoomx.pdf", plot = rarefaction, dpi = 300, height = 9, width = 11)

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/rarefaction_zoomxy.pdf", plot = rarefaction2, dpi = 300, height = 6, width = 11)
```

**D/w MK 3/14; need to generate rarefaction curves looking at multiple alpha diversity indices including Shannon and Simpson indices. I don't see a way to do this in ggrare. Found separate code here that I will try: https://github.com/joey711/phyloseq/issues/143 (nr0cinu)

```{r}
#pull in decontaminated phyloseq that is not pruned:
meta <- readRDS("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/phy.kaiju.decontam.04192022.rds")

calculate_rarefaction_curves <- function(meta, measures, depths) {
  require('plyr') # ldply
  require('reshape2') # melt

  estimate_rarified_richness <- function(meta, measures, depth) {
    if(max(sample_sums(meta)) < depth) return()
    meta <- prune_samples(sample_sums(meta) >= depth, meta)

    rarified_meta <- rarefy_even_depth(meta, depth, verbose = FALSE)

    alpha_diversity <- estimate_richness(rarified_meta, measures = measures)

    # as.matrix forces the use of melt.array, which includes the Sample names (rownames)
    molten_alpha_diversity <- melt(as.matrix(alpha_diversity), varnames = c('Sample', 'Measure'), value.name = 'Alpha_diversity')

    molten_alpha_diversity
  }
  
  names(depths) <- depths # this enables automatic addition of the Depth to the output by ldply
  rarefaction_curve_data <- ldply(depths, estimate_rarified_richness, meta = meta, measures = measures, .id = 'Depth', .progress = ifelse(interactive(), 'text', 'none'))

  # convert Depth from factor to numeric
  rarefaction_curve_data$Depth <- as.numeric(levels(rarefaction_curve_data$Depth))[rarefaction_curve_data$Depth]

  rarefaction_curve_data
}

rarefaction_curve_data <- calculate_rarefaction_curves(meta, c('Observed', 'Shannon', 'Simpson'), rep(c(1, 10, 100, 1000, 1:100 * 10000), each = 10)) #step count here should be 10K if I'm interpreting code correctly 
summary(rarefaction_curve_data)

rarefaction_curve_data_summary <- ddply(rarefaction_curve_data, c('Depth', 'Sample', 'Measure'), summarise, Alpha_diversity_mean = mean(Alpha_diversity), Alpha_diversity_sd = sd(Alpha_diversity))

#save this dataframe so we don't have to run it again if R crashes
write.csv(rarefaction_curve_data_summary, '/Users/swetapatel/OneDrive - Duke University/Fogarty coding/q_rare_summary.csv')

rarefaction_curve_data_summary <- read.csv('/Users/swetapatel/OneDrive - Duke University/Fogarty coding/q_rare_summary.csv')

q_rare_facet <- ggplot(
  data = rarefaction_curve_data_summary,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity_mean,
    ymin = Alpha_diversity_mean - Alpha_diversity_sd,
    ymax = Alpha_diversity_mean + Alpha_diversity_sd,
  )
) + geom_point(
# ) + geom_line(
# ) + geom_pointrange(
) + facet_wrap(
  facets = ~ Measure,
  scales = 'free_y'
)

#save facet figure, then create separate dataframes for Shannon and Simpson and create zoomed in plots
ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/q_rarefaction_facet.pdf", plot = q_rare_facet, dpi = 300, height = 9, width = 11)

rare_shan <- subset(rarefaction_curve_data_summary, rarefaction_curve_data_summary$Measure == "Shannon")
rare_simp <- subset(rarefaction_curve_data_summary, rarefaction_curve_data_summary$Measure == "Simpson")

#Shannon figures
q_rare_shan <- ggplot(
  data = rare_shan,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity_mean,
    ymin = Alpha_diversity_mean - Alpha_diversity_sd,
    ymax = Alpha_diversity_mean + Alpha_diversity_sd,
  )
) + geom_pointrange() + labs(x="Depth", y = "Shannon index")

p1 <- q_rare_shan + coord_cartesian(xlim = c(0, 500000))
p2 <- q_rare_shan + coord_cartesian(xlim = c(0, 250000))
p3 <- q_rare_shan + coord_cartesian(xlim = c(0, 100000))
p4 <- q_rare_shan + coord_cartesian(xlim = c(0, 50000))
p5 <- q_rare_shan + coord_cartesian(xlim = c(0, 10000))

q_shan <- plot_grid(q_rare_shan, p2, p4, p5, labels = "AUTO", nrow = 1, ncol = 4, align = "h", axis = "b")
q_shan

#Simpson figures
q_rare_simp <- ggplot(
  data = rare_simp,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity_mean,
    ymin = Alpha_diversity_mean - Alpha_diversity_sd,
    ymax = Alpha_diversity_mean + Alpha_diversity_sd,
  )
) + geom_pointrange() + labs(x="Depth", y = "Simpson index")

p2 <- q_rare_simp + coord_cartesian(xlim = c(0, 250000))
p4 <- q_rare_simp + coord_cartesian(xlim = c(0, 50000))
p5 <- q_rare_simp + coord_cartesian(xlim = c(0, 10000))

q_simp <- plot_grid(q_rare_simp, p2, p4, p5, labels = "AUTO", nrow = 1, ncol = 4, align = "h", axis = "b")
q_simp

shan_simp <- plot_grid(q_shan, q_simp, labels = "AUTO", nrow = 2, ncol = 1, align = "h", axis = "b")
shan_simp

#save zoomed shan and simp:
ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/q_rarefaction_ss_zoom.pdf", plot = shan_simp, dpi = 300, height = 9, width = 11)

#Per MK 3/14, will focus on panel Ds from our combined shan_simp figure and 1) make points smaller, 2) add lines, and 3) try and create custom ticks on x axis
q_rare_shan2 <- ggplot(
  data = rare_shan,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity_mean,
    ymin = Alpha_diversity_mean - Alpha_diversity_sd,
    ymax = Alpha_diversity_mean + Alpha_diversity_sd,
    group = Sample,
  )
) + geom_point(size=0.5) + geom_line() + labs(x="Depth", y = "Shannon index") +
  scale_x_continuous(limits= c(0, 10000), breaks = c(500, 1000, 2500, 5000, 10000))

q_rare_simp2 <- ggplot(
  data = rare_simp,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity_mean,
    ymin = Alpha_diversity_mean - Alpha_diversity_sd,
    ymax = Alpha_diversity_mean + Alpha_diversity_sd,
    group = Sample,
  )
) + geom_point(size=0.5) + geom_line() + labs(x="Depth", y = "Simpson index") +
  scale_x_continuous(limits= c(0, 10000), breaks = c(500, 1000, 2500, 5000, 10000))

shan_simp2 <- plot_grid(q_rare_shan2, q_rare_simp2, labels = "AUTO", nrow = 1, ncol = 2, align = "h", axis = "b")

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/shan_simp_zoom10K.pdf", plot = shan_simp2, dpi = 300, height = 6, width = 10)

#add color, clean up figures so they are manuscript quality

#need to get rid of moms and color code participants by sib, clwh, heu, huu
rarefaction_curve_data_summary$subject <- "99"
rarefaction_curve_data_summary <- mutate(rarefaction_curve_data_summary, subject = case_when(grepl("MOM", Sample) ~ "mom", grepl("CHI", Sample) ~ "child", grepl("SIB", Sample) ~ "sib"))

table(rarefaction_curve_data_summary$subject) #visually reviewed dataframe and looks like it matches up

#Now get rid of the moms 
rare_kids <- subset(rarefaction_curve_data_summary, rarefaction_curve_data_summary$subject != "mom")
table(rare_kids$subject) #moms are removed

#Now create a new variable to color code by HIV or sib status
rare_kids$class <- "99"
rare_kids <- mutate(rare_kids, class = case_when(grepl("B.*CHI", Sample) ~ "CLWH", grepl("E.", Sample) ~ "HEU children", grepl("U.", Sample) ~ "HUU children", grepl("B.*SIB", Sample) ~ "HEU siblings"))
rare_kids$class <- as.factor(rare_kids$class)
rare_kids$class <- reorder(rare_kids$class, new.order=c("CLWH", "HEU children", "HUU children", "HEU siblings"))
table(rare_kids$class) #All the numbers add up

#Create color palate with 4 colors
chisp_rare <- c("#800000FF", "#155F83FF", "#FFA319FF", "#8A9045FF")
show_col(chisp_rare)

rare_shan <- subset(rare_kids, rare_kids$Measure == "Shannon")
rare_simp <- subset(rare_kids, rare_kids$Measure == "Simpson")

#Shannon panel 
q_rare_shan2 <- ggplot(
  data = rare_shan,
  mapping = aes(x = Depth, y = Alpha_diversity_mean, group = Sample, color = class
  )
) + geom_point(size=0.5) + geom_line() + labs(x="Depth", y = "Shannon index") +
  scale_x_continuous(limits= c(0, 10000))+
  scale_color_manual(values = chisp_rare) +
  theme_classic() +
  theme(
    legend.text = element_text(size = 10),
    legend.title = element_blank(),
    axis.text = element_text(size = 10),
    axis.title.y = element_text(size = 12),
    axis.title.x = element_text(size = 12))

#Simpson panel with legend
q_rare_simp2 <- ggplot(
  data = rare_simp,
  mapping = aes(x = Depth, y = Alpha_diversity_mean, group = Sample, color = class
  )
) + geom_point(size=0.5) + geom_line() + labs(x="Depth", y = "Simpson index") +
  scale_x_continuous(limits= c(0, 10000))+
  scale_color_manual(values = chisp_rare) +
  theme_classic() +
  theme(
    legend.text = element_text(size = 10),
    legend.title = element_blank(),
    axis.text = element_text(size = 10),
    axis.title.y = element_text(size = 12),
    axis.title.x = element_text(size = 12))

shan_simp2 <- plot_grid(q_rare_shan2, q_rare_simp2, labels = "AUTO", nrow = 1, ncol = 2, align = "h", axis = "b")
#rel_widths = c(1,1.2)

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/rarefaction_color.pdf", plot = shan_simp2, dpi = 300, height = 5, width = 11)
```

#### Sensitivity analysis discarding all samples with <10K reads (for LM appeal)
- will use sample_sums in our pruned object and create a new pseq object that only contains samples with >10K reads (n=  130)

Step 1: create 10Kpruned object
- original plots and pruning decisions generated from decontaminated Kaiju pseq object documented in 'Fog KAIJU 10-6-22.R' file
```{r}
#Pruning to >10K
chisib10K = prune_samples(sample_sums(chisib_prune)>=10000, chisib_prune)
#how many species do we have in our pruned sample? 
ntaxa(chisib10K)  #6709; unchanged from agglom, but we are down to 131 participants (153 - 131 = 22)

#metadata for 131 participants
meta10K <- data.frame(sample_data(chisib10K))

#save pruned pseq object
saveRDS(chisib10K, file = "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/phy.chisib10K_03032024.rds", ascii = FALSE, version = NULL,
        compress = FALSE, refhook = NULL)

chisib10K <- readRDS("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/phy.chisib10K_03032024.rds")

#create pseq object with only the kids
chi10k = subset_samples(chisib10K, subject == "child") #121 kids in analysis (down from 143)
#child metadata
chi10kmeta <- data.frame(sample_data(chi10k))
```

##### Alpha diversity analyses: >10K reads
```{r}
################
#ALPHA DIVERSITY
################
#Create dataframe with diversity indices for each specimen using PRUNED pseq object
diversity <- estimate_richness(chi10k, measures = c("Shannon", "Simpson", "Chao1", "Observed"))

setDT(diversity, keep.rownames = TRUE)[]
colnames(diversity) <- c("sample_id", "Observed", "Chao1", "se.chao1", "Shannon", "Simpson")
rownames(diversity) <- diversity$sample_id
diversity <- merge(diversity, chi10kmeta, by="sample_id")

class(diversity$hiv) #character; need to make factor to relevel
diversity$hiv <- as.factor(diversity$hiv)
diversity$hiv <- relevel(diversity$hiv, ref = "Unexposed")

class(diversity$wood) #integer; needs to be factor
diversity$wood <- as.factor(diversity$wood)
diversity$wood <- relevel(diversity$wood, ref = "0")

class(diversity$season) #character; needs to be factor
diversity$season <- as.factor(diversity$season)
diversity$season <- relevel(diversity$season, ref = "Dry")

class(diversity$abx) #character; needs to be factor
diversity$abx <- as.factor(diversity$abx)
diversity$abx <- relevel(diversity$abx, ref = "No")

class(diversity$uri_rec2) #character; needs to be factor
diversity$uri_rec2 <- as.factor(diversity$uri_rec2)
diversity$uri_rec2 <- relevel(diversity$uri_rec2, ref = "No")

########
#SHANNON
########
hist(diversity$Shannon) #slight right skew but rel normal
summary(diversity$Shannon) #median (IQR) 1.77 (1.46, 2.04); lower than full cohort: 1·85 (1·51, 2·26)
summary(div_10k <- lm(Shannon ~ hiv, data = diversity)) #no difference in univariable analysis
summary(div_10kmv <- lm(Shannon ~ age + hiv + wood + season + abx + uri_rec2, data = diversity)) #no difference in mv analysis

######
#CHAO1
######
#Is our Chao1 data normally distributed? 
histogram(diversity$Chao1) #v right skewed
qqnorm(diversity$Chao1, pch = 1, frame = FALSE)
qqline(diversity$Chao1, col = "steelblue", lwd = 2)

#What does it look like if we log transform it? Will this allow for parametric testing?
diversity$Chao1_log <- log(diversity$Chao1)
histogram(diversity$Chao1_log)
qqnorm(diversity$Chao1_log, pch = 1, frame = FALSE)
qqline(diversity$Chao1_log, col = "steelblue", lwd = 2)
  #Looks very normal! Can use linear regression on transformed data

#Chao1 median and IQR for all samples
summary(diversity$Chao1) # median (IQR) = 166 (91, 299); higher than full cohort: 137 (68, 261)

#analyses: no significant associations between covariables of interest and log transformed Chao1
summary(aov(Chao1_log ~ hiv, data = diversity))
summary(div_mv <- lm(Chao1_log ~ age + hiv + wood + season + abx + uri_rec2, data = diversity))

#################
#OBSERVED SPECIES
#################
tapply(diversity$Observed, diversity$hiv, summary)
kruskal.test(Observed ~ hiv, data = diversity) #p = 0.22
summary(obs_mv <- lm(Observed ~ age + hiv + wood + season + abx + uri_rec2, data = diversity))
```
**Summary: no variables were associated with measures of alpha diversity (Shannon index, log transformed Chao1, number of observed species) among children with >10K reads**

[] do we need a boxplot? Can revisit and add code once complete remainder of analyses

##### Beta diversity measures

- CLR transform for PCOA plots and PERMANOVA

```{r}
chi10k_clr <- microbiome::transform(chi10k, "clr")
```

**Plots**

```{r}
#First: create new HIV variable as factor, then relevel
table(chi10kmeta$hiv)
chi10kmeta$hiv2 <- NA
chi10kmeta$hiv2 [chi10kmeta$hiv == "Infected"] <- "CLWH"
chi10kmeta$hiv2 [chi10kmeta$hiv == "Exposed Uninfected"] <- "HEU children"
chi10kmeta$hiv2 [chi10kmeta$hiv == "Unexposed"] <- "HUU children"
table(chi10kmeta$hiv2)
class(chi10kmeta$hiv2)
chi10kmeta$hiv2 <- as.factor(chi10kmeta$hiv2)
chi10kmeta$hiv2 <- reorder(chi10kmeta$hiv2, new.order=c("CLWH", "HEU children", "HUU children"))

#Import back into pseq (source: https://jacobrprice.github.io/2017/08/26/phyloseq-to-vegan-and-back.html)
sample_data(chi10k_clr) <- as.data.frame(chi10kmeta)
#now see if it worked:
bloop <- data.frame(sample_data(chi10k_clr))
table(bloop$hiv2)
#success!
remove(bloop)

#Now ordinate and create PCOA plot
clr_pcoa <- ordinate(chi10k_clr, method = "PCoA", distance = "euclidean")
b1 <- plot_ordination(chi10k_clr, clr_pcoa, color = "hiv2") +
  geom_point(size = 0.005) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values=c('#800000FF','#155F83FF', '#FFA319FF'))+
  stat_ellipse(aes(linetype=hiv2), geom="polygon", alpha=0, type="t", level=0.8, size=0.5) +
  xlab("PC1 (15.3%)") + ylab("PC2 (7.9%)") +
  scale_linetype_manual(values= c("dashed", "dashed", "dashed"))

b1

#save for MK review:
ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/PCoA_all_10K.pdf", plot = b1, dpi = 300, height = 6, width = 6)

```

**PERMANOVA**

- will run individual permanovas for each covariable of interest, adjusted for age; for all multivariable comparisons, using 'by = margin' (see discussion in 5/20/23 RMD file)

```{r}
#Create OTU table
otu <- as.data.frame(otu_table(chi10k_clr, taxa_are_rows=TRUE))

#Need species in columns and observations in rows: transpose using t() function and save as a dataframe
  #Source: https://github.com/edamame-course/2015-tutorials/blob/master/final/Ashley_Intro_to_R.md
otu_trans <- data.frame(t(otu))

#Calculate distance and save as a matrix
BC.dist=vegdist(otu_trans, method="euclidean")

####
#HIV
####
adonis2(BC.dist ~ hiv, data = chi10kmeta, permutations = 1000)
  #Significant difference by HIV status: R2 = 0.022, P = 0.042; unchanged

adonis2(BC.dist ~ hiv + age, data = chi10kmeta, permutations = 1000, by = "margin")
  #results are pretty consistent: HIV: p = 0.044, R2 = 0.023; age: p = 0.07, R2 = 0.12

####
#Age
####
adonis2(BC.dist ~ age, data = chi10kmeta, permutations = 1000)
  #NO significant difference in overall composition by age (P = 0.09, R2 = 0.012); change from prior

#######
#Season
#######
adonis2(BC.dist ~ season, data = chi10kmeta, permutations = 1000)
  #Significant difference in overall composition by season remains (P = 0.007, R2 = 0.017)

adonis2(BC.dist ~ season + age, data = chi10kmeta, permutations = 1000, by = "margin")
#Significant difference by season (P = 0.01, R^2 = 0.017) but not age (P = 0.08, R2 = 0.011) 

#####
#Wood
#####
adonis2(BC.dist ~ wood, data = chi10kmeta, permutations = 1000)
  #NEW significant difference by wood smoke exposure among children (P = 0.045, R2=0.013)

adonis2(BC.dist ~ wood + age, data = chi10kmeta, permutations = 1000, by = "margin")
#NEW signif difference by wood smoke exposure (P = 0.045, R2 = 0.013) but NOT age: (P = 0.08, R^2 = 0.012)

####
#Abx
####
adonis2(BC.dist ~ abx, data = chi10kmeta, permutations = 1000)
  #Trend but no significant difference in overall composition by abx exposure (P = 0.08, R2=0.011); unchanged

adonis2(BC.dist ~ abx + age, data = chi10kmeta, permutations = 1000, by = "margin")
#No significant difference in overall composition by abx exposure or age

###########
#Recent URI
###########
adonis2(BC.dist ~ uri_rec2, data = chi10kmeta, permutations = 1000)
#Significant difference in composition among children with recent URI (P = 0.004, R2=0.019); unchanged

adonis2(BC.dist ~ uri_rec2 + age, data = chi10kmeta, permutations = 1000, by = "margin")
#Significant difference in composition by recent URI sx status (R2 = 0.018, p=0.005) but not age (R2 = 0.11, p = 0.09) 

##############
#Multivariable
##############
#Significant variables from univariable analyses: HIV, season, wood, URI. Age is NO LONGER significantly associated; but given known effect of age on microbiome composition, is it appropriate to leave it out of the multivariable model? Will run both versions and d/w MK.

adonis2(BC.dist ~ hiv + season + wood + uri_rec2, data = chi10kmeta, permutations = 1000, by = "margin")
#In a multivariable model [age NOT included], HIV status (P=0.04, R2=0.023), season (P=0.006, R2=0.017), and recent URI (P=0.007, R2=0.016) were a/w differences in overall microbiome composition.  Wood smoke exposure was associated in a univariable model but not a mv model (p=0.13, R2=0.010).

adonis2(BC.dist ~ hiv + season + wood + uri_rec2 + age, data = chi10kmeta, permutations = 1000, by = "margin")

#In a multivariable model [age included], HIV status (P=0.03, R2=0.023), season (P=0.008, R2=0.017), and recent URI (P=0.01, R2=0.016) were a/w differences in overall microbiome composition. Age (p=0.08, R2=0.011) and wood smoke exposure (p=0.12, R2=0.010) were not associated.
```

##### MaAsLin2 models
```{r}
chi10k.filter <- filter_taxa(chi10k, function(Abundance) mean(Abundance)>=50, TRUE)
ntaxa(chi10k.filter) #59 (compared to 43; why more taxa?)

otu <- as.data.frame(otu_table(chi10k.filter, taxa_are_rows=TRUE))
#need to remove the unidentified species from here: row 12, 31, 37, 51, 52, 55
  #source: https://stackoverflow.com/questions/12328056/how-do-i-delete-rows-in-a-data-frame
#corresponding respectively to: Corynebacterium sp. KPL1859, Streptococcus sp. SK643, Paenibacillus sp. P22, Alkalibacterium sp. 20, uncultured Clostridium sp., Nocardioides sp. Root122
otu <- otu[-c(12, 31, 37, 51, 52, 55),]

#Corynebacterium sp. KPL1995 -> Corynebacterium pseudodiphtheriticum
#Corynebacterium sp. KPL1986 -> Corynebacterium accolens
#Haemophilus sp. C1 -> Haemophilus haemolyticus

row.names(otu)[3] <- "Haemophilus haemolyticus"
row.names(otu)[11] <- "Corynebacterium pseudodiphtheriticum"
row.names(otu)[12] <- "Corynebacterium accolens"

#are our variables factors in chi10kmeta? No
class(chi10kmeta$hiv) #character; need to make factor to relevel
chi10kmeta$hiv <- as.factor(chi10kmeta$hiv)
chi10kmeta$hiv <- relevel(chi10kmeta$hiv, ref = "Unexposed")

class(chi10kmeta$wood) #integer; needs to be factor
chi10kmeta$wood <- as.factor(chi10kmeta$wood)
chi10kmeta$wood <- relevel(chi10kmeta$wood, ref = "0")

class(chi10kmeta$season) #character; needs to be factor
chi10kmeta$season <- as.factor(chi10kmeta$season)
chi10kmeta$season <- relevel(chi10kmeta$season, ref = "Dry")

class(chi10kmeta$abx) #character; needs to be factor
chi10kmeta$abx <- as.factor(chi10kmeta$abx)
chi10kmeta$abx <- relevel(chi10kmeta$abx, ref = "No")

class(chi10kmeta$uri_rec2) #character; needs to be factor
chi10kmeta$uri_rec2 <- as.factor(chi10kmeta$uri_rec2)
chi10kmeta$uri_rec2 <- relevel(chi10kmeta$uri_rec2, ref = "No")

#multivariable MaAsLin2 with HUU as reference
fit_data2 = Maaslin2(
  input_data = otu, 
  input_metadata = chi10kmeta, 
  min_prevalence = 0.2,
  max_significance = 0.2,
  output = "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/maaslin_mv_10k_HUUref", 
  fixed_effects = c("hiv", "season", "age", "abx", "wood", "uri_rec2"),
  reference = c("hiv,Unexposed"))

#Maaslin only allows for pairwise comparisons, so we can only compare HUU to CLWH and HUU to HEU. What if we set HEU as the reference to see if anything is differentially abundant between HEU and CLWH?
fit_data2 = Maaslin2(
  input_data = otu, 
  input_metadata = chi10kmeta, 
  min_prevalence = 0.2,
  max_significance = 0.2,
  output = "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/maaslin_mv_10k_HEUref", 
  fixed_effects = c("hiv", "season", "age", "abx", "wood", "uri_rec2"),
  reference = c("hiv,Exposed Uninfected"))
```

- Using the same parameters as we did for our original analysis (filtering at 50 counts, removing unclassified species, q threshold of 0.2, min_prevalence of 0.2), we end up with more taxa in our OTU dataset and differing results of our differential abundance analysis. In our original analysis, C. pseudodiphtheriticum was differentially more abundant in HUU compared to CLWH, but in the current analysis restricted to kids with >10K reads, there were no species differentially abundant between HUU and CLWH. 4 species were differentially abundant between HEU and HUU kids (all less abundant in HEU kids). The species differentially abundant by season (several Moraxella spp) and age (strep perioris and infantis, m. luteus) were consistent with prior model. 

- When we re-ran our Maaslin2 model with HEU as the reference, we did see some consistency with our prior results in that C. aurimucosum, C. accolens, and M. luteus were less abundant in CLWH compared to HEU children, but several other Corynebacterium spp were less abundant in CLWH as well (C. oculi, C. efficiens). Moraxella caviae more abundant in CLWH compared to HEU children.

- Is it sound to adjust filtering threshold so we end up with roughly 40 species and then repeat the analysis? Tried below but not convinced this is sound; d/w MK. 

```{r}
chi10k.filter2 <- filter_taxa(chi10k, function(Abundance) mean(Abundance)>=60, TRUE)
ntaxa(chi10k.filter2)

otu2 <- as.data.frame(otu_table(chi10k.filter2, taxa_are_rows=TRUE))
#need to remove the unidentified species from here: row 10, 23, 35, 36, 38
  #source: https://stackoverflow.com/questions/12328056/how-do-i-delete-rows-in-a-data-frame
#corresponding respectively to: Corynebacterium sp. KPL1859, Streptococcus sp. SK643, Alkalibacterium sp. 20, uncultured Clostridium sp., Nocardioides sp. Root122

otu2 <- otu2[-c(10, 23, 35, 36, 38),]

#Corynebacterium sp. KPL1995 -> Corynebacterium pseudodiphtheriticum
#Corynebacterium sp. KPL1986 -> Corynebacterium accolens
#Haemophilus sp. C1 -> Haemophilus haemolyticus

row.names(otu2)[3] <- "Haemophilus haemolyticus"
row.names(otu2)[9] <- "Corynebacterium pseudodiphtheriticum"
row.names(otu2)[10] <- "Corynebacterium accolens"

#multivariable MaAsLin2 with HUU as reference
fit_data2 = Maaslin2(
  input_data = otu2, 
  input_metadata = chi10kmeta, 
  min_prevalence = 0.2,
  max_significance = 0.2,
  output = "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/maaslin_mv_filt60_10k_HUUref", 
  fixed_effects = c("hiv", "season", "age", "abx", "wood", "uri_rec2"),
  reference = c("hiv,Unexposed"))

#Maaslin only allows for pairwise comparisons, so we can only compare HUU to CLWH and HUU to HEU. What if we set HEU as the reference to see if anything is differentially abundant between HEU and CLWH?
fit_data2 = Maaslin2(
  input_data = otu2, 
  input_metadata = chi10kmeta, 
  min_prevalence = 0.2,
  max_significance = 0.2,
  output = "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/maaslin_mv_filt60_10k_HEUref", 
  fixed_effects = c("hiv", "season", "age", "abx", "wood", "uri_rec2"),
  reference = c("hiv,Exposed Uninfected"))
```

- When we redo our analysis with a higher filtering cutoff (to end up with roughly the same number of spp we did in our original analysis), removing unclassified species, q threshold of 0.2, min_prevalence of 0.2), we end up with differing results of our differential abundance analysis. In our original analysis, C. pseudodiphtheriticum was differentially more abundant in HUU compared to CLWH, but in the current analysis restricted to kids with >10K reads, there were no species differentially abundant between HUU and CLWH. M. oblonga was less abundant in HEU compared to HUU kids. The species differentially abundant by season (several Moraxella spp) and age (strep perioris and infantis, m. luteus) were consistent with prior model. In the model with HEU as reference, C. accolens, C. aurimucosum, and M. luteus were less abundant in CLWH compared to HEU, and M. caviae was more abundant. M. ovis and M. oblonga were more abundant in HUU children and S. aureus was less abundant in HUU children compared to HEU children. 

##### K-medoid clusters 

- NOTE: comes before differential abundance testing in manuscript, but placing after Maaslin2 in code for consistency with prior code

##### Relative abundance plots

- as with original analysis, will make a master melted_df containing all child and sib data with >10K reads, and then subset from this master file to make any necessary relative abundance plots

```{r}
phy.otus <- transform_sample_counts(chisib10K, function(Abundance) Abundance/sum(Abundance))
otu_df <- psmelt(phy.otus)
otu_df$OTU <- as.character(otu_df$OTU)
otu_abundances <- aggregate(otu_df$Abundance, by=list(OTU=otu_df$OTU), FUN=sum)
otu_abundances$x <- (otu_abundances$x)/(nsamples(chisib10K))
names(otu_abundances)[names(otu_abundances) == 'x'] <- 'otu_Ab'
sum(otu_abundances$otu_Ab)  #sums to 1
nrow(otu_abundances)        #6709 species
remove(phy.otus, otu_df)

# Agglomerate taxa into species --> already done, so no need to do again
# Transform to relative abundances
chisib.rel <- transform_sample_counts(chisib10K, function(Abundance) Abundance/sum(Abundance))
sample_sums(chisib.rel)  # This is a sanity check to make sure that relative abundance was calculated for each sample prior to pooling data
#All samples add to 1
melted_df_all <- psmelt(chisib.rel)

#Now create a dataframe of species abundances that we can then arrange in order. Will rename everything that is not the top 10 as "other"
melted_df_all$Species <- as.character(melted_df_all$Species)
species_abundances <- aggregate(melted_df_all$Abundance, by=list(Species=melted_df_all$Species,
                                                           OTU=melted_df_all$OTU), FUN=mean)
species_abundances <- aggregate(species_abundances$x, by=list(Species=species_abundances$Species), FUN=sum)
names(species_abundances)[names(species_abundances) == 'x'] <- 'species_Ab'
sum(species_abundances$species_Ab)    # Should sum to 1 (sum of relative abundances of species)
nrow(species_abundances)      # Corresponds to # of unique species: 6706 ***WHY is this different from 6709?###

#create rule for top 10 species
species_abundances <- arrange(species_abundances, desc(species_Ab))  
TOPSpecies <- unique(species_abundances$Species[1:10])
species_df <- species_abundances[species_abundances$Species %in% TOPSpecies,]
species_df$Species <- factor(species_df$Species, levels = species_df$Species[order(-species_df$species_Ab)])

#rename all non-top 10 species in melted_df as "other" and see if this works
melted_df_all$Species[!(melted_df_all$Species %in% TOPSpecies)] <- "Other"
sum(melted_df_all$Abundance) #sums to 131 exactly; n=131 so tracks?
table(melted_df_all$Species)

#so far so good! all of our top 10 species are consistent
```

- next, will create df just for the children (excluding sibs)

```{r}
#Subset out melted_df from melted_df_all for most of our analyses and sub-analyses: 
melted_df <- subset(melted_df_all, melted_df_all$subject == "child")
table(melted_df$Species)

#To generate our standard color palette for this manuscript: 
library(RColorBrewer)
library(scales)
chisp2 <- c("#293352", "#800000FF", "#767676FF", "#FFA319FF", "#8A9045FF", "#155F83FF", "#C16622FF", "#4D004B", "#8F3931FF", "#58593FFF", "#350E20FF")
show_col(chisp2)

#source for rotating x labels: https://stackoverflow.com/questions/1330989/rotating-and-spacing-axis-labels-in-ggplot2
melted_df$subject_abundances <- (melted_df$Abundance)/(length(unique(melted_df$Sample))) 
class(melted_df$subject_abundances) #numeric variable
summary(melted_df$subject_abundances)
which(is.na(melted_df$subject_abundances)) #No NAs here

#abbreviate HIV status
melted_df$hiv2 <- NA
melted_df$hiv2 [melted_df$hiv == "Infected"] <- "CLWH"
melted_df$hiv2 [melted_df$hiv == "Exposed Uninfected"] <- "HEU children"
melted_df$hiv2 [melted_df$hiv == "Unexposed"] <- "HUU children"
table(melted_df$hiv2)
table(melted_df$hiv)
which(is.na(melted_df$hiv2)) #Nothing missing

#matches up with table(melted_df$hiv)
melted_df$hiv2 <- as.factor(melted_df$hiv2)
melted_df$hiv2 <- reorder(melted_df$hiv2, new.order=c("CLWH", "HEU children", "HUU children"))
melted_df$Species <- as.factor(melted_df$Species)
###FIRST, create figure without re-ordering anything, screenshot it, and then create a new version that is re-ordered and compare the 2. The point of this is to make sure we're not causing any errors in the renaming process

##############
#NO REORDERING
##############
summary(melted_df$subject_abundances)
which(is.na(melted_df$subject_abundances)) #nothing missing

library(ggtext)
raw10K <- ggplot(melted_df[order(melted_df$Species, decreasing = TRUE),], aes(x=hiv2, y=subject_abundances, fill=Species)) +
  geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2)

# #save for comparison as png
# png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig1b_raw.png", width = 9, height = 5, units = 'in', res = 600)
# all_raw
# dev.off()
```

- Now will create our updated barplot with the species reordered

```{r}
#reorder species names
melted_df$Species <- reorder(melted_df$Species, new.order=c("Corynebacterium accolens", "Corynebacterium propinquum", "Corynebacterium pseudodiphtheriticum", "Dolosigranulum pigrum", "Haemophilus influenzae", "Moraxella catarrhalis", "Moraxella lincolnii", "Moraxella nonliquefaciens", "Staphylococcus aureus", "Streptococcus pneumoniae", "Other"))

library(ggtext)
b2 <- ggplot(melted_df[order(melted_df$Species, decreasing = TRUE),], aes(x=hiv2, y=subject_abundances, fill=Species)) +
  geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  xlab("HIV Status") + ylab("Mean relative abundance") 

b2
#this looks right when compared to prior figure 1b and species_df abundances (overall stacked rows eyeball as expected based on overall abundances)

#Create a 10K version of Fig 1 A-B (PCoA plot and rel abundance plot by HIV status)
b12 <- plot_grid(b1, b2, labels = "AUTO", nrow = 1, ncol = 2, align = "h", axis = "b", rel_widths = c(1,1.3))

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig1A-B_10K.pdf", plot = b12, dpi = 300, height = 5, width = 11)

```

**How can we check our numbers?**

There is probably a more efficient way to do this, but for now will create individual dataframes for our top 10 species and compare abundances by our variable of interest (hiv and cd4 status for fig 1, etc) to make sure what we are visualizing in the plot makes sense.

Creating species-specific dataframes:

```{r}
#Corynebacterium accolens
cor_acc <- filter(melted_df, Species == "Corynebacterium accolens")
summary(cor_acc$Abundance)
tapply(cor_acc$Abundance, cor_acc$hiv2, summary)

#Corynebacterium propinquum
cor_p <- filter(melted_df, Species == "Corynebacterium propinquum")
summary(cor_p$Abundance) #matches up with species_df abundance
tapply(cor_p$Abundance, cor_p$hiv2, summary)

#Corynebacterium pseudodiphtheriticum
cor_ps <- filter(melted_df, Species == "Corynebacterium pseudodiphtheriticum")
summary(cor_ps$Abundance)
tapply(cor_ps$Abundance, cor_ps$hiv2, summary)

#Dolosigranulum pigrum
dpig <- filter(melted_df, Species == "Dolosigranulum pigrum")
summary(dpig$Abundance)
tapply(dpig$Abundance, dpig$hiv2, summary)

#Haemophilus influenzae
hflu <- filter(melted_df, Species == "Haemophilus influenzae")
summary(hflu$Abundance)
tapply(hflu$Abundance, hflu$hiv2, summary)

#Moraxella cattarhalis
mcat <- filter(melted_df, Species == "Moraxella catarrhalis")
summary(mcat$Abundance)
tapply(mcat$Abundance, mcat$hiv2, summary) 

#Moraxella lincolnii
mlin <- filter(melted_df, Species == "Moraxella lincolnii")
summary(mlin$Abundance)
tapply(mlin$Abundance, mlin$hiv2, summary) 

#Moraxella nonliquefaciens
mnon <- filter(melted_df, Species == "Moraxella nonliquefaciens")
summary(mnon$Abundance)
tapply(mnon$Abundance, mnon$hiv2, summary)

#Staphylococcus aureus
sa <- filter(melted_df, Species == "Staphylococcus aureus")
summary(sa$Abundance)
tapply(sa$Abundance, sa$hiv2, mean)

#Streptococcus pneumoniae
spneum <- filter(melted_df, Species == "Streptococcus pneumoniae")
summary(spneum$Abundance)
tapply(spneum$Abundance, spneum$hiv2, mean)

#Other
oth <- filter(melted_df, Species == "Other")
summary(oth$Abundance)
aggregate(Abundance ~ Species, data = oth, sum)
#sum of all Other abundances = 16.81235
aggregate(Abundance ~ Species + hiv2, data = oth, sum)
table(chi10kmeta$hiv2)
#CLWH: 4.921470/37 = 0.1330127
#HEU:  7.020649/43 = 0.1632709
#HUU: 4.870231/41 = 0.1187861
```

##### Subanalysis of factors associated with microbiome composition among CLWH

###### Creating new variables

```{r}
hiv = subset_samples(chi10k, hiv == "Infected") #37 participants
chi_hiv <- data.frame(sample_data(hiv))

##############
#NEW VARIABLES
##############
#How many kids receiving TMP-SMX?
table(chi_hiv$child_tmpsmx) #12 kids on TMP-SMX

#Need to make a binary variable for tmpsmx use
chi_hiv$tmpsmx <- NA
chi_hiv$tmpsmx [chi_hiv$child_tmpsmx == 1] <- "Yes"
chi_hiv$tmpsmx[chi_hiv$child_tmpsmx == 2] <- "No"
table(chi_hiv$tmpsmx)

#Don't need new variable, but how many kids had received abx for acute infxn?
table(chi_hiv$abx) #5 kids with abx exposure

#I want the most recent viral load values for my analysis. Will try creating a new variable to ID which value is more recent first:
#missing 3 vl1 and 14vl2's
chi_hiv$newest_vL <- NA
chi_hiv$newest_vL [chi_hiv$child_vl1_weeks < chi_hiv$child_vl2_weeks] <- 1
chi_hiv$newest_vL [chi_hiv$child_vl2_weeks < chi_hiv$child_vl1_weeks] <- 2
chi_hiv$newest_vL [chi_hiv$child_vl2_weeks < 0] <- 1  #we want to include the viral loads collected BEFORE enrollment where possible
chi_hiv$newest_vL [is.na(chi_hiv$child_vl_2)] <- 1    #if we are missing vl_2, we will use vl_1
chi_hiv$newest_vL [is.na(chi_hiv$child_vl_1) & is.na(chi_hiv$child_vl_2)] <- NA
table(chi_hiv$newest_vL)

#Next, can we pull the viral load that was most recent using our newest_vL variable?
#source: https://stackoverflow.com/questions/67944727/creating-new-variable-by-selecting-column-based-on-value-of-another-column
#Need to do this with rows that have values (aka need to exclude the NA viral load children for this to work)
#For future phyloseq analyses: will need to remove B-04, B-07, B-52
chi_viral <- subset(chi_hiv, !is.na(newest_vL))
chi_viral$Var1 <- chi_viral$child_vl_1
chi_viral$Var2 <- chi_viral$child_vl_2

chi_viral <- chi_viral %>% rowwise() %>%
  mutate(newest_vlnum = get(paste0('Var', newest_vL)))
#now we have a dataframe of the 34/37 children with viral load measurements and 1 column containing the most recent viral loads for each
summary(chi_viral$newest_vlnum)
table(chi_viral$newest_vlnum) #23 with viral suppression, 11 without viral suppression

#make new variable for viral suppression or not, with vL < 400 = suppressed
chi_viral$vl_suppr <- NA
chi_viral$vl_suppr [chi_viral$newest_vlnum < 400] <- "1"
chi_viral$vl_suppr [chi_viral$newest_vlnum >= 400] <- "0"
table(chi_viral$vl_suppr)

#REPEAT the above steps to get date associated with most recent viral load, most recent CD4, and date a/w most recent CD4
#Viral load collection date
chi_viral$Var1 <- chi_viral$child_vl1_weeks
chi_viral$Var2 <- chi_viral$child_vl2_weeks

chi_viral <- chi_viral %>% rowwise() %>%
  mutate(newest_vldate = get(paste0('Var', newest_vL)))   #now have info for when vL were checked in relation to enrollment

summary(chi_viral$newest_vldate)

chi_viral <- within(chi_viral, rm(Var1, Var2))

#CD4
#missing CD4 data for B-45 and B-52
#will focus on percentages since this is what is presented for kids
chi_hiv$newest_cd4 <- NA
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks1 < chi_hiv$chi_cd4_weeks2] <- 1
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks2 < chi_hiv$chi_cd4_weeks1] <- 2
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks2 < 0] <- 1  #we want to include the viral loads collected BEFORE enrollment where possible
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_2)] <- 1    #if we are missing cd4_2, we will use cd4_1
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_1)] <- 2    #one child is missing cd4_1 but has cd4_2
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_1) & is.na(chi_hiv$chi_cd4_perc_2)] <- NA
table(chi_hiv$newest_cd4)

#create new dataframe with children missing CD4 data removed
chi_cd4 <- subset(chi_hiv, !is.na(newest_cd4)) #35/37 kids have CD4 measurements
chi_cd4$Var1 <- chi_cd4$chi_cd4_perc_1
chi_cd4$Var2 <- chi_cd4$chi_cd4_perc_2

chi_cd4 <- chi_cd4 %>% rowwise() %>%
  mutate(newest_cd4perc = get(paste0('Var', newest_cd4)))

summary(chi_cd4$newest_cd4perc)
table(chi_cd4$newest_cd4perc)

#make new variable for immune competent CD4 or not, with CD4 percentage >=25% = immune competent
chi_cd4$cd4nL <- NA
chi_cd4$cd4nL [chi_cd4$newest_cd4perc >=25] <- "1"
chi_cd4$cd4nL [chi_cd4$newest_cd4perc < 25] <- "0"
table(chi_cd4$cd4nL)  #7 kids with CD4% < 25

#REPEAT the above steps to get date associated with most recent CD4
#CD4 collection date
chi_cd4$Var1 <- chi_cd4$chi_cd4_weeks1
chi_cd4$Var2 <- chi_cd4$chi_cd4_weeks2

chi_cd4 <- chi_cd4 %>% rowwise() %>%
  mutate(newest_cd4date = get(paste0('Var', newest_cd4)))   #now have info for when cd4 were checked in relation to enrollment

summary(chi_cd4$newest_cd4date)

chi_cd4 <- within(chi_cd4, rm(Var1, Var2))
```

- skipped demographic comparisons by viral suppression and CD4 status for now

###### Alpha diversity
```{r}
#Create dataframe with diversity indices for each specimen using HIV pseq object
diversity1 <- estimate_richness(hiv, measures = c("Shannon", "Simpson", "Chao1", "Observed"))

setDT(diversity1, keep.rownames = TRUE)[]
colnames(diversity1) <- c("sample_id", "Observed", "Chao1", "se.chao1", "Shannon", "Simpson")
rownames(diversity1) <- diversity1$sample_id
diversity <- merge(diversity1, chi_viral, by="sample_id")
div_cd4 <- merge(diversity1, chi_cd4, by="sample_id")
div_all <- merge(diversity1, chi_hiv, by="sample_id")

#################
#OBSERVED SPECIES
#################

#How many species on average per sample?
hist(div_all$Observed)
summary(div_all$Observed) 
#Median (IQR) # spec: 147 (89, 222) among CLWH with >10K reads

#what does it look like by TMP-SMX use? NOTE: N=37
tapply(div_all$Observed, div_all$tmpsmx, summary)   
wilcox.test(Observed ~ tmpsmx, data = div_all)  #P=0.23 with warning (ties)

#what does it look like by viral supp? NOTE: N=34
tapply(diversity$Observed, diversity$vl_suppr, summary)   
wilcox.test(Observed ~ vl_suppr, data = diversity)  #P=0.61 with warning (ties)

#what does it look like by cd4 classification? NOTE: N=35
tapply(div_cd4$Observed, div_cd4$cd4nL, summary)   
wilcox.test(Observed ~ cd4nL, data = div_cd4)  #P=0.85 with warning (ties)

########
#SHANNON
########

#Is our SDI data normally distributed? 
shapiro.test(div_all$Shannon)  #yes! P=0.43
histogram(div_all$Shannon)  #looks good
qqnorm(div_all$Shannon, pch = 1, frame = FALSE)
qqline(div_all$Shannon, col = "steelblue", lwd = 2)

#Shannon diversity median and IQR for all samples
summary(div_all$Shannon)
#Median (IQR) SDI: 1.73 (1.44, 2.08)

#Viral suppression
tapply(diversity$Shannon, diversity$vl_suppr, summary)
t.test(Shannon ~ vl_suppr, data = diversity)        #P=0.97

#TMP-SMX use
tapply(div_all$Shannon, div_all$tmpsmx, summary)
t.test(Shannon ~ tmpsmx, data = div_all)  #P=0.22

#CD4 (continuous and binary)
summary(div <- lm(Shannon ~ newest_cd4perc, data = div_cd4))  #P=0.43
tapply(div_cd4$Shannon, div_cd4$cd4nL, summary)
t.test(Shannon ~ cd4nL, data = div_cd4)  #P=0.86

#Abx use
tapply(div_all$Shannon, div_all$abx, summary)
t.test(Shannon ~ abx, data = div_all)  #P=0.58

#MULTIVARIABLE MODEL
  #viral load
summary(div_mv <- lm(Shannon ~ vl_suppr + tmpsmx + abx, data = diversity))
  #cd4
summary(div_mv <- lm(Shannon ~ cd4nL + tmpsmx + abx, data = div_cd4))

######
#CHAO1
######

#Is our Chao1 data normally distributed? 
shapiro.test(div_all$Chao1)  #no, because p<0.05; also looks skewed on histogram and QQ is wonky
histogram(div_all$Chao1)
qqnorm(div_all$Chao1, pch = 1, frame = FALSE)
qqline(div_all$Chao1, col = "steelblue", lwd = 2)

#What does it look like if we log transform it? Will this allow for parametric testing?
diversity$Chao1_log <- log(diversity$Chao1)
div_all$Chao1_log <- log(div_all$Chao1)
div_cd4$Chao1_log <- log(div_cd4$Chao1)

shapiro.test(div_all$Chao1_log) 
histogram(div_all$Chao1_log)
qqnorm(div_all$Chao1_log, pch = 1, frame = FALSE)
qqline(div_all$Chao1_log, col = "steelblue", lwd = 2)
#Looks better! Can use t.test and linear regression on transformed data

#Chao1 median and IQR for all included CLWH
summary(diversity$Chao1)
#Median (IQR) Chao1: 143 (84, 243)

#Viral suppression
tapply(diversity$Chao1_log, diversity$vl_suppr, summary)
t.test(Chao1_log ~ vl_suppr, data = diversity)        #P=0.81

#TMP-SMX use
tapply(div_all$Chao1_log, div_all$tmpsmx, summary)
t.test(Chao1_log ~ tmpsmx, data = div_all)  #P=0.46

#CD4 (continuous and binary)
summary(div <- lm(Chao1_log ~ newest_cd4perc, data = div_cd4))  #P=0.95
tapply(div_cd4$Chao1_log, div_cd4$cd4nL, summary)
t.test(Chao1_log ~ cd4nL, data = div_cd4)  #P=0.80

#abx 
tapply(div_all$Chao1_log, div_all$abx, summary)
t.test(Chao1_log ~ abx, data = div_all)  #P=0.55

#MULTIVARIABLE MODEL
  #viral load
summary(div_mv <- lm(Chao1_log ~ vl_suppr + tmpsmx + abx, data = diversity))
  #cd4
summary(div_mv <- lm(Chao1_log ~ cd4nL + tmpsmx + abx, data = div_cd4))

#Skipping boxplots
remove(div, div_all, div_cd4, diversity, diversity1)

```

**Summary: No association between CD4 status, viral suppression, TMP-SMX use, and abx exposure and measures of alpha diversity among CLWH with >/=10K reads in our study**

###### Beta diversity: PCoA plots

```{r}
#remove kids missing viral load data: B-04, B-07, B-52 and add chi_viral as sample data (confirmed that these are the same study IDs in 10K pruned data)
vl <- hiv
chi_viral <- as.data.frame(chi_viral)
row.names(chi_viral) <- chi_viral$sample_id
chi_viral <- sample_data(chi_viral)
head(chi_viral)
vl <- merge_phyloseq(vl, chi_viral)
#now see if it worked:
bloop <- data.frame(sample_data(vl))
#not working for some reason on 6/2/22; merged new sample data with pseq FIRST, then removed samples and it worked
#success!
remove(bloop)
vl = subset_samples(vl, sample_id != "B.04.CHI" & sample_id != "B.07.CHI" & sample_id != "B.52.CHI")
chi_viral <- data.frame(sample_data(vl))

#do the same to create a CD4 pseq object to plot
cd4 <- hiv
chi_cd4 <- as.data.frame(chi_cd4)
row.names(chi_cd4) <- chi_cd4$sample_id
chi_cd4 <- sample_data(chi_cd4)
head(chi_cd4)
cd4 <- merge_phyloseq(cd4, chi_cd4)
#now see if it worked:
bloop <- data.frame(sample_data(cd4))
table(bloop$cd4nL)
#success!
remove(bloop)
#missing: B-45 and B-52
cd4 = subset_samples(cd4, sample_id != "B.45.CHI" & sample_id != "B.52.CHI")
chi_cd4 <- data.frame(sample_data(cd4))

#Now transform our pseq objects
vl_clr <- microbiome::transform(vl, "clr")
cd4_clr <- microbiome::transform(cd4, "clr")
# hiv_clr <- microbiome::transform(hiv, "clr")

#PCoA plots#
#VIRAL SUPPRESSION
clr_pcoa <- ordinate(vl_clr, method = "PCoA", distance = "euclidean")
lines <- c("dashed", "dashed")
clr_vl <- plot_ordination(vl_clr, clr_pcoa, color = "vl_suppr", title = "Euclidean PCoA Plot by Viral Suppression") +
  geom_point(size = 2) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values=c('#800000FF','#155F83FF'))+
  stat_ellipse(aes(linetype=vl_suppr), geom="polygon", alpha=0, type="t", level=0.8, size=0.7) +
  xlab("PC1 (16.6%)") + ylab("PC2 (11.7%)") +
  scale_linetype_manual(values=lines)

clr_vl

# png(file="clr_vlsuppr.png",
#     width = 5, height = 4, units = 'in', res=600)
# clr_vl
# dev.off()

#CD4 (binary) using cd4_clr
chi_cd4$cd4cat <- NA
chi_cd4$cd4cat [chi_cd4$cd4nL=="1"] <- "Normal CD4"
chi_cd4$cd4cat [chi_cd4$cd4nL=="0"] <- "Low CD4"
table(chi_cd4$cd4cat)
class(chi_cd4$cd4cat)
chi_cd4$cd4cat <- as.factor(chi_cd4$cd4cat)
chi_cd4$cd4cat <- reorder(chi_cd4$cd4cat, new.order=c("Low CD4", "Normal CD4"))

#Import back into pseq (source: https://jacobrprice.github.io/2017/08/26/phyloseq-to-vegan-and-back.html)
sample_data(cd4_clr) <- as.data.frame(chi_cd4)
#now see if it worked:
bloop <- data.frame(sample_data(cd4_clr))
table(bloop$cd4cat)
#success!
remove(bloop)

clr_pcoa <- ordinate(cd4_clr, method = "PCoA", distance = "euclidean")
lines <- c("dashed", "dashed")
b3 <- plot_ordination(cd4_clr, clr_pcoa, color = "cd4cat") +
  geom_point(size = 2) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values=c('#800000FF','#155F83FF'))+
  stat_ellipse(aes(linetype=cd4cat), geom="polygon", alpha=0, type="t", level=0.8, size=0.7) +
  xlab("PC1 (16.3%)") + ylab("PC2 (11.4%)") +
  scale_linetype_manual(values=lines)
  
# png(file="clr_cd4nl.png",
#     width = 6, height = 4, units = 'in', res=600)
b3
# dev.off()

```

###### Beta diversity: Permanova

```{r}
#PERMANOVA: viral load & TMP-SMX
otu <- as.data.frame(otu_table(vl_clr, taxa_are_rows=TRUE))
#Need species in columns and observations in rows: transpose using t() function and save as a dataframe
#Source: https://github.com/edamame-course/2015-tutorials/blob/master/final/Ashley_Intro_to_R.md
otu_trans <- data.frame(t(otu))
chi_viral <- data.frame(sample_data(vl_clr))
BC.dist=vegdist(otu_trans, method="euclidean")

#viral suppression + age
adonis2(BC.dist ~ vl_suppr + age, data = chi_viral, permutations = 1000)
#No significant difference by viral suppression status among children (P = 0.89) or age (P=0.20)

#PERMANOVA: CD4, TMPSMX, abx
otu <- as.data.frame(otu_table(cd4_clr, taxa_are_rows=TRUE))
#Need species in columns and observations in rows: transpose using t() function and save as a dataframe
#Source: https://github.com/edamame-course/2015-tutorials/blob/master/final/Ashley_Intro_to_R.md
otu_trans <- data.frame(t(otu))
chi_cd4 <- data.frame(sample_data(cd4_clr))
BC.dist=vegdist(otu_trans, method="euclidean")

#CD4 (binary) + age
adonis2(BC.dist ~ cd4nL + age, data = chi_cd4, permutations = 1000) #P=0.08, R = 0.039 for CD4; age P = 0.21 and R2 = 0.034

#TMPSMX use + age
adonis2(BC.dist ~ tmpsmx + age, data = chi_cd4, permutations = 1000) #P=0.41, R2=0.030

#abx with age
adonis2(BC.dist ~ abx + age, data = chi_cd4, permutations = 1000) #P=0.61, R2=0.026

#clean up
remove(hiv_clr, cd4_clr, vl_clr, otu, otu_trans, clr_cd4, clr_pcoa, clr_tmpsmx, clr_vl)
remove(BC.dist, lines)
```

**Summary: No difference in overall microbiome composition by viral suppression status, TMP-SMX use, or abx exposure among CLWH. The PCoA plot of CD4 classification showed clustering by CD4, but this was not statistically significant by PERMANOVA (P=0.08, R^2^ = 0.039)**

###### Relative abundance plot by CD4 classification

```{r}
#create a subsetted df of melted_df that only has CLWH
melt_cd4 <- subset(melted_df, melted_df$hiv == "Infected")
table(melt_cd4$Species)
sum(melt_cd4$Abundance) #confirmed n=37

#create our CD4 variables in that dataframe
melt_cd4$newest_cd4 <- NA
melt_cd4$newest_cd4 [melt_cd4$chi_cd4_weeks1 < melt_cd4$chi_cd4_weeks2] <- 1
melt_cd4$newest_cd4 [melt_cd4$chi_cd4_weeks2 < melt_cd4$chi_cd4_weeks1] <- 2
melt_cd4$newest_cd4 [melt_cd4$chi_cd4_weeks2 < 0] <- 1  #we want to include the cd4 collected BEFORE enrollment where possible
melt_cd4$newest_cd4 [is.na(melt_cd4$chi_cd4_perc_2)] <- 1    #if we are missing cd4_2, we will use cd4_1
melt_cd4$newest_cd4 [is.na(melt_cd4$chi_cd4_perc_1)] <- 2    #one child is missing cd4_1 but has cd4_2
melt_cd4$newest_cd4 [is.na(melt_cd4$chi_cd4_perc_1) & is.na(melt_cd4$chi_cd4_perc_2)] <- NA
table(melt_cd4$newest_cd4)

#create new dataframe without B-45 and B-52 (aka new df with children missing CD4 data removed)
melt_cd4 <- subset(melt_cd4, !is.na(newest_cd4))
#confirm sum
sum(melt_cd4$Abundance)
melt_cd4$Var1 <- melt_cd4$chi_cd4_perc_1
melt_cd4$Var2 <- melt_cd4$chi_cd4_perc_2

melt_cd4 <- melt_cd4 %>% rowwise() %>%
  mutate(newest_cd4perc = get(paste0('Var', newest_cd4)))

summary(melt_cd4$newest_cd4perc)
table(melt_cd4$newest_cd4perc)

#make new variable for immune competent CD4 or not, with CD4 percentage >=25% = immune competent
melt_cd4$cd4nL <- NA
melt_cd4$cd4nL [melt_cd4$newest_cd4perc >=25] <- "1"
melt_cd4$cd4nL [melt_cd4$newest_cd4perc < 25] <- "0"
table(melt_cd4$cd4nL)  #11 kids with CD4% < 25 (when you divide sum by 6709 rows per participant)

#New variable for CD4 cutoff
melt_cd4$cd4cat <- NA
melt_cd4$cd4cat [melt_cd4$cd4nL==1] <- "Normal CD4"
melt_cd4$cd4cat [melt_cd4$cd4nL==0] <- "Low CD4"
table(melt_cd4$cd4cat)
class(melt_cd4$cd4cat)
melt_cd4$cd4cat <- as.factor(melt_cd4$cd4cat)
melt_cd4$cd4cat <- reorder(melt_cd4$cd4cat, new.order=c("Low CD4", "Normal CD4"))
```

- Now recreate Figure 1D with the reformatted labels
```{r}
b4 <- ggplot(melt_cd4[order(melt_cd4$Species, decreasing = TRUE),], aes(x=cd4cat, y=subject_abundances, fill=Species)) +
  geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  xlab("HIV Status") + ylab("Mean relative abundance")

b4

fig1_10k <- plot_grid(b1, b2, b3, b4, labels = "AUTO", nrow = 2, ncol = 2, align = "h", axis = "b", rel_widths = c(1,1.3))

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig1_10K_subanalysis.pdf", plot = fig1_10k, dpi = 300, height = 9, width = 11)

#save as png too for ease of opening
png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig1_10K_subanalysis.png",
    width = 11, height = 9, units = 'in', res = 600)
fig1_10k
dev.off()
```

Will check our figure against the mean relative abundances of each species

```{r}
cor_acc <- filter(melt_cd4, Species == "Corynebacterium accolens")
summary(cor_acc$Abundance)
tapply(cor_acc$Abundance, cor_acc$cd4cat, mean)

#Corynebacterium propinquum
cor_p <- filter(melt_cd4, Species == "Corynebacterium propinquum")
tapply(cor_p$Abundance, cor_p$cd4cat, mean)

#Corynebacterium pseudodiphtheriticum
cor_ps <- filter(melt_cd4, Species == "Corynebacterium pseudodiphtheriticum")
tapply(cor_ps$Abundance, cor_ps$cd4cat, mean)

#Dolosigranulum pigrum
dpig <- filter(melt_cd4, Species == "Dolosigranulum pigrum")
tapply(dpig$Abundance, dpig$cd4cat, mean)

#Haemophilus influenzae
hflu <- filter(melt_cd4, Species == "Haemophilus influenzae")
tapply(hflu$Abundance, hflu$cd4cat, mean)

#Moraxella cattarhalis
mcat <- filter(melt_cd4, Species == "Moraxella catarrhalis")
tapply(mcat$Abundance, mcat$cd4cat, mean) 

#Moraxella lincolnii
mlin <- filter(melt_cd4, Species == "Moraxella lincolnii")
tapply(mlin$Abundance, mlin$cd4cat, mean) 

#Moraxella nonliquefaciens
mnon <- filter(melt_cd4, Species == "Moraxella nonliquefaciens")
tapply(mnon$Abundance, mnon$cd4cat, mean)

#Staphylococcus aureus
sa <- filter(melt_cd4, Species == "Staphylococcus aureus")
tapply(sa$Abundance, sa$cd4cat, mean)

#Streptococcus pneumoniae
spneum <- filter(melt_cd4, Species == "Streptococcus pneumoniae")
tapply(spneum$Abundance, spneum$cd4cat, mean)

#Other
oth <- filter(melt_cd4, Species == "Other")
summary(oth$Abundance)
aggregate(Abundance ~ Species + cd4cat, data = oth, sum)
table(chi_cd4$cd4cat)
#Low CD4: 0.9770166/7 = 0.1395738
#Normal CD4: 3.6488923/28 = 0.1303176
```

##### Differential abundance analyses: LefSe

- will filter at 50, remove unidentified species from the otu table, import back into pseq object, and re-run as we did before, using the MicrobiomeMarker package

- microbiomeMarker tutorial: https://github.com/yiluheihei/microbiomeMarker/blob/master/vignettes/microbiomeMarker-vignette.Rmd (under differential analysis section) AND https://yiluheihei.github.io/microbiomeMarker/reference/run_lefse.html

```{r}
library(microbiomeMarker)
cd4.filter <- filter_taxa(cd4, function(Abundance) mean(Abundance)>=50, TRUE)
ntaxa(cd4.filter) #50

otu <- as.data.frame(otu_table(cd4.filter, taxa_are_rows=TRUE))
#need to remove the unidentified species from here: row 11, 26, 31, 43, 45
  #source: https://stackoverflow.com/questions/12328056/how-do-i-delete-rows-in-a-data-frame
#corresponding respectively to: Corynebacterium sp. KPL1859, Streptococcus sp. SK643,  Paenibacillus sp. P22, Clostridiales bacterium KA00274, Prevotellaceae bacterium Marseille-P2826
otu <- otu[-c(11, 26, 31, 43, 45),]

#Corynebacterium sp. KPL1995 -> Corynebacterium pseudodiphtheriticum
#Corynebacterium sp. KPL1986 -> Corynebacterium accolens
#Haemophilus sp. C1 -> Haemophilus haemolyticus

row.names(otu)[3] <- "Haemophilus haemolyticus"
row.names(otu)[10] <- "Corynebacterium pseudodiphtheriticum"
row.names(otu)[11] <- "Corynebacterium accolens"

#Kept getting error messages when I tried to import the new OTU table into cd4.filter; ?maybe because otu table and taxtable were now mismatched. So cleared out the relevant rows from the taxtable and then made a new pseq object: cd4.filter2
  #source: https://github.com/joey711/phyloseq/issues/1018
bloop <- data.frame(sample_data(cd4.filter))
taxtable <- as.data.frame(as(tax_table(cd4.filter),"matrix"),stringsAsFactors=FALSE)
taxtable <- taxtable[-c(11, 26, 31, 43, 45),]
rownames(taxtable) <- rownames(otu)

otumat = otu_table(as.matrix(otu), taxa_are_rows=T)
taxmat = tax_table(as.matrix(taxtable))
bloop <- sample_data(bloop)

cd4.filter2 = phyloseq(otumat, taxmat, bloop)
#Make sure it worked --> success!
otu <- as.data.frame(otu_table(cd4.filter2, taxa_are_rows=TRUE))
bloop <- data.frame(sample_data(cd4.filter2))
table(bloop$cd4nL)

#Now we can run lefse on our filtered object
cd4lefse.f <- run_lefse(
  cd4.filter2,
  group = "cd4nL",
  subgroup = NULL,
  taxa_rank = "all",
  kw_cutoff = 0.01,
  wilcoxon_cutoff = 0.01,
  multigrp_strat = FALSE,
  lda_cutoff = 2
)

#Keep getting the following error: "Error in sig_otus[, wilcoxon_p] : incorrect number of dimensions." Re-created the phyloseq object and confirmed otus and taxtable matched.
  #what if we run lefse with our filtered CD4 pseq object without unidentified species removed? Get same error message. But if we run on our unfiltered CD4 object, it works

cd4lefse.f <- run_lefse(
  cd4.filter,
  wilcoxon_cutoff = 0.01,
  group = "cd4nL",
  kw_cutoff = 0.01,
  multigrp_strat = TRUE,
  taxa_rank = "Species",
  lda_cutoff = 2
)

test <- run_lefse(
  cd4,
  wilcoxon_cutoff = 0.01,
  group = "cd4nL",
  kw_cutoff = 0.01,
  multigrp_strat = TRUE,
  taxa_rank = "Species",
  lda_cutoff = 2
)

test
head(marker_table(test))

test.f <- as.data.frame(as(marker_table(test),"matrix"),stringsAsFactors=FALSE)

#reviewed lefse tutorial and changed taxa_rank from "Species" to 'all' for our cd4lefse.f using cd4.filter2 and got results: N. lactamica is differentially abundant and is more abundant in children with normal CD4. This is the OPPOSITE of what we previously found (N. lactamica was more abundant in CLWH with low CD4)
cd4lefse.f
head(marker_table(cd4lefse.f))

cd4lefse.f <- as.data.frame(as(marker_table(cd4lefse.f),"matrix"),stringsAsFactors=FALSE)
write.csv(cd4lefse.f, '/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/mm_lefse_cd4_filt_10K.csv')
```

- results of this LefSe analysis are contradictory to our prior results: N. lactamica was more abundant in children with normal CD4. This is the OPPOSITE of what we previously found (N. lactamica was more abundant in CLWH with low CD4)