---
title: "HIV Microbiome Analyses - Patel et al"
author: "Sweta Patel"
date: "5/15/2023"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


**General order of steps**
1. Basic demographics
2. Alpha diversity analyses
3. Transform data (CLR) and do Beta-diversity analyses (PCoA plots, PERMANOVA)
  #UPDATE 5/5: the order covariables are listed in PERMANOVA affects the results
4. Generate relative abundance plots using NON-transformed data
5. UPDATE 5/21: Filter on mean relative abundance of 50 and remove NOS species
6. Differential abundance analyses using Maaslin and FILTERED data
    #set Q value to 0.1-0.2
7. Logistic regression analyses (Aim 2)

*Note: these analyses utilize a pruned and decontaminated phyloseq object containing a cleaned taxtable and new variables created from the original variables recorded in REDCap. All coding can be found in the "Fog KAIJU 10-6-23" file.*

```{r include=FALSE}
#Loading packages 
library(plyr)
library(dplyr)
library(ggplot2)
library(phyloseq)
library(tidyr)
library(vegan)
library(metagenomeSeq)
library(httr)
library(gridExtra)
library(data.table)
library(RColorBrewer)
library(scales)
library(gsubfn)
library(readxl)
library(microbiome)
library(MASS)
library(Maaslin2)
library(cowplot)
library(cluster)
library(devtools)
library(SpiecEasi)
library(pairwiseAdonis)
```

```{r include=FALSE}
set.seed(1234)
#can't set wd in RMD file; entered below code into console directly
# setwd("OneDrive - Duke University/Fogarty coding/Sequencing/")
```

### Importing and cleaning data

```{r, results = FALSE}
pruned <- readRDS("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/phy.pruned05092022.rds")
#confirm the taxtable, etc looks good:
test <- as.data.frame(as(tax_table(pruned),"matrix"),stringsAsFactors=FALSE)  #success!
meta_prune <- data.frame(sample_data(pruned))
meta_prune <- within(meta_prune, rm(sib_weight, sib_sex, sib_height, sib_muac, sib_race, sib_bfeed_1,
                                  sib_bfeed_2, sib_uri_recent, sib_uri_current, sib_pcr_1, sib_pcr_2,
                                  sib_bcg, sib_hepb, sib_clinic, sib_clinic_dx___5, sib_clinic_dx_oth,
                                  sib_hosp, sib_meds, sib_meds_name, sibmo, sibyr, sib_vl1_weeks, sib_vl_weeks2,
                                  sib_med_days, sib_dpt, sib_pcv, sib_rota, sib_polio, sib_measles, sib_age, samp_storage,
                                  enr_clinic, enr_clinic_other, mat_np))
remove(test)
```

### Demographics of participants with sequencing results

N= 272 participants with sequencing results (of 309 enrolled participants)

```{r}
child <- subset(meta_prune, subject == "child") #n=143
table(child$hiv)  #50 HUU kids, 49 HEU kids, 44 HIV+ kids

#Age
hist(child$age)
summary(child$age)
tapply(child$age, child$hiv, summary)
kruskal.test(child$age, child$hiv)
table(child$age)

#Sex
table(child$sex)  #74 girls
table(child$sex, child$hiv)
prop.table(table(child$sex, child$hiv), 2)
summary(table(child$sex, child$hiv))

#Maternal age
hist(child$mat_age2)  #looks quite normal, but will still present median and IQR for consistency
summary(child$mat_age2)
tapply(child$mat_age2, child$hiv, summary)
kruskal.test(child$mat_age2, child$hiv)

#Season
table(child$season) 
table(child$season, child$hiv)
prop.table(table(child$season, child$hiv), 2)
summary(table(child$season, child$hiv))

#Maternal education
table(child$mat_educ)
table(child$mat_educ2)
table(child$mat_educ2, child$hiv)
prop.table(table(child$mat_educ2, child$hiv), 2)
summary(table(child$mat_educ2, child$hiv))
fisher.test(child$mat_educ2, child$hiv)

#Electricity
table(child$elec) #106 children have electricity
table(child$elec, child$hiv)
prop.table(table(child$elec, child$hiv), 2)
summary(table(child$elec, child$hiv))

#Wood
table(child$wood) #90 children in households using wood
table(child$wood, child$hiv)
prop.table(table(child$wood, child$hiv), 2)
summary(table(child$wood, child$hiv))

#Num household members
hist(child$hhsize)
summary(child$hhsize)
tapply(child$hhsize, child$hiv, summary)
kruskal.test(child$hhsize, child$hiv)

#Num children <5
hist(child$num_child)
summary(child$num_child)
tapply(child$num_child, child$hiv, summary)
kruskal.test(child$num_child, child$hiv)

#antibiotics
table(child$abx) #24 children received abx
table(child$abx, child$hiv)
prop.table(table(child$abx, child$hiv), 2)
summary(table(child$abx, child$hiv))

#Clinic in past 3 months
table(child$clinic)
table(child$clinic, child$hiv)
prop.table(table(child$clinic, child$hiv), 2)
summary(table(child$clinic, child$hiv))

#URI in past 1 month
table(child$uri_recent)
table(child$uri_recent, child$hiv)
prop.table(table(child$uri_recent, child$hiv), 2)
summary(table(child$uri_recent, child$hiv))

#URI currently (don't need both this and uri recent in table 1; need to choose)
table(child$uri_current)
table(child$uri_current, child$hiv)
prop.table(table(child$uri_current, child$hiv), 2)
summary(table(child$uri_current, child$hiv))

#Hospital in past 3 months -->> 4 children hospitalized, all 4 HIV+
  #*MENTION IN MANUSCRIPT
table(child$hosp)
table(child$hosp, child$hiv)
table(child$hosp_dx)  #1 with lower respiratory infx, 1 with gastroenteritis, 2 with other
table(child$hosp_dx_oth)  #malnutrition, sepsis + chronic suppurative OM

#PCV doses
table(child$pcv)
table(child$pcv, child$hiv)
prop.table(table(child$pcv, child$hiv), 2)
summary(table(child$pcv, child$hiv))
fisher.test(child$pcv, child$hiv)

#HiB
table(child$dpt)
table(child$dpt, child$hiv)
prop.table(table(child$dpt, child$hiv), 2)
summary(table(child$dpt, child$hiv))
fisher.test(child$dpt, child$hiv)

#breakdown of subjects (not for table 1, but for sequencing figure)
table(meta_prune$subject)

#Lancet Microbe reviewer requesting breakdown of children by clinic enrollment; need to pull this data from prior deidentified csv file:
clinic <- read.csv("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/metadata_clean_10-7-21.csv")
table(clinic$enr_clinic) #need to whittle this down to only the children with sequencing data 
#dataframe with only study id, enr_clinic, and enr_clinic_other
test <- clinic[c("study_id", "enr_clinic", "enr_clinic_other")]
#standardize study_id
test$study_id <- gsubfn("-", ".", test$study_id)
#merge
clin2 <- merge(child, test, by = "study_id")
table(clin2$enr_clinic) #1 = BBCOE; 2 = Mogoditshane; 3 = Old Naledi; 4 = Lesirane; 9 = Other; and all other = Mochudi
table(clin2$enr_clinic_other)
table(clin2$enr_clinic, clin2$hiv)
prop.table(table(clin2$enr_clinic, clin2$hiv), 2)
#Created supplementary table with this information

#Additional requests from LM reviewer: breastfeeding and growth status by HIV 
#breastfeeding
table(child$bfeed_1) #49 kids exclusively breastfed, 59 formula fed, and 35 received both breastmilk and formula
table(child$bfeed_1, child$hiv)
prop.table(table(child$bfeed_1, child$hiv), 2)
# summary(table(child$bfeed_1, child$hiv))
fisher.test(child$bfeed_1, child$hiv)

#nutrition: will use mid upper arm circumference (MUAC) as surrogate
###why missing MUAC for B-23 and E-23?
hist(child$muac)
summary(child$muac)
tapply(child$muac, child$hiv, summary)
kruskal.test(child$muac, child$hiv)

#CLWH have lower median muac (15) compared to HEU (15.5) or HUU (15.85) kids; p <0.001 by K-W test. What if we look at it by nutrition category?
child$muaccat <- NA
child$muaccat [child$muac < 12.5] <- "GAM"
child$muaccat [child$muac >= 12.5] <-  "Normal"
table(child$muaccat)
#only 1 child with general acute malnutrition, no kids with severe malnutrition by MUAC

###Asked MK about growth percentile with available variables 9/19; need to calculate Z scores for height and weight.
#R package available to calculate 
#source for zscorer: https://github.com/nutriverse/zscorer
library(zscorer)
#sex needs to be coded as male (1) and female (2)
child$sex2 <- NA
child$sex2 [child$sex == 0] <- 1
child$sex2 [child$sex == 1] <- 2
table(child$sex)
table(child$sex2) #matches up
#age needs to be in days, not months
child$age_d <- child$age*(30.4375) #confirmed that B-01 age and age_d match up

#weight for age
child2 <- addWGSR(data=child, sex="sex2", firstPart = "weight", secondPart = "age_d", index="wfa")
summary(child2$wfaz)  #ERROR: weight for U21 listed as 80.5 kg, corresponding to a z score of 37.6. Likely an error; will double check data entry with Charity
#in the meantime, drop U21 from our wfa z score calculations
child3 <- subset(child2, study_id !="U.21")
child3 <- addWGSR(data=child3, sex="sex2", firstPart = "weight", secondPart = "age_d", index="wfa")
summary(child3$wfaz)
hist(child3$wfaz) #roughly normal, slight left skew
tapply(child3$wfaz, child3$hiv, summary)
kruskal.test(child3$wfaz, child3$hiv)

#height for age
child2 <- addWGSR(data=child, sex="sex2", firstPart = "height", secondPart = "age_d", index="hfa")
summary(child2$hfaz)
hist(child2$hfaz) #roughly normal, slight left skew
tapply(child2$hfaz, child2$hiv, summary)
kruskal.test(child2$hfaz, child2$hiv)

#CLWH have significantly lower wfaz and hfaz compared to HEU and HUU children.

#Reviewer also asked about duration since abx, which we have already as a separate variable.
table(child$meds_name)
table(child$abx)
#generating summary statistics of subset of data: https://stackoverflow.com/questions/4835830/r-language-how-do-i-print-see-summary-statistics-for-sample-subset
summary(subset(child, abx == "Yes", select=med_days)) #among children who received abx, median (IQR) duration between abx Rx and enrollment = 35 (20, 62) days
```

### HIV-specific variables
```{r}
chi_hiv <- subset(child, hiv == "Infected")
#restrict to just HIV variables for easier cleaning and analysis
chi_hiv <- chi_hiv[c("child_pcr_1", "child_vl_1", "child_pcr_2", "child_vl_2", "chi_cd4_num_1", 
                     "chi_cd4_perc_1", "chi_cd4_num_2", "chi_cd4_perc_2", "chi_cd4_weeks1", 
                     "chi_cd4_weeks2", "child_vl1_weeks", "child_vl2_weeks",
                     "child_arv", "child_arv_meds", "child_tmpsmx", "abx")]

#How many kids receiving TMP-SMX? NOTE: we did not include these kids in the abx category. Need to re-do analyses with this included?
table(chi_hiv$child_tmpsmx) #15 kids on TMP-SMX
#manual inspection: 10 kids on TMP-SMX without other abx exposure

#I want the most recent viral load values for my analysis. Will try creating a new variable to ID which value is more recent first:
#missing 3 vl1 and 14vl2's
chi_hiv$newest_vL <- NA
chi_hiv$newest_vL [chi_hiv$child_vl1_weeks < chi_hiv$child_vl2_weeks] <- 1
chi_hiv$newest_vL [chi_hiv$child_vl2_weeks < chi_hiv$child_vl1_weeks] <- 2
chi_hiv$newest_vL [chi_hiv$child_vl2_weeks < 0] <- 1  #we want to include the viral loads collected BEFORE enrollment where possible
chi_hiv$newest_vL [is.na(chi_hiv$child_vl_2)] <- 1    #if we are missing vl_2, we will use vl_1
chi_hiv$newest_vL [is.na(chi_hiv$child_vl_1) & is.na(chi_hiv$child_vl_2)] <- NA
table(chi_hiv$newest_vL)

#Next, can we pull the viral load that was most recent using our newest_vL variable?
  #source: https://stackoverflow.com/questions/67944727/creating-new-variable-by-selecting-column-based-on-value-of-another-column
  #Need to do this with rows that have values (aka need to exclude the NA viral load children for this to work)
#For future phyloseq analyses: will need to remove B-04, B-07, B-52
chi_viral <- subset(chi_hiv, !is.na(newest_vL))
chi_viral$Var1 <- chi_viral$child_vl_1
chi_viral$Var2 <- chi_viral$child_vl_2

chi_viral <- chi_viral %>% rowwise() %>%
  mutate(newest_vlnum = get(paste0('Var', newest_vL)))
#now we have a dataframe of the 41 children with viral load measurements and 1 column containing the most recent viral loads for each
summary(chi_viral$newest_vlnum)
table(chi_viral$newest_vlnum) #27 with viral suppression, 14 without viral suppression

#make new variable for viral suppression or not, with vL < 400 = suppressed
chi_viral$vl_suppr <- NA
chi_viral$vl_suppr [chi_viral$newest_vlnum < 400] <- "1"
chi_viral$vl_suppr [chi_viral$newest_vlnum >= 400] <- "0"
table(chi_viral$vl_suppr)

#REPEAT the above steps to get date associated with most recent viral load, most recent CD4, and date a/w most recent CD4
#Viral load collection date
chi_viral$Var1 <- chi_viral$child_vl1_weeks
chi_viral$Var2 <- chi_viral$child_vl2_weeks

chi_viral <- chi_viral %>% rowwise() %>%
  mutate(newest_vldate = get(paste0('Var', newest_vL)))   #now have info for when vL were checked in relation to enrollment

summary(chi_viral$newest_vldate)

chi_viral <- within(chi_viral, rm(Var1, Var2))

#CD4
  #missing CD4 data for B-45 and B-52
  #will focus on percentages since this is what is presented for kids
chi_hiv$newest_cd4 <- NA
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks1 < chi_hiv$chi_cd4_weeks2] <- 1
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks2 < chi_hiv$chi_cd4_weeks1] <- 2
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks2 < 0] <- 1  #we want to include the viral loads collected BEFORE enrollment where possible
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_2)] <- 1    #if we are missing cd4_2, we will use cd4_1
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_1)] <- 2    #one child is missing cd4_1 but has cd4_2
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_1) & is.na(chi_hiv$chi_cd4_perc_2)] <- NA
table(chi_hiv$newest_cd4)

#create new dataframe without B-45 and B-52 (aka new df with children missing CD4 data removed)
chi_cd4 <- subset(chi_hiv, !is.na(newest_cd4))
chi_cd4$Var1 <- chi_cd4$chi_cd4_perc_1
chi_cd4$Var2 <- chi_cd4$chi_cd4_perc_2

chi_cd4 <- chi_cd4 %>% rowwise() %>%
  mutate(newest_cd4perc = get(paste0('Var', newest_cd4)))

summary(chi_cd4$newest_cd4perc)
table(chi_cd4$newest_cd4perc)

#make new variable for immune competent CD4 or not, with CD4 percentage *** = immune competent
  #See sub-analyses file for code

#arvs
table(chi_hiv$child_arv)
table(chi_hiv$child_arv_meds)
```

### Number of reads by subject class and HIV status

*using pruned and agglomerated reads*

```{r}
summary(meta_prune$paired.reads)   #Median (IQR) of 73,166 (29,605; 217,115) reads per sample
#mean 183,726 reads per sample
hist(meta_prune$paired.reads)

#Did maternal samples contain more reads than child or sib reads? --> the opposite! Why?
#children = petri dishes
tapply(meta_prune$paired.reads, meta_prune$subject, summary)
kruskal.test(meta_prune$paired.reads, meta_prune$subject)

#number of reads
hist(child$paired.reads)
summary(child$paired.reads)
tapply(child$paired.reads, child$hiv, summary)
kruskal.test(paired.reads ~ hiv, data = child) 

#Sum of sequences
sum(sample_sums(pruned))     #35,101,748 sequences obtained 
mean(sample_sums(pruned))    #Mean of 129,050.5 sequences per sample
median(sample_sums(pruned))  #Median of 45,235 sequences per sample
summary(sample_sums(pruned)) #IQR of sequencing reads per sample: 15730, 146428
nsamples(pruned)        #272 NP samples 
ntaxa(pruned) #6709 taxa
```

### Alpha diversity measures

*Create dataframe*

```{r}
#Create dataframe with diversity indices for each specimen using PRUNED pseq object
diversity <- estimate_richness(pruned, measures = c("Shannon", "Simpson", "Chao1", "Observed"))

setDT(diversity, keep.rownames = TRUE)[]
colnames(diversity) <- c("sample_id", "Observed", "Chao1", "se.chao1", "Shannon", "Simpson")
rownames(diversity) <- diversity$sample_id
diversity <- merge(diversity, meta_prune, by="sample_id")
```

#### Observed species

```{r}
#How many species on average per sample? Did that differ by subject or by HIV status (among children only)?
hist(diversity$Observed)
summary(diversity$Observed) 
sd(diversity$Observed)
#Median (IQR) # spec: 106 (53, 260)
#Mean (SD): 216 (300)

#what does it look like by subject? Moms have significantly fewer species per sample
tapply(diversity$Observed, diversity$subject, summary)   
kruskal.test(Observed ~ subject, data = diversity)

#what about by HIV status of the child? No signif diff
div_child <- subset(diversity, subject == "child")
tapply(div_child$Observed, div_child$hiv, summary)
kruskal.test(Observed ~ hiv, data = div_child) 

#Export study IDs for SRA file:
sra <- as.data.frame(div_child$sample_id, div_child$month, div_child$year)
sra <- subset(div_child, select = c("sample_id", "month", "year"))
write.csv(sra, '/Users/swetapatel/OneDrive - Duke University/Fogarty coding/sample_ids.csv')
```

#### Shannon

```{r}
#Is our SDI data normally distributed? 
shapiro.test(diversity$Shannon)  #technically no, because p<0.05; looks pretty good on histogram but QQ is wonky
histogram(diversity$Shannon)
qqnorm(diversity$Shannon, pch = 1, frame = FALSE)
qqline(diversity$Shannon, col = "steelblue", lwd = 2)

#Shannon diversity median and IQR for all samples
summary(diversity$Shannon)
sd(diversity$Shannon)
#Median (IQR) SDI: 1.85 (1.51, 2.26)
#Mean (SD) 1.93 (0.66)  

#what does it look like by subject? 
tapply(diversity$Shannon, diversity$subject, summary)
kruskal.test(Shannon ~ subject, data = diversity) #child signif lower (p=0.01)
#Since our data is normal-ish in distribution, what does the ANOVA look like?
subj.aov <- aov(Shannon ~ subject, data = diversity)
summary(subj.aov)   #P value 0.007; SIGNIFICANT
table(diversity$subject)

#what about by HIV status of the child?
tapply(div_child$Shannon, div_child$hiv, summary)
kruskal.test(Shannon ~ hiv, data = div_child)   #no significant difference by HIV status using KW test (P=0.71)
  #ANOVA
hiv.aov <- aov(Shannon ~ hiv, data = div_child)
summary(hiv.aov)  #P=0.37, not significant

#Age of the child
div_age <- lm(Shannon ~ age, data = div_child)
summary(div_age)  #p=0.98

#Wood smoke exposure
tapply(div_child$Shannon, div_child$wood, summary)
wilcox.test(Shannon ~ wood, data = div_child)   #p=0.30
t.test(Shannon ~ wood, data = div_child)        #P=0.73

#Season
tapply(div_child$Shannon, div_child$season, summary)
wilcox.test(Shannon ~ season, data = div_child)   #p=0.69
t.test(Shannon ~ season, data = div_child)  #P=0.70

#Receipt of abx in prior 3 months 
tapply(div_child$Shannon, div_child$abx, summary)
wilcox.test(Shannon ~ abx, data = div_child)   #p=0.86
t.test(Shannon ~ abx, data = div_child)   #P=0.71

#Current URI
tapply(div_child$Shannon, div_child$uri_cur2, summary)
wilcox.test(Shannon ~ uri_cur2, data = div_child) #p=0.65
t.test(Shannon ~ uri_cur2, data = div_child) #p=0.37

#Recent URI
tapply(div_child$Shannon, div_child$uri_rec2, summary)
wilcox.test(Shannon ~ uri_rec2, data = div_child) #p=0.30
t.test(Shannon ~ uri_rec2, data = div_child) #p=0.31

#Household members
summary(div_hh <- lm(Shannon ~ hhsize, data = div_child)) #p=0.47

#MULTIVARIABLE MODEL
#Need to set reference for HIV variable
class(div_child$hiv) #character; need to make factor to relevel
div_child$hiv <- as.factor(div_child$hiv)
div_child$hiv <- relevel(div_child$hiv, ref = "Unexposed")
summary(div_mv <- lm(Shannon ~ age + hiv + wood + season + abx + uri_rec2, data = div_child))
summary(div_mv <- lm(Shannon ~ age + hiv + wood + season + abx + uri_rec2 + hhsize, data = div_child))
```

#### Chao1

```{r}
#Is our Chao1 data normally distributed? 
shapiro.test(diversity$Chao1)  #no, because p<0.05; also looks skewed on histogram and QQ is wonky
histogram(diversity$Chao1)
qqnorm(diversity$Chao1, pch = 1, frame = FALSE)
qqline(diversity$Chao1, col = "steelblue", lwd = 2)

#What does it look like if we log transform it? Will this allow for parametric testing?
diversity$Chao1_log <- log(diversity$Chao1)
shapiro.test(diversity$Chao1_log) 
histogram(diversity$Chao1_log)
qqnorm(diversity$Chao1_log, pch = 1, frame = FALSE)
qqline(diversity$Chao1_log, col = "steelblue", lwd = 2)
  #Looks very normal! Can use t.test and linear regression on transformed data
div_child <- subset(diversity, subject == "child")

#Chao1 median and IQR for all samples
summary(diversity$Chao1)
sd(diversity$Chao1)
#Median (IQR) Chao1: 106 (53, 260)
#Mean (SD) 216 (300.4)

#what does it look like by subject? 
tapply(diversity$Chao1, diversity$subject, summary)
tapply(diversity$Chao1_log, diversity$subject, summary) #log mean of moms is lower
summary(aov(Chao1_log ~ subject, data = diversity)) #significant difference (p=0.03)

#what about by HIV status of the child?
tapply(div_child$Chao1, div_child$hiv, summary)
tapply(div_child$Chao1_log, div_child$hiv, summary)
summary(aov(Chao1_log ~ hiv, data = div_child)) #no significant difference by HIV status using log values and ANOVA (p=0.2)

#Age of the child
summary(div_age <- lm(Chao1_log ~ age, data = div_child)) #p=0.38

#Wood smoke exposure
tapply(div_child$Chao1, div_child$wood, summary)
t.test(Chao1_log ~ wood, data = div_child)    #p=0.93

#Season
tapply(div_child$Chao1, div_child$season, summary)
t.test(Chao1_log ~ season, data = div_child)    #p=0.43

#Receipt of abx in prior 3 months 
tapply(div_child$Chao1, div_child$abx, summary)
t.test(Chao1_log ~ abx, data = div_child)    #p=0.92

#Current URI  #p=0.06
tapply(div_child$Chao1_log, div_child$uri_cur2, summary)
t.test(Chao1_log ~ uri_cur2, data = div_child)

#Recent URI 
tapply(div_child$Chao1_log, div_child$uri_rec2, summary)
t.test(Chao1_log ~ uri_rec2, data = div_child)  #p=0.69

#Household members
summary(div_hh <- lm(Chao1_log ~ hhsize, data = div_child)) #p=0.42

#MULTIVARIABLE MODEL
#log with HUU as reference value
class(div_child$hiv) #character; need to make factor to relevel
div_child$hiv <- as.factor(div_child$hiv)
div_child$hiv <- relevel(div_child$hiv, ref = "Unexposed")
summary(div_mv <- lm(Chao1_log ~ age + hiv + wood + season + abx + uri_rec2, data = div_child))
summary(div_mv <- lm(Chao1_log ~ age + hiv + wood + season + abx + uri_cur2, data = div_child))
summary(div_mv <- lm(Chao1_log ~ age + hiv + wood + season + abx + uri_rec2 + hhsize, data = div_child))
```

#### Alpha diversity boxplots

- Helpful resource for boxplots: http://www.sthda.com/english/wiki/ggplot2-box-plot-quick-start-guide-r-software-and-data-visualization

```{r}
#make new variable with HIV status abbreviated
div_child$hiv2 <- NA
div_child$hiv2 [div_child$hiv == "Infected"] <- "CLWH"
div_child$hiv2 [div_child$hiv == "Exposed Uninfected"] <- "HEU children"
div_child$hiv2 [div_child$hiv == "Unexposed"] <- "HUU children"
table(div_child$hiv2)
#matches up with table(relative_df$hiv)
div_child$hiv2 <- as.factor(div_child$hiv2)
div_child$hiv2 <- reorder(div_child$hiv2, new.order=c("CLWH", "HEU children", "HUU children"))

#Shannon boxplot 
s1a2 <- ggplot(div_child, aes(x=hiv2, y=Shannon, fill=hiv2)) + 
  geom_boxplot() + labs(x="HIV status", y = "Shannon index")+
  scale_fill_manual(values=c('#800000FF','#155F83FF', '#FFA319FF')) +
  theme_classic() +
  theme(
    legend.text = element_text(size = 10),
    legend.position = "none",
    legend.title = element_blank(),
    axis.text = element_text(size = 10),
    axis.title.y = element_text(size = 12),
    axis.ticks.x = element_blank(),
    axis.title.x = element_blank(),
    axis.text.x = element_blank())

#Chao1 by HIV (to create legend we will use for our combined figure)
s1b <- ggplot(div_child, aes(x=hiv2, y=Chao1, fill=hiv2)) + 
  geom_boxplot() + labs(x="HIV status", y = "Chao1 richness")+
  scale_fill_manual(values=c('#800000FF','#155F83FF', '#FFA319FF')) +
  theme_classic() +
  theme(
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.title = element_blank(),
    axis.text = element_text(size = 10),
    axis.title.y = element_text(size = 12),
    axis.title.x = element_blank(),
    axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black"))

#Chao1 boxplot with no legend
s1b2 <- ggplot(div_child, aes(x=hiv2, y=Chao1, fill=hiv2)) + 
  geom_boxplot() + labs(x="HIV status", y = "Chao1 richness")+
  scale_fill_manual(values=c('#800000FF','#155F83FF', '#FFA319FF')) +
  theme_classic() +
  theme(
    legend.text = element_text(size = 10),
    legend.position = "none",
    legend.title = element_blank(),
    axis.text = element_text(size = 10),
    axis.title.y = element_text(size = 12),
    axis.ticks.x = element_blank(),
    axis.title.x = element_blank(),
    axis.text.x = element_blank())

#Get legend only to make it easier to size all parts of the figure
legend <- get_legend(s1b)

#combine the versions using cowplot to create supplemental Figure S1
prow2 <- plot_grid(s1a2, s1b2, labels = "AUTO", nrow = 1, ncol = 2, align = "h", axis = "b", rel_widths = c(1,1.1))
FigS1_v2 <- plot_grid(prow2, legend, rel_widths = c(2, 0.5))

png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/FigS1_v2.png",
    width = 7, height = 3.5, units = 'in', res = 600)
FigS1_v2
dev.off()

#save as .eps file to comply with Lancet Microbe requirements
ggsave(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/FigS1_v2.eps", width = 7, height = 3.5, units = "in")
FigS1_v2
dev.off()

#clean up environment
remove(div_age, div_mv, hiv.aov, m1, subj.aov)
```
- source for saving as .eps: https://stackoverflow.com/questions/5142842/export-a-graph-to-eps-file-with-r

### Beta diversity measures

*In order to compare overall composition by HIV status, we will first CLR transform our data and use euclidean distances for comparison*

- Sources: https://rdrr.io/github/microbiome/microbiome/man/transform.html
  https://microbiome.github.io/tutorials/Preprocessing.html

```{r}
pruned_clr <- microbiome::transform(pruned, "clr")
chi_clr = subset_samples(pruned_clr, subject == "child")
```

#### Beta diversity plots

- will save in .eps format for consistency with Lancet Microbe requirements

```{r}
#First: create new HIV variable as factor, then relevel
table(child$hiv)
child$hiv2 <- NA
child$hiv2 [child$hiv == "Infected"] <- "CLWH"
child$hiv2 [child$hiv == "Exposed Uninfected"] <- "HEU children"
child$hiv2 [child$hiv == "Unexposed"] <- "HUU children"
table(child$hiv2)
class(child$hiv2)
child$hiv2 <- as.factor(child$hiv2)
child$hiv2 <- reorder(child$hiv2, new.order=c("CLWH", "HEU children", "HUU children"))

#Import back into pseq (source: https://jacobrprice.github.io/2017/08/26/phyloseq-to-vegan-and-back.html)
sample_data(chi_clr) <- as.data.frame(child)
#now see if it worked:
bloop <- data.frame(sample_data(chi_clr))
table(bloop$hiv2)
#success!
remove(bloop)

#Now ordinate and create PCOA plot
clr_pcoa <- ordinate(chi_clr, method = "PCoA", distance = "euclidean")
p1 <- plot_ordination(chi_clr, clr_pcoa, color = "hiv2") +
  geom_point(size = 0.005) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values=c('#800000FF','#155F83FF', '#FFA319FF'))+
  stat_ellipse(aes(linetype=hiv2), geom="polygon", alpha=0, type="t", level=0.8, size=0.5) +
  xlab("PC1 (14.4%)") + ylab("PC2 (8.9%)") +
  scale_linetype_manual(values= c("dashed", "dashed", "dashed"))

p1
```

#### Permanova

- NOTE: since initial analysis, adonis has been depreciated so all analyses here updated with adonis2

```{r}
#Create OTU table
otu <- as.data.frame(otu_table(chi_clr, taxa_are_rows=TRUE))

#Need species in columns and observations in rows: transpose using t() function and save as a dataframe
  #Source: https://github.com/edamame-course/2015-tutorials/blob/master/final/Ashley_Intro_to_R.md
otu_trans <- data.frame(t(otu))
child <- data.frame(sample_data(chi_clr))

#Calculate distance and save as a matrix
BC.dist=vegdist(otu_trans, method="euclidean")

#HIV
# adonis(BC.dist ~ hiv, data = child, permutations = 1000)
  #Significant difference by HIV status among children (P = 0.043, R^2 = 0.019)
adonis2(BC.dist ~ hiv, data = child, permutations = 1000)
  #Significant difference by HIV status remains: R2 = 0.019, P = 0.039

#HIV + age (per 6/9/23 MK discussion to account for age in all PERMANOVAs)
#NOTE re: adonis2 'by' function: "by = "terms" will assess significance for each term (sequentially from first to last), setting by = "margin" will assess the marginal effects of the terms (each marginal term analysed in a model with all other variables)"
#Source: https://search.r-project.org/CRAN/refmans/vegan/html/adonis.html
#Previously, we used the by = "terms" function and ordered terms in multivariable models based on the R2 from each univariable model. However, it is worth exploring the by = "margin" so we can look at the marginal effects of all variables together.
# adonis(BC.dist ~ hiv + age, data = child, permutations = 1000)
  #Significant difference by HIV status among children (P = 0.042, R^2 = 0.019)
    #Significant difference by age as well (P = 0.03, R^2 = 0.012)

#NEW adonis2 code with by = "terms" and by = "margins"
adonis2(BC.dist ~ hiv + age, data = child, permutations = 1000, by = "terms")
  #Significant difference by HIV status among children (P = 0.036, R2 = 0.019) and by age (P = 0.023, R2 = 0.012
adonis2(BC.dist ~ hiv + age, data = child, permutations = 1000, by = "margin")
  #results are pretty consistent: HIV: p = 0.043, R2 = 0.019; age: p = 0.02, R2 = 0.12

#Age
adonis2(BC.dist ~ age, data = child, permutations = 1000)
  #Significant difference in overall composition by age (P = 0.03, R2 = 0.012)

#Season
  #Significant difference in overall composition by season (P < 0.001, R2 = 0.016)
adonis2(BC.dist ~ season, data = child, permutations = 1000)
  #Season + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis2(BC.dist ~ season + age, data = child, permutations = 1000, by = "terms")
adonis2(BC.dist ~ season + age, data = child, permutations = 1000, by = "margin")
#Significant difference by season (P = 0.002, R^2 = 0.016) and age (P = 0.03, R2 = 0.011) by "terms"
#Consistent by "margin": season (P = 0.004, R^2 = 0.016) and age (P = 0.03, R2 = 0.011)

#Wood
adonis2(BC.dist ~ wood, data = child, permutations = 1000)
  #No significant difference by wood smoke exposure among children (P = 0.12)
#Wood + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis2(BC.dist ~ wood + age, data = child, permutations = 1000, by = "terms")
#No signif difference by wood smoke exposure (P = 0.09) but age remains signif: (P = 0.02, R^2 = 0.012)
adonis2(BC.dist ~ wood + age, data = child, permutations = 1000, by = "margin") #consistent

#Abx
adonis2(BC.dist ~ abx, data = child, permutations = 1000)
  #Trend but no significant difference in overall composition by abx exposure (P = 0.08)
#Abx + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis2(BC.dist ~ abx + age, data = child, permutations = 1000, by = "terms")
#Trend but no significant difference in overall composition by abx exposure (P = 0.06) and age remains significant; same for by = "margin"
adonis2(BC.dist ~ abx + age, data = child, permutations = 1000, by = "margin")

#Recent URI
adonis2(BC.dist ~ uri_rec2, data = child, permutations = 1000)
#Significant difference in composition among children with recent URI (P = 0.008, R2=0.015)
  #Recent URI + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis2(BC.dist ~ uri_rec2 + age, data = child, permutations = 1000, by = "terms")
adonis2(BC.dist ~ uri_rec2 + age, data = child, permutations = 1000, by = "margin")
#Significant difference in composition by recent URI sx status (R2 = 0.015, p=0.004) and age (R2 = 0.11, p = 0.04) using both "terms" and "margin"

#Multivariable
  #NOTE: order of variables affects results when using by = "terms" per link above
#start with model that lists covariables by effect size from univariable analysis (signif variables only)
  #HIV R2 = 0.019, season R2 = 0.016, recent URI R2=0.015, age: R2 = 0.012
adonis2(BC.dist ~ hiv + season + uri_rec2 + age, data = child, permutations = 1000, by = "terms")
pm <- adonis2(BC.dist ~ hiv + season + uri_rec2 + age, data = child, permutations = 1000, by = "margin")
pm
  #In a multivariable model, HIV status (P=0.04, R2=0.019), season (P=0.002, R2=0.016), recent URI (P=0.008, R2=0.013), and age (P=0.03, R2=0.011) were a/w differences in overall microbiome composition using both by = options (above R2 and p values are from by = "margin")
```

Statistical reviewer from Lancet Microbe requested 95% CI for each R2 we presented. 

From web review appears the most straightforward way to obtain these values is through bootstrapping:
https://stats.stackexchange.com/questions/175026/formula-for-95-confidence-interval-for-r2
https://rstudio-pubs-static.s3.amazonaws.com/288625_b0cf85c5be974eb795cef778535289d8.html

General tutorial on bootstrapping:
https://stats.oarc.ucla.edu/r/faq/how-can-i-generate-bootstrap-statistics-in-r/

HOWEVER, I could not figure out how to bootstrap a microbiome PERMANOVA (code deleted but available in prior versions on Github) and reached out to Pixu Shi. Her response (8/12): "There is no CI for PERMANOVA because CI is for a parameter and PERMANOVA is nonparametric. The closest thing we can get is to run PERMANOVA 100 times, and report the 97.5 and 2.5 percentile of the F stat or p-value."

Looks like we can use the permustats function from the vegan package for this: https://search.r-project.org/CRAN/refmans/vegan/html/permustats.html

```{r}
#extract permutation results and observed statistics from our permanova model
perm <- permustats(pm)
#summary statistics; NOTE: only presents upper limit (97.5%), suggesting that default for adonis2 is a one-sided test.
summary(perm)
#can use 'alternative' command to get both the upper and lower limit values, but now our p value is 2-sided and as a result is no longer significant for hiv and age. This is ok; F statistics are always one-sided so we will report our one-sided p value and use the 2-sided statistics for the coverage interval alone. Confirmed the appropriateness of one-sided p values for F stats with Pixu on 8/14
summary(perm, alternative="two.sided")
```

**UPDATE** Upon further discussion with Pixu, we determined that generating a 95% CI is not possible for a nonparametric permutational test (see email from Pixu 8/17): "For the PERMANOVA R2, we cannot get an interval estimate. The R2 of covariate X is calculated as the sum of square of X divided by the total sum of square. The permutation test will generate a distribution of R2 under the null, which is used to obtain a p-value, but cannot be used to obtain interval estimate for R2 like a parametric test. Same can be said for the F statistics. The interval from permustats() is not an interval estimate for the F statistic, as you can see it doesn’t even cover your observed F when p-value is very small. The interval from permustats() is actually the upper quantile of F statistic under the null (achieved by permutation). In short, PERMANOVA is a nonparametric permutation test that doesn’t have interval estimate for R2 or F statistic."

We also received a reviewer comment requesting unsupervised clustering to clarify if there are some hidden groups among the data, so we will add this analysis. We tried Dirichlet Multinomial Models as suggested by the reviewer, but this was unsuccessful and we ultimately successfully used k-medioid clustering as done by McCumber et al in CID paper, using the code shared on his github: https://github.com/alexmccumber/BRAVE_Kids

#### MaAsLin2 models
```{r}
chi.filter <- filter_taxa(chi_prune, function(Abundance) mean(Abundance)>=50, TRUE)
ntaxa(chi.filter) #43

otu <- as.data.frame(otu_table(chi.filter, taxa_are_rows=TRUE))
#need to remove the unidentified species from here: row 11, 25, 37, 38, 40
  #source: https://stackoverflow.com/questions/12328056/how-do-i-delete-rows-in-a-data-frame
#corresponding respectively to: Corynebacterium sp. KPL1859, Streptococcus sp. SK643, Alkalibacterium sp. 20, uncultured Clostridium sp., Nocardioides sp. Root122
otu <- otu[-c(11, 25, 37, 38, 40),]

#Corynebacterium sp. KPL1995 -> Corynebacterium pseudodiphtheriticum
#Corynebacterium sp. KPL1986 -> Corynebacterium accolens
#Haemophilus sp. C1 -> Haemophilus haemolyticus

row.names(otu)[3] <- "Haemophilus haemolyticus"
row.names(otu)[10] <- "Corynebacterium pseudodiphtheriticum"
row.names(otu)[11] <- "Corynebacterium accolens"

#original MaAsLin2 with HUU as reference
fit_data2 = Maaslin2(
  input_data = otu, 
  input_metadata = child, 
  min_prevalence = 0.2,
  max_significance = 0.2,
  output = "maaslin_mv_kaiju_filt50_v4", 
  fixed_effects = c("hiv", "season", "age", "abx", "wood", "uri_rec2"),
  reference = c("hiv,Unexposed"))

#Maaslin only allows for pairwise comparisons, so we can only compare HUU to CLWH and HUU to HEU. What if we set HEU as the reference to see if anything is differentially abundant between HEU and CLWH?
fit_data2 = Maaslin2(
  input_data = otu, 
  input_metadata = child, 
  min_prevalence = 0.2,
  max_significance = 0.2,
  output = "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/maaslin_mv_kaiju_filt50_v5_HEUref", 
  fixed_effects = c("hiv", "season", "age", "abx", "wood", "uri_rec2"),
  reference = c("hiv,Exposed Uninfected"))
```

#### MaAsLin2 barplot (previously figure 3, now figure 2 in manuscript)
```{r}
#UPDATE AUG 3: use Maaslin2 results to create barplot
maaslin <- read.csv("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/maaslin_mv_kaiju_filt50_v4/significant_results.tsv", sep = "\t")
#clean up data: remove "." from species name and make names factors so we can reorder them
class(maaslin$feature)
maaslin$feature <- gsub(".", " ", maaslin$feature, fixed = TRUE)
maaslin$feature <- gsub("sp  KPL1995", "pseudodiphtheriticum", maaslin$feature, fixed = TRUE)
maaslin$feature <- as.factor(maaslin$feature)
#rename feature to species
maaslin <- dplyr::rename(maaslin, Species = feature)
#create new variable for positive or neg coefficient that we can use to add color to our figures
maaslin$posneg <- NA
maaslin$posneg [maaslin$coef >= 0] <- "Pos"
maaslin$posneg [maaslin$coef < 0] <- "Neg"
table(maaslin$posneg)

#Do the same for our new (Oct 23) Maaslin model with HEU as the reference
maas_heu <- read.csv("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/maaslin_mv_kaiju_filt50_v5_HEUref/significant_results.tsv", sep = "\t")
class(maas_heu$feature)
maas_heu$feature <- gsub(".", " ", maas_heu$feature, fixed = TRUE)
maas_heu$feature <- as.factor(maas_heu$feature)
#rename feature to species
maas_heu <- dplyr::rename(maas_heu, Species = feature)
#create new variable for positive or neg coefficient that we can use to add color to our figures
maas_heu$posneg <- NA
maas_heu$posneg [maas_heu$coef >= 0] <- "Pos"
maas_heu$posneg [maas_heu$coef < 0] <- "Neg"
table(maas_heu$posneg)

#will create 4 horizontal barplots and use cowplot to align them into 1 figure
#MAKE NEW VAR for coef + or neg and then use that for your barplot fill
#season
m_seas <- subset(maaslin, metadata == "season")
#relevel so all negative values grouped together
m_seas$Species <- reorder(m_seas$Species, new.order=c("Micrococcus luteus", "Kocuria polaris", "Moraxella bovis", "Moraxella caprae", "Moraxella catarrhalis", "Moraxella caviae", "Moraxella cuniculi", "Moraxella lacunata", "Moraxella lincolnii", "Moraxella macacae", "Moraxella nonliquefaciens", "Moraxella oblonga", "Moraxella ovis"))

m_s <- ggplot(m_seas, aes(x=coef, y=Species, fill=posneg)) +
  geom_bar(stat="identity") +
  ggtitle("Rainy season") +
  scale_fill_manual(values=c('#800000FF', '#155F83FF')) +
  scale_x_continuous(breaks = seq(-6, 4, 1), limits = c(-5, 3)) +
  theme_classic() +
  theme(
    plot.title = element_text(size = 12, hjust = 0.5, face = "bold"),
    # legend.text = element_text(size = 12),
    legend.position = "none",
    # legend.title = element_blank(),
    axis.text = element_text(size = 10),
    axis.text.y = element_text(face = "italic"),
    axis.title = element_blank()) +
  geom_errorbar(aes(xmin=coef-stderr, xmax=coef+stderr), width=.2) 

#age
m_age <- subset(maaslin, metadata == "age")

m_a <- ggplot(m_age, aes(x=coef, y=Species, fill=posneg)) +
  geom_bar(stat="identity") +
  ggtitle("Age") +
  scale_fill_manual(values=c('#800000FF', '#155F83FF')) +
  scale_x_continuous(breaks = seq(-6, 4, 1), limits = c(-5, 3)) +
  theme_classic() +
  theme(
    plot.title = element_text(size = 12, hjust = 0.5, face = "bold"),
    # legend.text = element_text(size = 12),
    legend.position = "none",
    # legend.title = element_blank(),
    axis.text = element_text(size = 10),
    axis.text.y = element_text(face = "italic"),
    axis.title = element_blank()) +
  geom_errorbar(aes(xmin=coef-stderr, xmax=coef+stderr), width=.2) 

#recent URI
m_uri <- subset(maaslin, metadata == "uri_rec2")

m_u <- ggplot(m_uri, aes(x=coef, y=Species, fill=posneg)) +
  geom_bar(stat="identity") +
  ggtitle("Recent URI symptoms") +
  scale_fill_manual(values=c('#155F83FF')) +
  scale_x_continuous(breaks = seq(-6, 4, 1), limits = c(-5, 3)) +
  theme_classic() +
  theme(
    plot.title = element_text(size = 12, hjust = 0.5, face = "bold"),
    # legend.text = element_text(size = 12),
    legend.position = "none",
    # legend.title = element_blank(),
    axis.text = element_text(size = 10),
    axis.text.y = element_text(face = "italic"),
    axis.title = element_blank()) +
  geom_errorbar(aes(xmin=coef-stderr, xmax=coef+stderr), width=.2) 

#hiv
m_hiv <- subset(maaslin, metadata == "hiv")

#Our Figure 2 shows effect sizes for differentially abundant taxa with a q<0.2; we see that our effect sizes for all other variables (season, age, etc) in the 2 different maaslin models we ran (one with HUU as ref, one with HEU as ref) are identical, so it seems reasonable to combine the HEU-HIV and HUU-HIV results into one panel as long as we have some way to differentiate the reference groups
m_heu <- subset(maas_heu, metadata == "hiv")

m_hiv2 <- rbind(m_hiv, m_heu)
#relevel so all Coryne grouped together
m_hiv2$Species <- reorder(m_hiv2$Species, new.order=c("Micrococcus luteus", "Corynebacterium pseudodiphtheriticum", "Corynebacterium aurimucosum", "Corynebacterium accolens"))
#now we will bind them together to make one combined hiv dataframe and make a figure from that 

#Source to split C. pseudo into 2 lines on the Y axis for space: https://stackoverflow.com/questions/20123147/add-line-break-to-axis-labels-and-ticks-in-ggplot
m_h2 <- ggplot(m_hiv2, aes(x=coef, y=Species, fill=posneg)) +
  geom_bar(stat="identity") +
  ggtitle("HIV infection") +
  scale_fill_manual(values=c('#800000FF')) +
  scale_x_continuous(breaks = seq(-6, 4, 1), limits = c(-5, 3)) +
  scale_y_discrete(labels=function(x){sub("\\s", "\n", x)}) +
  theme_classic() +
  theme(
    plot.title = element_text(size = 12, hjust = 0.5, face = "bold"),
    # legend.text = element_text(size = 12),
    legend.position = "none",
    # legend.title = element_blank(),
    axis.text = element_text(size = 10),
    axis.text.y = element_text(face = "italic"),
    axis.title = element_blank()) +
  geom_errorbar(aes(xmin=coef-stderr, xmax=coef+stderr), width=.2) 


#combine into 1 cowplot
right <- plot_grid(m_h2, m_u, m_a,
          labels = c('B', 'C', 'D'),
          nrow = 3, 
          label_size = 12,
          rel_heights = c(1.5, 0.8, 1.2),
          align = "v")

fig3 <- plot_grid(m_s, right,
                  labels = c('A', '', '', ''),
                  nrow = 1,
                  label_size = 14, 
                  rel_heights = c(1, 1.2))
                  # align = "v", axis = "b")

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig2NEW(maaslin)_V2b.pdf", plot = fig3, dpi = 300, width = 12, height = 5)
```

### K-medioid clustering

```{r}
chi_prune = subset_samples(pruned, subject == "child")
phy.otus <- transform_sample_counts(chi_prune, function(Abundance) Abundance/sum(Abundance))
otu_df <- psmelt(phy.otus)
otu_df$OTU <- as.character(otu_df$OTU)
otu_abundances <- aggregate(otu_df$Abundance, by=list(OTU=otu_df$OTU), FUN=sum)
otu_abundances$x <- (otu_abundances$x)/(nsamples(chi_prune))
names(otu_abundances)[names(otu_abundances) == 'x'] <- 'otu_Ab'
sum(otu_abundances$otu_Ab)  #sums to 1
nrow(otu_abundances)        #6709 species
remove(phy.otus, otu_df)

# Agglomerate taxa into species --> already done, so no need to do again
# Transform to relative abundances
chi.rel <- transform_sample_counts(chi_prune, function(Abundance) Abundance/sum(Abundance))
sample_sums(chi.rel)  # This is a sanity check to make sure that relative abundance was calculated for each sample prior to pooling data
#All samples add to 1
melted_df <- psmelt(chi.rel)

#clean up melted_df
melted_df <- within(melted_df, rm(sib_weight, sib_sex, sib_height, sib_muac, sib_race, sib_bfeed_1,
                                    sib_bfeed_2, sib_uri_recent, sib_uri_current, sib_pcr_1, sib_pcr_2,
                                    sib_bcg, sib_hepb, sib_clinic, sib_clinic_dx___5, sib_clinic_dx_oth,
                                    sib_hosp, sib_meds, sib_meds_name, sibmo, sibyr, sib_vl1_weeks, sib_vl_weeks2,
                                    sib_med_days, sib_dpt, sib_pcv, sib_rota, sib_polio, sib_measles, 
                                    sib_age, samp_storage, enr_clinic, enr_clinic_other, mat_np))
```


```{r}
# Function to make matrix usable for vegan from phyloseq object
vegan_otu <- function(physeq) {
    OTU <- otu_table(physeq)
    if (taxa_are_rows(OTU)) {
        OTU <- t(OTU)
    }
    return(as(OTU, "matrix"))
}

#Aggregate at species level and drop NA reads for the distance and clustering
test <- as.data.frame(as(tax_table(chi.rel),"matrix"),stringsAsFactors=FALSE)

chi.spec = tax_glom(chi.rel, taxrank = "Species") 

DistType = t(as.data.frame(vegan_otu(chi.spec)))

data.dist = vegdist(vegan_otu(chi.spec), method = "bray")

pam.clustering=function(x,k) { # x is a distance matrix and k the number of clusters
                         require(cluster)
                         cluster = as.vector(pam(as.dist(x), k, diss=TRUE)$clustering)
                         return(cluster)
                        }

#See how many clusters is ideal (range from 1 to 20)

nclusters=NULL
library(clusterSim)
for (k in 1:20) { 
		if (k==1) {
			nclusters[k]=NA 
		} else {
			data.cluster_temp=pam.clustering(data.dist, k)
			nclusters[k]=index.G1(t(DistType), data.cluster_temp,  d = data.dist,
			centrotypes = "medoids")
		}
	}
```

**Stop after this plot and see how many clusters you have**
```{r}
plot(nclusters, type="h", xlab="k clusters", ylab="CH index")
```

- it looks like 5 clusters is the "best" number (source: https://stats.stackexchange.com/questions/52838/what-is-an-acceptable-value-of-the-calinski-harabasz-ch-criterion)

```{r}
#Set new number of clusters = to k based on above plot
data.cluster=pam.clustering(data.dist, k=5)

nclusters = index.G1(t(DistType), data.cluster, d = data.dist, centrotypes = "medoids")

#look at number to see if it is highest average (or could change mean to median) <- note from Alex; not sure how to interpret
obs.silhouette=mean(silhouette(data.cluster, data.dist)[,3])  #where does the 3 come from?
obs.silhouette=median(silhouette(data.cluster, data.dist)[,3])
  #tutorial on silhouette: https://stat.ethz.ch/R-manual/R-devel/library/cluster/html/silhouette.html
test <- silhouette(data.cluster, data.dist)
summary(test)

library(ade4)
obs.pcoa=dudi.pco(data.dist, scannf=F, nf=3)
  #Get warning for non euclidean distance; we intentionally used non-transformed data and bray-curtis (not euclidean) measures, so not unexpected

#reading about dudi.pco: scannf = TRUE should give you a bar plot of eigenvalues; if FALSE, then nf = number of axes to be kept
  #how do we know that 3 axes is the right number? 
# obs.pcoa=dudi.pco(data.dist, scannf=T) #if using scannf=T, you have to specify the number of axes no matter what
#https://pbil.univ-lyon1.fr/R/pdf/course2.pdf --> tutorial on picking the number of axes using our obs.pcoa
  #also walks you through eigenvalues
obs.pcoa$eig
sum(obs.pcoa$eig) #0.33
barplot(obs.pcoa$eig)
(kip <- 100 * obs.pcoa$eig/sum(obs.pcoa$eig))
cumsum(kip) #viewing the data in a 3d space extracts only 44.5% of the info; we'd have to view in 40 axes to see 99% of the contained information. On the other hand, Alex's PC% were relatively small and the whole point of these plots is dimension reduction. Can d/w MK; do we need to re-do with a higher number of axes?

s.class(obs.pcoa$li, fac=as.factor(data.cluster), grid=T,  col = c(1:5))
  #gives you a cool figure!

p = cbind(obs.pcoa$li,data.cluster)

p$data.cluster = as.factor(p$data.cluster)

library(RColorBrewer)
library(scales)
chisp2 <- c("#293352", "#800000FF", "#767676FF", "#FFA319FF", "#8A9045FF", "#155F83FF", "#C16622FF", "#4D004B", "#8F3931FF", "#58593FFF", "#350E20FF")
show_col(chisp2)
chisp3 <- c("#293352", "#800000FF","#8A9045FF", "#FFA319FF", "#4D004B")
show_col(chisp3) #bit dark, let's make a lighter set of 5 for the cluster boxplots
chisp4 <- c("#155F83FF", "#8F3931FF", "#FFA319FF", "#8A9045FF", "#C16622FF")
show_col(chisp4)

library(ggpubr)
clust <- ggplot(data = p, aes(x = A1, y = A2)) +
  geom_point(data = p, aes(color = data.cluster), alpha=1, size=1.0) + 
  scale_color_manual(values = chisp3) +
  theme_classic() +
    theme(
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 10))+
  stat_ellipse(aes(color = data.cluster)) +
  stat_stars(aes(color = data.cluster), show.legend = T) +
  xlab("PC1 (22.6%)") +
  ylab("PC2 (17.3%)") +
  labs(color = "Microbiome Profile")+ 
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1))

clust
#Question: how to tell PC %? Standard axes are A1 and A2 instead of PC% and PC% --> created a pcoa plot by cluster below and used those PC%; make sure this is ok

#save for ease of review; once you figure out how to update the axes accurately and finalize the figure, will need to save as an eps file (sample code below)
png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/clust_v2.png",
    width = 12, height = 9, units = 'in', res = 600)
clust
dev.off()

# #save as .eps file to comply with Lancet Microbe requirements
# ggsave(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/clust.eps", width = 7, height = 3.5, units = "in")
# clust
# dev.off()
```

- Now that we have generated our clusters, we need to add this data to our metadata and see if any of our variables of interest are associated with these clusters

```{r}
sample_data(chi.spec)$cluster = as.factor(data.cluster)

sampleDF = sample_data(chi.spec) %>%
  data.frame(.)

#save sampleDF as a csv file in case we need to return to it
write.csv(sampleDF, '/Users/swetapatel/OneDrive - Duke University/Fogarty coding/metadata_clust.csv')

#To get PC% for axes for above figure:
#Create new PCoA with clusters as variable, using euclidean distances generated from CLR transforming chi.spec
chi.spec_clr <- microbiome::transform(chi.spec, "clr")
clr_pcoa <- ordinate(chi.spec_clr, method = "PCoA", distance = "euclidean")
p1 <- plot_ordination(chi.spec_clr, clr_pcoa, color = "cluster") +
  geom_point(size = 0.005) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values = chisp4)+
  stat_ellipse(aes(color = cluster)) +
  stat_stars(aes(color = cluster), show.legend = T) +
  xlab("PC1 (14.3%)") +
  ylab("PC2 (8.9%)") +
  labs(color = "Microbiome Profile")
#gives us axes of 8.9% (y) and 14.3% (x); we will input these back into our figure above and make sure this is appropriate with MK. Weird though because 2 different distances used (euclidean here, bray above) and the clusters look different; what does our pcoa look like if we use bray here too?
clr_pcoa2 <- ordinate(chi.spec, method = "PCoA", distance = "bray")
p1_clust <- plot_ordination(chi.spec, clr_pcoa2, color = "cluster") +
  geom_point(size = 0.005) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values = chisp4)+
  stat_ellipse(aes(color = cluster)) +
  stat_stars(aes(color = cluster), show.legend = T) +
  xlab("PC1 (22.6%)") +
  ylab("PC2 (17.3%)") +
  labs(color = "Microbiome Profile")

p1_clust

#looks the same!! so we can use these PC% in our figure above

#do our clusters associate with any of our demographic variables?
#age by cluster
ggplot(sampleDF, aes(x=cluster, y=age)) +
  geom_boxplot(fill=chisp4) +
  xlab('Cluster') + 
  ylab('Age (Months)') +
  theme_classic()

#HIV category by cluster
#you don't have to manually create frequencies; use stat = "count" and get rid of your y= in aes
#Create new category of HIV to match our prior figure labels:
sampleDF$hiv2 <- NA
sampleDF$hiv2 [sampleDF$hiv == "Infected"] <- "CLWH"
sampleDF$hiv2 [sampleDF$hiv == "Exposed Uninfected"] <- "HEU children"
sampleDF$hiv2 [sampleDF$hiv == "Unexposed"] <- "HUU children"
table(sampleDF$hiv2)
#matches up with table(relative_df$hiv)
sampleDF$hiv2 <- as.factor(sampleDF$hiv2)
sampleDF$hiv2 <- reorder(sampleDF$hiv2, new.order=c("CLWH", "HEU children", "HUU children"))

#create plot
hiv_clust <- ggplot(sampleDF, aes(fill=hiv2, x=cluster)) + geom_bar(stat= "count", position = "fill", width = 0.9) +
  scale_fill_manual(values = c('#800000FF','#155F83FF', '#FFA319FF')) +
  # scale_y_continuous(labels = waiver()) +  
  scale_x_discrete(labels=xlab) +
  labs(title=" ", x="Microbiome profile", y="Proportion of children", fill = "HIV status")+
  theme_classic() +
  theme(
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12))

#it looks like cluster 2 has a lot more CLWH; will do an analysis comparing cluster 2 to all other clusters by HIV status

#save for ease of review; if you include in manuscript, will need to save as an eps file (sample code below)
png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/HIV_by_clust.png",
    width = 12, height = 9, units = 'in', res = 600)
hiv_clust
dev.off()

# #save as .eps file to comply with Lancet Microbe requirements
# ggsave(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/FILENAMEHERE.eps", width = 7, height = 3.5, units = "in")
# p
# dev.off()

#Are there any other variables we should look at visually? What about wood and antibiotics? Not signif associated below but it makes sense that they could affect microbiome composition
sampleDF$wood2 <- NA
sampleDF$wood2 [sampleDF$wood == "0"] <- "No"
sampleDF$wood2 [sampleDF$wood == "1"] <- "Yes"
table(sampleDF$wood2)
#matches up with table(relative_df$hiv)
sampleDF$wood2 <- as.factor(sampleDF$wood2)
sampleDF$wood2 <- reorder(sampleDF$wood2, new.order=c("Yes", "No"))
wood_clust <- ggplot(sampleDF, aes(fill=wood2, x=cluster)) + geom_bar(stat = "count", position = "fill", width = 0.9) + 
  scale_fill_manual(values = c('#800000FF','#155F83FF'))+
  scale_x_discrete(labels=xlab) +
  labs(title=" ", x="Microbiome profile", y="Proportion of children", fill = "Wood smoke exposure")+
  theme_classic() +
  theme(
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12))

wood_clust
#can try 2 comparisons: 1+5 vs 2-4, and 2 vs all other clusters 

class(sampleDF$abx)
sampleDF$abx <- as.factor(sampleDF$abx)
sampleDF$abx <- reorder(sampleDF$abx, new.order=c("Yes", "No"))
abx_clust <- ggplot(sampleDF, aes(fill=abx, x=cluster)) + geom_bar(stat = "count", position = "fill", width = 0.9) + 
  scale_fill_manual(values = c('#800000FF','#155F83FF'))+
  scale_x_discrete(labels=xlab) +
  labs(title=" ", x="Microbiome profile", y="Proportion of children", fill = "Antibiotic exposure")+
  theme_classic() +
  theme(
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12))

abx_clust
#if we group 1 and 5 together and 2-4 together, what does that look like?

#Since we are doing all these split comparisons based on our bar plot results, we will save these using cowplot in case we want to include them as a supplemental figure
fig_barclust <- plot_grid(hiv_clust, wood_clust, abx_clust, labels = c('A', 'B', 'C'), align = "h", axis = "b", label_size = 12)
fig_barclust

#save as .eps file to comply with Lancet Microbe requirements
ggsave(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/barclust_combined.eps", width = 9, height = 5, units = "in")
fig_barclust
dev.off()
```

#### Statistical testing of demographic variables by biotype (as done in CID paper); will use same variables as used in Table 1 (plus HIV). Included in Table (table # TBD for manuscript, likely supplemental)

```{r}
#Age
tapply(sampleDF$age, sampleDF$cluster, summary)
kruskal.test(sampleDF$age, sampleDF$cluster) #p=0.04

#Sex
table(sampleDF$sex, sampleDF$cluster)
prop.table(table(sampleDF$sex, sampleDF$cluster), 2)
summary(table(sampleDF$sex, sampleDF$cluster))

#HIV status!
table(sampleDF$hiv, sampleDF$cluster)
prop.table(table(sampleDF$hiv, sampleDF$cluster), 2)
summary(table(sampleDF$hiv, sampleDF$cluster))
fisher.test(sampleDF$hiv, sampleDF$cluster)
#received following error: Error in fisher.test(sampleDF$hiv, sampleDF$cluster) : 
  #FEXACT error 7(location). LDSTP=18630 is too small for this problem,
 # (pastp=64.7107, ipn_0:=ipoin[itp=201]=9, stp[ipn_0]=62.0229).
#Increase workspace or consider using 'simulate.p.value=TRUE'
fisher.test(sampleDF$hiv, sampleDF$cluster, simulate.p.value = TRUE)

#Maternal age
tapply(sampleDF$mat_age2, sampleDF$cluster, summary)
kruskal.test(sampleDF$mat_age2, sampleDF$cluster)

#Season
table(sampleDF$season, sampleDF$cluster)
prop.table(table(sampleDF$season, sampleDF$cluster), 2)
summary(table(sampleDF$season, sampleDF$cluster)) #p=0.008!

#Maternal education
table(sampleDF$mat_educ2)
table(sampleDF$mat_educ2, sampleDF$cluster)
prop.table(table(sampleDF$mat_educ2, sampleDF$cluster), 2)
# summary(table(sampleDF$mat_educ2, sampleDF$cluster))
fisher.test(sampleDF$mat_educ2, sampleDF$cluster)

#Electricity
table(sampleDF$elec, sampleDF$cluster)
prop.table(table(sampleDF$elec, sampleDF$cluster), 2)
# summary(table(sampleDF$elec, sampleDF$cluster))
fisher.test(sampleDF$elec, sampleDF$cluster)

#Wood 
table(sampleDF$wood, sampleDF$cluster)
prop.table(table(sampleDF$wood, sampleDF$cluster), 2)
summary(table(sampleDF$wood, sampleDF$cluster))

#Num household members
tapply(sampleDF$hhsize, sampleDF$cluster, summary)
kruskal.test(sampleDF$hhsize, sampleDF$cluster)

#antibiotics
table(sampleDF$abx) #24 children received abx
table(sampleDF$abx, sampleDF$cluster)
prop.table(table(sampleDF$abx, sampleDF$cluster), 2)
# summary(table(sampleDF$abx, sampleDF$cluster))
fisher.test(sampleDF$abx, sampleDF$cluster)

#URI in past 1 month
table(sampleDF$uri_recent) #59 kids with URI sx
table(sampleDF$uri_rec2, sampleDF$cluster)
prop.table(table(sampleDF$uri_rec2, sampleDF$cluster), 2)
summary(table(sampleDF$uri_rec2, sampleDF$cluster))

#PCV doses
sampleDF$pcv3 <- NA
sampleDF$pcv3 [sampleDF$pcv == 0 | sampleDF$pcv == 1 | sampleDF$pcv == 2] <- "No"
sampleDF$pcv3 [sampleDF$pcv == 3] <- "Yes"
table(sampleDF$pcv3)
table(sampleDF$pcv) #confirms numbers match up
table(sampleDF$pcv3, sampleDF$cluster)
prop.table(table(sampleDF$pcv3, sampleDF$cluster), 2)
# summary(table(sampleDF$pcv3, sampleDF$cluster))
fisher.test(sampleDF$pcv3, sampleDF$cluster)

#HiB
sampleDF$dpt3 <- NA
sampleDF$dpt3 [sampleDF$dpt == 0 | sampleDF$dpt == 1 | sampleDF$dpt == 2] <- "No"
sampleDF$dpt3 [sampleDF$dpt == 3] <- "Yes"
table(sampleDF$dpt3)
table(sampleDF$dpt) #confirms numbers match up
table(sampleDF$dpt3, sampleDF$cluster)
prop.table(table(sampleDF$dpt3, sampleDF$cluster), 2)
# summary(table(sampleDF$dpt3, sampleDF$cluster))
fisher.test(sampleDF$dpt3, sampleDF$cluster)

#breastfeeding
table(sampleDF$bfeed_1) #49 kids exclusively breastfed, 59 formula fed, and 35 received both breastmilk and formula
sampleDF$bfeed_any <- NA
sampleDF$bfeed_any [sampleDF$bfeed_1 == 1 | sampleDF$bfeed_1 == 3] <- "Yes"
sampleDF$bfeed_any [sampleDF$bfeed_1 == 2] <- "No"
table(sampleDF$bfeed_any)
table(sampleDF$bfeed_any, sampleDF$cluster)
prop.table(table(sampleDF$bfeed_any, sampleDF$cluster), 2)
summary(table(sampleDF$bfeed_any, sampleDF$cluster))

###Asked MK about growth percentile with available variables 9/19; need to calculate Z scores for height and weight.
#R package available to calculate 
#source for zscorer: https://github.com/nutriverse/zscorer
library(zscorer)
#sex needs to be coded as male (1) and female (2)
sampleDF$sex2 <- NA
sampleDF$sex2 [sampleDF$sex == 0] <- 1
sampleDF$sex2 [sampleDF$sex == 1] <- 2
table(sampleDF$sex)
table(sampleDF$sex2) #matches up
#age needs to be in days, not months
sampleDF$age_d <- sampleDF$age*(30.4375) #confirmed that B-01 age and age_d match up

#weight for age
#ERROR: weight for U21 listed as 80.5 kg, corresponding to a z score of 37.6. Likely an error; will double check data entry with Charity
#in the meantime, drop U21 from our wfa z score calculations
sampleDF2 <- subset(sampleDF, study_id !="U.21")
sampleDF2 <- addWGSR(data=sampleDF2, sex="sex2", firstPart = "weight", secondPart = "age_d", index="wfa")
summary(sampleDF2$wfaz)
hist(sampleDF2$wfaz) #roughly normal, slight left skew
tapply(sampleDF2$wfaz, sampleDF2$cluster, summary)
kruskal.test(sampleDF2$wfaz, sampleDF2$cluster)

#height for age
sampleDF3 <- addWGSR(data=sampleDF, sex="sex2", firstPart = "height", secondPart = "age_d", index="hfa")
summary(sampleDF3$hfaz)
hist(sampleDF3$hfaz) #roughly normal, slight left skew
tapply(sampleDF3$hfaz, sampleDF3$cluster, summary)
kruskal.test(sampleDF3$hfaz, sampleDF3$cluster)
```

Combining profiles to look for specific associations given the barplots above for HIV status, wood smoke exposure, and antibiotic exposure:

```{r}
sampleDF$clust15 <- NA
sampleDF$clust15 [sampleDF$cluster == "1" | sampleDF$cluster == "5"] <- "C-D"
sampleDF$clust15 [sampleDF$cluster == "2" | sampleDF$cluster == "3" |sampleDF$cluster == "4"] <- "Other"
table(sampleDF$clust15)

sampleDF$clust2 <- NA
sampleDF$clust2 [sampleDF$cluster == "2"] <- "SP-HI"
sampleDF$clust2 [sampleDF$cluster == "1" | sampleDF$cluster == "3" |sampleDF$cluster == "4" |sampleDF$cluster == "5"] <- "Other"
table(sampleDF$clust2)

#HIV 
#by cluster 2
table(sampleDF$hiv, sampleDF$clust2)
prop.table(table(sampleDF$hiv, sampleDF$clust2), 2)
summary(table(sampleDF$hiv, sampleDF$clust2)) #P=0.013!!!

#by cluster 1+5 
table(sampleDF$hiv, sampleDF$clust15)
prop.table(table(sampleDF$hiv, sampleDF$clust15), 2)
summary(table(sampleDF$hiv, sampleDF$clust15)) #no significant association

#wood smoke 
#by cluster 2
table(sampleDF$wood, sampleDF$clust2)
prop.table(table(sampleDF$wood, sampleDF$clust2), 2)
summary(table(sampleDF$wood, sampleDF$clust2)) #P = 0.045

#by cluster 1+5 
table(sampleDF$wood, sampleDF$clust15)
prop.table(table(sampleDF$wood, sampleDF$clust15), 2)
summary(table(sampleDF$wood, sampleDF$clust15)) #P = 0.014

#antibiotics
#by cluster 2
table(sampleDF$abx, sampleDF$clust2)
prop.table(table(sampleDF$abx, sampleDF$clust2), 2)
summary(table(sampleDF$abx, sampleDF$clust2)) #P = 0.047 but may be incorrect (some n < 10 but not <5; has cutoff changed?)
fisher.test(sampleDF$abx, sampleDF$clust2) #P = 0.08

#by cluster 1+5 
table(sampleDF$abx, sampleDF$clust15)
prop.table(table(sampleDF$abx, sampleDF$clust15), 2)
summary(table(sampleDF$abx, sampleDF$clust15)) #P = 0.008
```

#### Multivariable analyses of associations between clinical variables and microbiome profiles.

A priori, we would expect antibiotic exposure and URI sx to associate with both HIV and microbiome composition (even though we do not see this in our demographics table). Wood smoke shouldn't be affected by HIV status (except potentially as proxy for low SES --> less access to healthcare but this is a bit of a stretch as 61% of our study participants were exposed to wood smoke). Thus, we will adjust for age, abx exposure, and URI sx when looking for associations between HIV and microbiome profile. We need binary outcomes for multivariable logistic regression, so will look at cluster 2 yes/no and clusters 1/5 yes/no by HIV status. Can also look at clusters using multinomial logistic regression? Run and confirm with MK. 

Update: will run multivariable multinomial logistic regression adjusting for all of our prespecified confounders ***

```{r}
sampleDF <- read.csv("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/metadata_clust.csv")
#Need to create binary 0/1 variables for cluster 2 and cluster 1/5 vars

sampleDF$clust2b <- NA
sampleDF$clust2b [sampleDF$clust2 == "SP-HI"] <- 1
sampleDF$clust2b [sampleDF$clust2 == "Other"] <- 0
table(sampleDF$clust2b)

sampleDF$clust15b <- NA
sampleDF$clust15b [sampleDF$clust15 == "C-D"] <- 1
sampleDF$clust15b [sampleDF$clust15 == "Other"] <- 0
table(sampleDF$clust15b)

sampleDF$hiv <- as.factor(sampleDF$hiv)
sampleDF$hiv <- relevel(sampleDF$hiv, ref = "Unexposed")

sampleDF$abx <- as.factor(sampleDF$abx)
sampleDF$abx <- relevel(sampleDF$abx, ref = "No")

sampleDF$uri_rec2 <- as.factor(sampleDF$uri_rec2)
sampleDF$uri_rec2 <- relevel(sampleDF$uri_rec2, ref = "No")

summary(mylogit <- glm(clust2b ~ hiv + age + abx + uri_rec2, data = sampleDF, family = "binomial")) #significant assoc b/w cluster 2 and HIV infection

#we adjust all other analyses for season and wood smoke exposure as well, so for uniformity will do here: 
class(sampleDF$wood2)
sampleDF$wood2 <- as.factor(sampleDF$wood2)
sampleDF$wood2 <- relevel(sampleDF$wood2, ref = "No")

sampleDF$season <- as.factor(sampleDF$season)
sampleDF$season <- relevel(sampleDF$season, ref = "Dry")

summary(mylogit <- glm(clust2b ~ hiv + age + abx + uri_rec2 + season + wood2, data = sampleDF, family = "binomial")) #significant assoc remains b/w cluster 2 and HIV infection
exp(mylogit$coefficients) #OR (95% CI) for CLWH: 5.31 (1.66, 20.75)
exp(confint(mylogit))

###MULTINOMIAL LOGISTIC REGRESSION###
#To look at association b/w HIV and all clusters (unlikely to show anything given small n but we will try)
#Using code from our PCV-13 serotype analysis published in PLOS One in 2021
library(nnet)
sampleDF$cluster <- relevel(sampleDF$cluster, ref = "1")
summary(test1 <- multinom(cluster ~ hiv + age + uri_rec2 + season + wood2, data = sampleDF))
z1 <- summary(test1)$coefficients/summary(test1)$standard.errors
z1
#Calculate p-values using Wald tests (here z-tests).
#2-tailed z test
p1 <- (1 - pnorm(abs(z1), 0, 1)) * 2
p1 #HIV infection a/w cluster 2 (p=0.017) and age a/w cluster 3 (p=0.007)

#The exponentiated regression coefficients are relative risk ratios for a unit change in the predictor variable
## extract the coefficients from the model and exponentiate
exp(coef(test1))
#Obtaining 95% CI for our RR
exp(confint(test1))

#Why didn't we include abx in the multivariable model? It is one of our prespecified confounders.
summary(test2 <- multinom(cluster ~ hiv + age + uri_rec2 + season + wood2 + abx, data = sampleDF))
z2 <- summary(test2)$coefficients/summary(test2)$standard.errors
z2
#Calculate p-values using Wald tests (here z-tests).
#2-tailed z test
p2 <- (1 - pnorm(abs(z2), 0, 1)) * 2
p2 #HIV infection (p=0.016) and wood (p=0.04) a/w cluster 2 and age (p=0.007) and uri sx (p=0.02) a/w cluster 3

#The exponentiated regression coefficients are relative risk ratios for a unit change in the predictor variable
## extract the coefficients from the model and exponentiate
exp(coef(test2))
#Obtaining 95% CI for our RR
exp(confint(test2))
```
*Summary of multivariable analysis*
**In a multivariable logistic regression model adjusting for all covariables specified in our methods with presence/absence of cluster 2 as the outcome of interest, HIV infection was associated with presence of cluster 2 (S. pneumo - H. flu); OR (95% CI): 5.31 (1.66, 20.75).**

**In a multinomial logistic regression model with cluster membership as the outcome; cluster 1 set as the reference; and HIV status, age, wood smoke exposure, season, and uri sx as covariables, HIV was significantly associated with the presence of cluster 2; relative risk ratio (95% CI) 5.17 (1.30, 20.55) ; p=0.019, as was wood smoke (RRR (95% CI): 3.63 (1.16, 11.36), p=0.026). Age and URI sx were significantly associated with the presence of cluster 3; age relative risk ratio (95% CI) 0.95 (0.90, 0.99) ; p=0.012; URI sx RRR (95% CI): 4.64 (1.28, 16.80), p=0.019. Lastly,  season was associated with cluster 4: RRR (95%): 0.23 (0.08, 0.65), p=0.005.

- Microbiome factors: did alpha diversity differ by cluster? What species comprised each cluster?

```{r}
Shannon = estimate_richness(chi.spec, measures = "Shannon")%>%
  cbind(., sampleDF, by = "names") #used Alex's code but this gives you a new dataframe to use called "Shannon"

tapply(Shannon$Shannon, Shannon$cluster, summary)
kruskal.test(Shannon$Shannon, Shannon$cluster) #P<0.001

shan <- ggplot(Shannon, aes(x=cluster, y=Shannon, fill=cluster)) + 
  geom_boxplot() + labs(x="Microbiome profile", y = "Shannon index")+
  scale_fill_manual(values=chisp4) +
  theme_classic() +
  theme(
    legend.text = element_text(size = 10),
    legend.position = "none",
    legend.title = element_blank(),
    axis.text = element_text(size = 10),
    axis.title.y = element_text(size = 12),
    axis.title.x = element_text(size = 12))

#save for ease of review; if you include in manuscript, will need to save as an eps file (sample code below)
png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/SDI_by_clust.png",
    width = 12, height = 9, units = 'in', res = 600)
shan
dev.off()

#save as .eps file to comply with Lancet Microbe requirements
ggsave(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/SDI_by_clust.eps", width = 7, height = 3.5, units = "in")
shan
dev.off()

```

#### Cluster composition

```{r}
#Cluster composition
sample_data(chi.spec)$cluster = as.factor(data.cluster)

psC1 = subset_samples(chi.spec, sample_data(chi.spec)$cluster == 1)
speciesDF = as.data.frame(cbind((otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))
rownames(speciesDF) = NULL
speciesDF$Kingdom = NULL
speciesDF$Phylum=NULL
speciesDF$Class=NULL
speciesDF$Order=NULL
speciesDF$Family=NULL
row.names(speciesDF) <-  speciesDF$Species #keep getting an error message: "Warning: non-unique value when setting 'row.names': ‘NA;g;s’
  #Error in `.rowNamesDF<-`(x, value = value) : 
  #duplicate 'row.names' are not allowed
#may be due to the unnamed species in the table; will remove them and try again
speciesDF$Species <- gsub(";", NA, speciesDF$Species)
speciesDF <- speciesDF %>% drop_na(Species) #Go from 6709 --> 6162 rows
speciesDF$Genus=NULL
speciesDF$Species = NULL
speciesDF = mutate_all(speciesDF, function(x) as.numeric(as.character(x)))

speciesDF$C1 = rowMeans(speciesDF)

AvgDF1 = speciesDF$C1

#now repeat this process for clusters 2-5, saving AvgDF2-5 in the process so we can bind them all together at the end

psC1 = subset_samples(chi.spec, sample_data(chi.spec)$cluster == 2)
speciesDF = as.data.frame(cbind((otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))
speciesDF$Species <- gsub(";", NA, speciesDF$Species)
speciesDF <- speciesDF %>% drop_na(Species) #Go from 6709 --> 6162 rows
row.names(speciesDF) <-  speciesDF$Species 
speciesDF$Kingdom = NULL
speciesDF$Phylum=NULL
speciesDF$Class=NULL
speciesDF$Order=NULL
speciesDF$Family=NULL
speciesDF$Genus=NULL
speciesDF$Species = NULL
speciesDF = mutate_all(speciesDF, function(x) as.numeric(as.character(x)))
speciesDF$C1 = rowMeans(speciesDF)
AvgDF2 = speciesDF$C1

psC1 = subset_samples(chi.spec, sample_data(chi.spec)$cluster == 3)
speciesDF = as.data.frame(cbind((otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))
speciesDF$Species <- gsub(";", NA, speciesDF$Species)
speciesDF <- speciesDF %>% drop_na(Species) #Go from 6709 --> 6162 rows
row.names(speciesDF) <-  speciesDF$Species 
speciesDF$Kingdom = NULL
speciesDF$Phylum=NULL
speciesDF$Class=NULL
speciesDF$Order=NULL
speciesDF$Family=NULL
speciesDF$Genus=NULL
speciesDF$Species = NULL
speciesDF = mutate_all(speciesDF, function(x) as.numeric(as.character(x)))
speciesDF$C1 = rowMeans(speciesDF)
AvgDF3 = speciesDF$C1

psC1 = subset_samples(chi.spec, sample_data(chi.spec)$cluster == 4)
speciesDF = as.data.frame(cbind((otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))
speciesDF$Species <- gsub(";", NA, speciesDF$Species)
speciesDF <- speciesDF %>% drop_na(Species) #Go from 6709 --> 6162 rows
row.names(speciesDF) <-  speciesDF$Species 
speciesDF$Kingdom = NULL
speciesDF$Phylum=NULL
speciesDF$Class=NULL
speciesDF$Order=NULL
speciesDF$Family=NULL
speciesDF$Genus=NULL
speciesDF$Species = NULL
speciesDF = mutate_all(speciesDF, function(x) as.numeric(as.character(x)))
speciesDF$C1 = rowMeans(speciesDF)
AvgDF4 = speciesDF$C1

psC1 = subset_samples(chi.spec, sample_data(chi.spec)$cluster == 5)
speciesDF = as.data.frame(cbind((otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))
speciesDF$Species <- gsub(";", NA, speciesDF$Species)
speciesDF <- speciesDF %>% drop_na(Species) #Go from 6709 --> 6162 rows
row.names(speciesDF) <-  speciesDF$Species 
speciesDF$Kingdom = NULL
speciesDF$Phylum=NULL
speciesDF$Class=NULL
speciesDF$Order=NULL
speciesDF$Family=NULL
speciesDF$Genus=NULL
speciesDF$Species = NULL
speciesDF = mutate_all(speciesDF, function(x) as.numeric(as.character(x)))
speciesDF$C1 = rowMeans(speciesDF)
AvgDF5 = speciesDF$C1

#Now combine all your row means and pull out the 10 most common species for consistency with our other species barplots
AvgDF = cbind(AvgDF1, AvgDF2, AvgDF3, AvgDF4, AvgDF5) %>%
  data.frame(.)
rownames(AvgDF) = rownames(speciesDF)
AvgDF$Avg = rowMeans(AvgDF)

#taking the 10 most common based on the averages across profiles
fig = top_n(AvgDF, 10, Avg) 
Other = 1 - colSums(fig)
fig = rbind(fig, Other)
rownames(fig)[rownames(fig) == "11"] <- "Other"
fig$tax = rownames(fig)

fig = dplyr::select(fig, -Avg) %>%
  dplyr::rename('1' = AvgDF1,
                '2' = AvgDF2,
                '3' = AvgDF3,
                '4' = AvgDF4,
                '5' = AvgDF5)

library(reshape2)
f = melt(fig) 

f$tax = factor(f$tax, levels=c("Corynebacterium accolens", "Corynebacterium propinquum", "Corynebacterium pseudodiphtheriticum", "Dolosigranulum pigrum","Haemophilus influenzae","Moraxella catarrhalis", "Moraxella lincolnii", "Moraxella nonliquefaciens", "Staphylococcus aureus","Streptococcus pneumoniae","Other"))

#rename tax so it looks nicer in the table legend
colnames(f)[colnames(f) == "tax"] <- "Species"

clust_plot <- ggplot(f[order(f$Species, decreasing = TRUE),], aes(x=variable, y=value, fill=Species)) + 
  geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text=element_markdown(),
  legend.title = element_text(size=10),
  axis.title.y = element_text(size=10, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_text(size=10), axis.text.x = element_text(size=10, vjust = 1, hjust = 1, colour = "black")) + scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  xlab("Microbiome profile") + ylab("Mean relative abundance") 

clust_plot

#save for ease of review; if you include in manuscript, will need to save as an eps file (sample code below)
png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/cluster_rel_abund.png",
    width = 12, height = 9, units = 'in', res = 600)
clust_plot
dev.off()

#NOW: use cowplot to combine our PCoA plot and composition bar plot into one figure
fig5 <- plot_grid(clust, clust_plot, labels = c('A', 'B'), align = "h", axis = "b", label_size = 12, rel_widths = c(1, 1.5))
fig5

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig3NEW(clust).pdf", plot = fig5, dpi = 300, width = 11, height = 3.5)

#UPDATED 9-22-23 with smaller axes and legend fonts per MK:
ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig3NEW(clust)_V2.pdf", plot = fig5, dpi = 300, width = 11, height = 3.5)

# #save as .eps file to comply with Lancet Microbe requirements
# ggsave(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig5_clust.eps", width = 11, height = 3.5, units = "in")
# fig5
# dev.off()

#Save our cluster composition + relative abundances as a csv file
write.csv(fig, '/Users/swetapatel/OneDrive - Duke University/Fogarty coding/cluster_comp.csv')

```

#### What do our clusters look like by immune status among CLWH?

We originally repeated the clustering process with only CLWH and ended up with 6 clusters, but from d/w MK we should complete all sub-analyses using our original 5 clusters. Thus, all of the CLWH-specific clustering code was deleted from this version and we will start with the csv file we saved previously that includes a cluster variable. We will subset out our CLWH, generate the relevant variables (vL, CD4), and look for associations between HIV-specific variables and clusters among CLWH.

```{r}
sampleDF <- read.csv("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/metadata_clust.csv")

#subset out CLWH
chi_hiv <- subset(sampleDF, hiv2 == "CLWH") #so we can use our original code for variable creation

#Now create variables for viral load and CD4
#I want the most recent viral load values for my analysis. Will try creating a new variable to ID which value is more recent first:
#missing 3 vl1 and 14vl2's
chi_hiv$newest_vL <- NA
chi_hiv$newest_vL [chi_hiv$child_vl1_weeks < chi_hiv$child_vl2_weeks] <- 1
chi_hiv$newest_vL [chi_hiv$child_vl2_weeks < chi_hiv$child_vl1_weeks] <- 2
chi_hiv$newest_vL [chi_hiv$child_vl2_weeks < 0] <- 1  #we want to include the viral loads collected BEFORE enrollment where possible
chi_hiv$newest_vL [is.na(chi_hiv$child_vl_2)] <- 1    #if we are missing vl_2, we will use vl_1
chi_hiv$newest_vL [is.na(chi_hiv$child_vl_1) & is.na(chi_hiv$child_vl_2)] <- NA
table(chi_hiv$newest_vL)

#Next, can we pull the viral load that was most recent using our newest_vL variable?
  #source: https://stackoverflow.com/questions/67944727/creating-new-variable-by-selecting-column-based-on-value-of-another-column
  #Need to do this with rows that have values (aka need to exclude the NA viral load children for this to work)
#For future phyloseq analyses: will need to remove B-04, B-07, B-52
chi_viral <- subset(chi_hiv, !is.na(newest_vL))
chi_viral$Var1 <- chi_viral$child_vl_1
chi_viral$Var2 <- chi_viral$child_vl_2

chi_viral <- chi_viral %>% rowwise() %>%
  mutate(newest_vlnum = get(paste0('Var', newest_vL)))
#now we have a dataframe of the 41 children with viral load measurements and 1 column containing the most recent viral loads for each
summary(chi_viral$newest_vlnum)
table(chi_viral$newest_vlnum) #27 with viral suppression, 14 without viral suppression

#make new variable for viral suppression or not, with vL < 400 = suppressed
chi_viral$vl_suppr <- NA
chi_viral$vl_suppr [chi_viral$newest_vlnum < 400] <- "1"
chi_viral$vl_suppr [chi_viral$newest_vlnum >= 400] <- "0"
table(chi_viral$vl_suppr)

#REPEAT the above steps to get date associated with most recent viral load, most recent CD4, and date a/w most recent CD4
#Viral load collection date
chi_viral$Var1 <- chi_viral$child_vl1_weeks
chi_viral$Var2 <- chi_viral$child_vl2_weeks

chi_viral <- chi_viral %>% rowwise() %>%
  mutate(newest_vldate = get(paste0('Var', newest_vL)))   #now have info for when vL were checked in relation to enrollment

summary(chi_viral$newest_vldate)

chi_viral <- within(chi_viral, rm(Var1, Var2))

#CD4
  #missing CD4 data for B-45 and B-52
  #will focus on percentages since this is what is presented for kids
chi_hiv$newest_cd4 <- NA
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks1 < chi_hiv$chi_cd4_weeks2] <- 1
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks2 < chi_hiv$chi_cd4_weeks1] <- 2
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks2 < 0] <- 1  #we want to include the viral loads collected BEFORE enrollment where possible
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_2)] <- 1    #if we are missing cd4_2, we will use cd4_1
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_1)] <- 2    #one child is missing cd4_1 but has cd4_2
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_1) & is.na(chi_hiv$chi_cd4_perc_2)] <- NA
table(chi_hiv$newest_cd4)

#create new dataframe without B-45 and B-52 (aka new df with children missing CD4 data removed)
chi_cd4 <- subset(chi_hiv, !is.na(newest_cd4))
chi_cd4$Var1 <- chi_cd4$chi_cd4_perc_1
chi_cd4$Var2 <- chi_cd4$chi_cd4_perc_2

chi_cd4 <- chi_cd4 %>% rowwise() %>%
  mutate(newest_cd4perc = get(paste0('Var', newest_cd4)))

summary(chi_cd4$newest_cd4perc)
table(chi_cd4$newest_cd4perc)

#make new variable for immune competent CD4 or not, with CD4 percentage >=25% = immune competent
  #d/w MK re: potentially adding a cutoff at 15% for immunosuppression, but we only have 2 kids with CD4% < 15 so will stay with 25% binary for now
chi_cd4$cd4nL <- NA
chi_cd4$cd4nL [chi_cd4$newest_cd4perc >=25] <- "1"
chi_cd4$cd4nL [chi_cd4$newest_cd4perc < 25] <- "0"
table(chi_cd4$cd4nL) 

#arvs
table(chi_hiv$child_arv)
table(chi_hiv$child_arv_meds)
```

- Do any of our clusters associate with age or HIV-specific covariables (viral suppression, cd4 category, TMP-SMX use, abx use)?

Will create plots first to visually assess for associations.

```{r}
#do our clusters associate with any of our demographic variables?
#age by cluster
chi_hiv$cluster <- as.factor(chi_hiv$cluster)
a <- ggplot(chi_hiv, aes(x=cluster, y=age)) +
  geom_boxplot(fill=chisp4) +
  xlab('Cluster') + 
  ylab('Age (Months)') +
  theme_classic()

#viral suppression by cluster
b <- ggplot(chi_viral, aes(fill=vl_suppr, x=cluster)) + geom_bar(stat = "count", position = "fill", width = 0.9) + 
  scale_fill_manual(values = c('#800000FF','#155F83FF'))+
  xlab('Cluster')+
  theme_classic()

#CD4 category by cluster
c <- ggplot(chi_cd4, aes(fill=cd4nL, x=cluster)) + geom_bar(stat = "count", position = "fill", width = 0.9) + 
  scale_fill_manual(values = c('#800000FF','#155F83FF'))+
  xlab('Cluster')+
  theme_classic()

#TMP-SMX use by cluster
chi_hiv$tmpsmx <- NA
chi_hiv$tmpsmx [chi_hiv$child_tmpsmx == 1] <- "Yes"
chi_hiv$tmpsmx[chi_hiv$child_tmpsmx == 2] <- "No"
table(chi_hiv$tmpsmx)

d <- ggplot(chi_hiv, aes(fill=tmpsmx, x=cluster)) + geom_bar(stat = "count", position = "fill", width = 0.9) + 
  scale_fill_manual(values = c('#800000FF','#155F83FF'))+
  xlab('Cluster')+
  theme_classic()

#abx use by cluster
e <- ggplot(chi_hiv, aes(fill=abx, x=cluster)) + geom_bar(stat = "count", position = "fill", width = 0.9) +
  scale_fill_manual(values = c('#800000FF','#155F83FF'))+
  xlab('Cluster')+
  theme_classic()

#save as an image for reference while we do additional analyses:
clust_hivcomb <- plot_grid(a,b,c,d,e, labels = "AUTO", nrow = 3, ncol = 2, align = "hv", axis = "b")

clust_hivcomb

png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/clwh_clust5bar.png",
    width = 12, height = 9, units = 'in', res = 600)
clust_hivcomb
dev.off()

#Visual trends: cluster 3 has the highest number of viremic kids. Cluster 5 only has immunocompetent children, and clusters 1 and 4 have the highest number of virally suppressed kids and the second highest number of immunocompetent kids. We will check comparisons of 3 vs others, 1-4 vs others, 2-3 vs others, and 5 vs others. Will also look at 1-5 and 1 only for CD4 since these are our Coryne heavy groups (and 1 has the most C. pseudodiphtheriticum)

#create these variables (will make binary in anticipation of logistic regression analyses)
#create in both chi_viral and chi_cd4 dataframes (since we have a different N in each)
chi_viral$clust1 <- NA
chi_viral$clust1 [chi_viral$cluster == "1"] <- 1
chi_viral$clust1[chi_viral$cluster != "1"] <- 0
table(chi_viral$clust1)

chi_viral$clust3 <- NA
chi_viral$clust3 [chi_viral$cluster == "3"] <- 1
chi_viral$clust3[chi_viral$cluster != "3"] <- 0
table(chi_viral$clust3)

chi_viral$clust5 <- NA
chi_viral$clust5 [chi_viral$cluster == "5"] <- 1
chi_viral$clust5[chi_viral$cluster != "5"] <- 0
table(chi_viral$clust5)

chi_viral$clust23 <- NA
chi_viral$clust23 [chi_viral$cluster == "2" | chi_viral$cluster == "3"] <- 1
chi_viral$clust23[chi_viral$cluster == "1" | chi_viral$cluster == "4" | chi_viral$cluster == "5"] <- 0
table(chi_viral$clust23)

chi_viral$clust14 <- NA
chi_viral$clust14 [chi_viral$cluster == "1" | chi_viral$cluster == "4"] <- 1
chi_viral$clust14[chi_viral$cluster == "2" | chi_viral$cluster == "3" | chi_viral$cluster == "5"] <- 0
table(chi_viral$clust14)

chi_viral$clust15 <- NA
chi_viral$clust15 [chi_viral$cluster == "1" | chi_viral$cluster == "5"] <- 1
chi_viral$clust15 [chi_viral$cluster == "2" | chi_viral$cluster == "3" |chi_viral$cluster == "4"] <- 0
table(chi_viral$clust15)

chi_cd4$clust1 <- NA
chi_cd4$clust1 [chi_cd4$cluster == "1"] <- 1
chi_cd4$clust1[chi_cd4$cluster != "1"] <- 0
table(chi_cd4$clust1)

chi_cd4$clust3 <- NA
chi_cd4$clust3 [chi_cd4$cluster == "3"] <- 1
chi_cd4$clust3[chi_cd4$cluster != "3"] <- 0
table(chi_cd4$clust3)

chi_cd4$clust5 <- NA
chi_cd4$clust5 [chi_cd4$cluster == "5"] <- 1
chi_cd4$clust5[chi_cd4$cluster != "5"] <- 0
table(chi_cd4$clust5)

chi_cd4$clust23 <- NA
chi_cd4$clust23 [chi_cd4$cluster == "2" | chi_cd4$cluster == "3"] <- 1
chi_cd4$clust23[chi_cd4$cluster == "1" | chi_cd4$cluster == "4" | chi_cd4$cluster == "5"] <- 0
table(chi_cd4$clust23)

chi_cd4$clust14 <- NA
chi_cd4$clust14 [chi_cd4$cluster == "1" | chi_cd4$cluster == "4"] <- 1
chi_cd4$clust14[chi_cd4$cluster == "2" | chi_cd4$cluster == "3" | chi_cd4$cluster == "5"] <- 0
table(chi_cd4$clust14)

chi_cd4$clust15 <- NA
chi_cd4$clust15 [chi_cd4$cluster == "1" | chi_cd4$cluster == "5"] <- 1
chi_cd4$clust15 [chi_cd4$cluster == "2" | chi_cd4$cluster == "3" |chi_cd4$cluster == "4"] <- 0
table(chi_cd4$clust15)
```

Next, statistical testing to identify demographic differences by cluster among CLWH

```{r}
table(chi_hiv$cluster)

###DEMOGRAPHIC DIFFERENCES BY CLUSTER AMONG CLWH
#Age
tapply(chi_hiv$age, chi_hiv$cluster, summary)
kruskal.test(chi_hiv$age, chi_hiv$cluster) #p=0.16

#Sex
table(chi_hiv$sex, chi_hiv$cluster)
prop.table(table(chi_hiv$sex, chi_hiv$cluster), 2)
fisher.test(chi_hiv$sex, chi_hiv$cluster)

#TMP-SMX use
table(chi_hiv$tmpsmx, chi_hiv$cluster)
prop.table(table(chi_hiv$tmpsmx, chi_hiv$cluster), 2)
fisher.test(chi_hiv$tmpsmx, chi_hiv$cluster)

#Abx use
table(chi_hiv$abx, chi_hiv$cluster)
prop.table(table(chi_hiv$abx, chi_hiv$cluster), 2)
fisher.test(chi_hiv$abx, chi_hiv$cluster)

#Viral suppression
table(chi_viral$vl_suppr, chi_viral$cluster)
prop.table(table(chi_viral$vl_suppr, chi_viral$cluster), 2)
fisher.test(chi_viral$vl_suppr, chi_viral$cluster)

#CD4 status #p=0.09
table(chi_cd4$cd4nL, chi_cd4$cluster)
prop.table(table(chi_cd4$cd4nL, chi_cd4$cluster), 2)
fisher.test(chi_cd4$cd4nL, chi_cd4$cluster)

#2 group analysis by CD4 and viral suppression status: 
#viral suppression
#by cluster 1: p=0.27
table(chi_viral$vl_suppr, chi_viral$clust1)
prop.table(table(chi_viral$vl_suppr, chi_viral$clust1), 2)
fisher.test(chi_viral$vl_suppr, chi_viral$clust1)

#by clusters 1&4: p = 0.02
table(chi_viral$vl_suppr, chi_viral$clust14)
prop.table(table(chi_viral$vl_suppr, chi_viral$clust14), 2)
fisher.test(chi_viral$vl_suppr, chi_viral$clust14)

#by clusters 1&5: p = 1 (not surprising based on bar plots)
table(chi_viral$vl_suppr, chi_viral$clust15)
prop.table(table(chi_viral$vl_suppr, chi_viral$clust15), 2)
fisher.test(chi_viral$vl_suppr, chi_viral$clust15)

#by cluster 3: p = 0.26
table(chi_viral$vl_suppr, chi_viral$clust3)
prop.table(table(chi_viral$vl_suppr, chi_viral$clust3), 2)
fisher.test(chi_viral$vl_suppr, chi_viral$clust3)

#by cluster 5: p=0.32
table(chi_viral$vl_suppr, chi_viral$clust5)
prop.table(table(chi_viral$vl_suppr, chi_viral$clust5), 2)
fisher.test(chi_viral$vl_suppr, chi_viral$clust5)

#low vs normal CD4%
#by cluster 1: p=0.25
table(chi_cd4$cd4nL, chi_cd4$clust1)
prop.table(table(chi_cd4$cd4nL, chi_cd4$clust1), 2)
fisher.test(chi_cd4$cd4nL, chi_cd4$clust1)

#by clusters 1&4: p = 0.08
table(chi_cd4$cd4nL, chi_cd4$clust14)
prop.table(table(chi_cd4$cd4nL, chi_cd4$clust14), 2)
fisher.test(chi_cd4$cd4nL, chi_cd4$clust14)

#by clusters 1&5: p = 0.06
table(chi_cd4$cd4nL, chi_cd4$clust15)
prop.table(table(chi_cd4$cd4nL, chi_cd4$clust15), 2)
fisher.test(chi_cd4$cd4nL, chi_cd4$clust15)

#by cluster 3: p = 0.28
table(chi_cd4$cd4nL, chi_cd4$clust3)
prop.table(table(chi_cd4$cd4nL, chi_cd4$clust3), 2)
fisher.test(chi_cd4$cd4nL, chi_cd4$clust3)

#by cluster 5: p=0.30
table(chi_cd4$cd4nL, chi_cd4$clust5)
prop.table(table(chi_cd4$cd4nL, chi_cd4$clust5), 2)
fisher.test(chi_cd4$cd4nL, chi_cd4$clust5)

#by cluster 2 or 3: p = 0.004
table(chi_cd4$cd4nL, chi_cd4$clust23)
prop.table(table(chi_cd4$cd4nL, chi_cd4$clust23), 2)
fisher.test(chi_cd4$cd4nL, chi_cd4$clust23)

#Will not do any multivariable testing because our sample size is too small
```

### Co-occurrence plots (aka network analysis) to characterize interspecies interactions/associations 

- Got reviewer request for the following: "I suspect you will have a multiple associations with higher Corynebacterium such as an intact immune system, and a negative association with Staph/Strep (which will be highly associated with CD4+ count). Please see co-occurence plots built with networks of spearman correlations (PMID: 30093571)."

PMID links to a paper by Leo Segal's group: https://erj.ersjournals.com/content/52/4/1800810.long#sec-2.
Their description of the network correlation methods is as follows: "Co-occurrence between most abundant bacterial genera (>1% relative abundance in at least one sample) were assessed using SparCC [30] with 20 iterations and 500 bootstrap replicates to eliminate correlations where significance was driven by outliers and visualised using Cytoscape v3.0.2 [31]. Only biomarkers that passed false discovery rate correction were used for this analysis [32]."
- code used in paper: https://github.com/segalmicrobiomelab/ntm_bronchiectasis_microbiome/blob/master/NTM.r --> this does not have any of the correlation network code, which is not helpful.

Found multiple SpiecEasi tutorials: 
1. https://biovcnet.github.io/_pages/NetworkScience_SparCC.nb.html (helps you run sparcc and generate file that you can plug into cytoscape)
2. https://mibwurrepo.github.io/Microbial-bioinformatics-introductory-course-Material-2018/inference-of-microbial-ecological-networks.html (looks like you can create your network diagram within R itself)
3. SpiecEasi package github (with newest code, looks like): 

Note: installing SpiecEasi on a mac was a beast and required the following steps (recording here in case I need to install another package requiring compliation in the future). Took ~2.5h troubleshooting.
1. Downloaded gfortran from CRAN: https://cran.r-project.org/bin/macosx/tools/
2. This defaults to storing gfortran in the /opt/ folder in Terminal, but R looks for it in usr/local/; first I tried manually moving gfortran into usr/local using the sudo mv command in Terminal, but this still did not work
3. Created a Makevars file in the ~/.R folder and then pasted the following text into it, using a combination of these tutorials
- creating file and adding contents to it in Terminal: https://superuser.com/questions/728800/how-to-create-a-file-and-insert-a-line-in-it-using-os-x-terminal
- getting R to find gfortran (?I think this is what this code did):
https://stackoverflow.com/questions/69639782/installing-gfortran-on-macbook-with-apple-m1-chip-for-use-in-r
- what I added to terminal: 
Swetas-MacBook-Pro-3:.R swetapatel$ echo "FC = /opt/gfortran/bin/gfortran">> ~/.R/Makevars
Swetas-MacBook-Pro-3:.R swetapatel$ echo "F77 = /opt/gfortran/bin/gfortran" >> ~/.R/Makevars
Swetas-MacBook-Pro-3:.R swetapatel$ echo "FLIBS = -L/opt/gfortran/lib/gcc/x86_64-apple-darwin20.0" >> ~/.R/Makevars
4. Restarted R and installed SpiecEasi using "install_github("zdk123/SpiecEasi")"

**SPIEC EASI**
- will use counts, not relative abundances (source: https://github.com/zdk123/SpiecEasi/issues/64)
- will use our chi.prune pseq object (already tax.glom'd by species) and for consistency, we will use the same filtering criteria that we used for our Maaslin2 analyses (filter taxa with fewer than 50 counts across the dataset).

```{r}
#Filter and remove unidentified species 
chi.filter <- filter_taxa(chi_prune, function(Abundance) mean(Abundance)>=50, TRUE)
ntaxa(chi.filter) #43
otu <- as.data.frame(otu_table(chi.filter, taxa_are_rows=TRUE))
#need to remove the unidentified species from here: row 11, 25, 37, 38, 40
  #source: https://stackoverflow.com/questions/12328056/how-do-i-delete-rows-in-a-data-frame
#corresponding respectively to: Corynebacterium sp. KPL1859, Streptococcus sp. SK643, Alkalibacterium sp. 20, uncultured Clostridium sp., Nocardioides sp. Root122
otu <- otu[-c(11, 25, 37, 38, 40),]

#already set.seed at the beginning of our R session to 1234
#Spiec easi package in R has wrappers to work directly with pseq objects, so we will import our new otu table back into chi.filter and give it a shot
#import updated otutable into pseq object (convert to matrix first) and take a look
TT <- as.matrix(otu)
otu2 <- otu_table(TT, taxa_are_rows = TRUE)
class(otu2)
otu_table(chi.filter) <- otu2
#can we test it to see if it worked? 
test <- as.data.frame(otu_table(chi.filter, taxa_are_rows=TRUE)) #it worked
remove(TT, test, otu2)

#Spiec Easi
se.mb.chi1 <- spiec.easi(chi.filter, method='mb', lambda.min.ratio=1e-2, nlambda=20, pulsar.params=list(rep.num=500))
ig2.mb <- adj2igraph(getRefit(se.mb.chi1), vertex.attr = list(name=taxa_names(chi.filter)))
test <- plot_network(ig2.mb, chi.filter, type = 'taxa', color="Genus")
#I made a network! pretty cool. for some reason the left side keeps getting cut off, even with resizing. Saved as screenshot and a file:
png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/se_mb.png",
    width = 12, height = 9, units = 'in', res = 600)
test
dev.off()

#We should also see what our networks look like using glasso and sparcc (with the latter used in the Segal paper)
#glasso
se.gl.chi1 <- spiec.easi(chi.filter, method='glasso', lambda.min.ratio=1e-2, nlambda=20, pulsar.params=list(rep.num=500))
ig2.gl <- adj2igraph(getRefit(se.gl.chi1), vertex.attr = list(name=taxa_names(chi.filter)))
plot_network(ig2.gl, chi.filter, type = 'taxa', color="Genus") #screenshotted again

#sparcc: can't run from pseq object. Looks like we run using the otu table(?)
sparcc.chi1 <- sparcc(otu) #ERROR; need to transpose the OTU table so OTUs are columns and samples are rows
totu <- t(otu)
sparcc.chi1 <- sparcc(totu)
## Define arbitrary threshold for SparCC correlation matrix for the graph
sparcc.graph <- abs(sparcc.chi1$Cor) >= 0.3
diag(sparcc.graph) <- 0
library(Matrix)
sparcc.graph <- Matrix(sparcc.graph, sparse=TRUE)
## Create igraph objects
ig.sparcc <- adj2igraph(sparcc.graph)

## set size of vertex proportional to clr-mean
vsize    <- rowMeans(clr(otu, 1))+6
plot(ig.sparcc, vertex.size=vsize, vertex.label=NA, main="sparcc")

#We can evaluate the weights on edges networks using the terms from the underlying model. SparCC correlations can be used directly.

library(Matrix)
elist.sparcc <- summary(sparcc.graph*sparcc.chi1$Cor)
hist(elist.sparcc[,3], main='', xlab='edge weights')

#Lets look at the degree statistics from the sparcc network.
library(igraph)
dd.sparcc <- degree.distribution(ig.sparcc)

plot(0:(length(dd.sparcc)-1), dd.sparcc, ylim=c(0,.35), type='b',
      ylab="Frequency", xlab="Degree", main="Degree Distributions")

#Not sure about this quality; will use tutorial from biovcnet github
sparcc.chi1$Cor[1:5, 1:5]
#threw away names; will add back from otu dataframe
rownames(sparcc.chi1$Cor) <- colnames(totu)
colnames(sparcc.chi1$Cor) <- colnames(totu)
rownames(sparcc.chi1$Cov) <- colnames(totu)
colnames(sparcc.chi1$Cov) <- colnames(totu)
#look to confirm
sparcc.chi1$Cor[1:5, 1:5]

#Plotting SparCC correlation
#first generate helper function to get upper triangle of the correlation matrix
get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
}
#helper function to reorder matrix (?)
reorder_cormat <- function(cormat){
# Use correlation between variables as distance
dd <- as.dist((1-cormat)/2)
hc <- hclust(dd)
cormat <-cormat[hc$order, hc$order]
}

plotableSparcc <- sparcc.chi1$Cor %>% reorder_cormat %>% get_upper_tri() %>% reshape2::melt() %>% na.omit()
Sparcc_plot <- plotableSparcc %>% ggplot(aes(x = Var2, y = Var1, fill = value)) + geom_tile() + scale_fill_gradient2() + theme(axis.text.x = element_text(angle = 90, hjust = 1))
Sparcc_plot

#save sparcc plot
png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/sparcc_tile.png",
    width = 12, height = 9, units = 'in', res = 600)
Sparcc_plot
dev.off()

#calculate sparcc p values
#first have to boostrap the sparccc value
tp0 <- proc.time()
out2 <- sparccboot(totu, R = 1000)
tp1 <- proc.time()
tp1 - tp0

#calculate p values from the bootstrapped values
outP <- pval.sparccboot(out2)
data.frame(outP$cors, outP$pvals) %>% head

#these values are hard to use because they don’t tell you which p-values correspond to which species; source to fix: https://github.com/zdk123/SpiecEasi/issues/17

cors <- outP$cors
pvals <- outP$pvals
sparCCpcors <- diag(0.5, nrow = dim(sparcc.chi1$Cor)[1], ncol = dim(sparcc.chi1$Cor)[1])
sparCCpcors[upper.tri(sparCCpcors, diag=FALSE)] <- cors
sparCCpcors <- sparCCpcors + t(sparCCpcors)

sparCCpval <- diag(0.5, nrow = dim(sparcc.chi1$Cor)[1], ncol = dim(sparcc.chi1$Cor)[1])
sparCCpval[upper.tri(sparCCpval, diag=FALSE)] <- pvals
sparCCpval <- sparCCpval + t(sparCCpval)

rownames(sparCCpcors) <- colnames(totu)
colnames(sparCCpcors) <- colnames(totu)
rownames(sparCCpval) <- colnames(totu)
colnames(sparCCpval) <- colnames(totu)

sparCCpcors[1:5, 1:5]
sparCCpval[1:5, 1:5]

#now need to process and account for false discovery rates 
#add helper function first
reorder_cor_and_p <- function(cormat, pmat){
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
  pmat <- pmat[hc$order, hc$order]
  list(r = cormat, p = pmat)
}

reordered_all_sparcc <- reorder_cor_and_p(sparCCpcors, sparCCpval)
reordered_sparccCor <- reordered_all_sparcc$r
reordered_sparccP<- reordered_all_sparcc$p

sparccCor_processed <- reordered_sparccCor  %>% get_upper_tri() %>% reshape2::melt() %>% na.omit() %>% rename(cor = value)
sparccP_processed <- reordered_sparccP  %>% get_upper_tri() %>% reshape2::melt() %>% na.omit() %>% rename(p = value)

# join the two data frames

SparccP <- left_join(sparccCor_processed, sparccP_processed, by = c("Var1", "Var2")) %>%
  # # remove self correlations
  # filter(Var1 != Var2) %>% 
  # calculate the false discovery rate to adjust for multiple p values
  mutate(fdr = p.adjust(p, method = "BH"))

#what is our false discovery rate threshold? for Maaslin2, we used a q value of 0.2; here, given the high number of pairwise correlations, we can try 0.05 and see how that works.
fdrThresh <- 0.05 # fdr threshold

sparccOkP <- SparccP%>% filter(fdr < fdrThresh) 

SparccP_plot <- SparccP %>% ggplot(aes(x = Var2, y = Var1, fill = cor)) + geom_tile() + scale_fill_gradient2(low = "#800000FF", mid = "white", high = "#4D004B", midpoint = 0) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + geom_point(data = sparccOkP, shape = 8, size = 0.5)

SparccP_plot

png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/sparccP.png",
    width = 12, height = 9, units = 'in', res = 600)
SparccP_plot
dev.off()

#what does it look like if we use an fdr of 0.1? 
fdrThresh2 <- 0.1 # fdr threshold

sparccOkP2 <- SparccP%>% filter(fdr < fdrThresh2) 

SparccP_plot2 <- SparccP %>% ggplot(aes(x = Var2, y = Var1, fill = cor)) + geom_tile() + scale_fill_gradient2(low = "#800000FF", mid = "white", high = "#4D004B", midpoint = 0) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + geom_point(data = sparccOkP2, shape = 8, size = 0.5)

SparccP_plot2

png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/sparccP2.png",
    width = 12, height = 9, units = 'in', res = 600)
SparccP_plot2
dev.off()

#save our SparccP dataframe as a csv file
write.csv(SparccP, "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/SparCC.csv")
test <- read.csv("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/SparCC.csv") #looks reasonable
remove(test)

#Next: will import into cytoscape to generate network figure that includes detail on negative vs positive assoc and strength of association
#Great cytoscape tutorial on youtube by the same people who created the sparcc tutorial I used: https://www.youtube.com/watch?v=f67jKjOZJaQ
#FOR EASE OF VISUALIZATION: manually changed the taxonomy names in our sparCC csv file for the following species:
#Corynebacterium sp. KPL1995 -> Corynebacterium pseudodiphtheriticum
#Corynebacterium sp. KPL1986 -> Corynebacterium accolens
#Haemophilus sp. C1 -> Haemophilus haemolyticus

#then imported the updated sparCC csv file into cytoscape

#What does our corrected taxonomy sparCC csv file look like in R? Can we use it to regenerate our nice tile plot?
spartax <- read.csv("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/SparCC.csv")
spartax$X <- NULL
sparccOkP2 <- spartax %>% filter(fdr < fdrThresh2) 
SparccP_plot3 <- spartax %>% ggplot(aes(x = Var2, y = Var1, fill = cor)) + geom_tile() + scale_fill_gradient2(low = "#800000FF", mid = "white", high = "#4D004B", midpoint = 0) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + geom_point(data = sparccOkP2, shape = 8, size = 0.5)


#not working; probably best to fix the otu table and redo all steps to get a figure we can include in the supplement that has correct taxonomy names
#sparcc: run from otu table but need to update row names first
row.names(otu)[3] <- "Haemophilus haemolyticus"
row.names(otu)[10] <- "Corynebacterium pseudodiphtheriticum"
row.names(otu)[11] <- "Corynebacterium accolens"

totu <- t(otu)
sparcc.chi1 <- sparcc(totu)

sparcc.chi1$Cor[1:5, 1:5]
#threw away names; will add back from otu dataframe
rownames(sparcc.chi1$Cor) <- colnames(totu)
colnames(sparcc.chi1$Cor) <- colnames(totu)
rownames(sparcc.chi1$Cov) <- colnames(totu)
colnames(sparcc.chi1$Cov) <- colnames(totu)
#look to confirm
sparcc.chi1$Cor[1:5, 1:5]

#Plotting SparCC correlation
#first generate helper function to get upper triangle of the correlation matrix
get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
}
#helper function to reorder matrix (?)
reorder_cormat <- function(cormat){
# Use correlation between variables as distance
dd <- as.dist((1-cormat)/2)
hc <- hclust(dd)
cormat <-cormat[hc$order, hc$order]
}

plotableSparcc <- sparcc.chi1$Cor %>% reorder_cormat %>% get_upper_tri() %>% reshape2::melt() %>% na.omit()
Sparcc_plot <- plotableSparcc %>% ggplot(aes(x = Var2, y = Var1, fill = value)) + geom_tile() + scale_fill_gradient2() + theme(axis.text.x = element_text(angle = 90, hjust = 1))
Sparcc_plot

#save sparcc plot
png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/sparcc_tile_v2.png",
    width = 12, height = 9, units = 'in', res = 600)
Sparcc_plot
dev.off()

#calculate sparcc p values
#first have to boostrap the sparccc value
tp0 <- proc.time()
out2 <- sparccboot(totu, R = 1000)
tp1 <- proc.time()
tp1 - tp0

#calculate p values from the bootstrapped values
outP <- pval.sparccboot(out2)
data.frame(outP$cors, outP$pvals) %>% head

#these values are hard to use because they don’t tell you which p-values correspond to which species; source to fix: https://github.com/zdk123/SpiecEasi/issues/17

cors <- outP$cors
pvals <- outP$pvals
sparCCpcors <- diag(0.5, nrow = dim(sparcc.chi1$Cor)[1], ncol = dim(sparcc.chi1$Cor)[1])
sparCCpcors[upper.tri(sparCCpcors, diag=FALSE)] <- cors
sparCCpcors <- sparCCpcors + t(sparCCpcors)

sparCCpval <- diag(0.5, nrow = dim(sparcc.chi1$Cor)[1], ncol = dim(sparcc.chi1$Cor)[1])
sparCCpval[upper.tri(sparCCpval, diag=FALSE)] <- pvals
sparCCpval <- sparCCpval + t(sparCCpval)

rownames(sparCCpcors) <- colnames(totu)
colnames(sparCCpcors) <- colnames(totu)
rownames(sparCCpval) <- colnames(totu)
colnames(sparCCpval) <- colnames(totu)

sparCCpcors[1:5, 1:5]
sparCCpval[1:5, 1:5]

#now need to process and account for false discovery rates 
#add helper function first
reorder_cor_and_p <- function(cormat, pmat){
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
  pmat <- pmat[hc$order, hc$order]
  list(r = cormat, p = pmat)
}

reordered_all_sparcc <- reorder_cor_and_p(sparCCpcors, sparCCpval)
reordered_sparccCor <- reordered_all_sparcc$r
reordered_sparccP<- reordered_all_sparcc$p

sparccCor_processed <- reordered_sparccCor  %>% get_upper_tri() %>% reshape2::melt() %>% na.omit() %>% rename(cor = value)
sparccP_processed <- reordered_sparccP  %>% get_upper_tri() %>% reshape2::melt() %>% na.omit() %>% rename(p = value)

# join the two data frames

SparccP <- left_join(sparccCor_processed, sparccP_processed, by = c("Var1", "Var2")) %>%
  # # remove self correlations
  # filter(Var1 != Var2) %>% 
  # calculate the false discovery rate to adjust for multiple p values
  mutate(fdr = p.adjust(p, method = "BH"))

#what is our false discovery rate threshold? for Maaslin2, we used a q value of 0.2; here, will use 0.1.
fdrThresh2 <- 0.1 # fdr threshold

sparccOkP2 <- SparccP%>% filter(fdr < fdrThresh2) 

SparccP_plot2b <- SparccP %>% ggplot(aes(x = Var2, y = Var1, fill = cor)) + geom_tile() + scale_fill_gradient2(low = "#800000FF", mid = "white", high = "#4D004B", midpoint = 0) + theme(
  axis.text.x = element_text(angle = 45, hjust = 1),
  axis.title = element_blank()) + geom_point(data = sparccOkP2, shape = 8, size = 0.5) + labs(fill = "Correlation")

SparccP_plot2b

png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/sparccP2b.png",
    width = 12, height = 9, units = 'in', res = 600)
SparccP_plot2b
dev.off()

#save as .eps file in case:
ggsave(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/sparccP2b.eps", width = 9, height = 7, units = "in")
SparccP_plot2b
dev.off()

#save this SparCC output as a csv file so we can compare to our prior csv file (same code, but otu table was not corrected so figures included Corynebacterium sp. KPL1995 Corynebacterium sp. KPL1986 and Haemophilus sp. C1 instead of the correct spp names)
write.csv(SparccP, "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/SparCCredo_cortax.csv")
write.csv(SparccP, "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/SparCCredo_cortax2.csv")
#numbers are very slightly different, probably due to variation inherent to running the program and the bootstrapping; given this, ideally would redo our figures in cytoscape with SparCCredo_cortax. Unfortunately, something about changing the row names in our otu file messed everything up so the sparccredo_cortax.csv weirdly imports into cytoscape with quotations around all species names, despite trying multiple things (exporting as .txt and .tsv files, changing Var1 and Var2 from factors to character variables and then to strings); ultimately fixed this problem by cutting all our data and pasting it into a new sheet.
#Cytoscape file names that were generated with the corrected otu table all contain the term "redo"

#UPDATE 9-12-23: Need to fix our heatmap so it's easier for the reader:
#1. Put organisms in alphabetical order
#2. shift the axes so the tiles are right up against the axes 
#will attempt this with the SparCCredo_cortax2_test file we created (and used to create our correlation network in Cytoscape)
#Sources: reversing axes: https://stackoverflow.com/questions/28391850/reverse-order-of-discrete-y-axis-in-ggplot2

SparccPcsv <- read.csv("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/SparCCredo_cortax2_test.csv")

SparccP_plot2b <- SparccP %>% ggplot(aes(x = Var2, y = Var1, fill = cor)) + geom_tile() + scale_fill_gradient2(low = "#800000FF", mid = "white", high = "#4D004B", midpoint = 0) + theme(
  axis.text.x = element_text(angle = 45, hjust = 1),
  axis.title = element_blank()) + geom_point(data = sparccOkP2, shape = 8, size = 0.5) + labs(fill = "Correlation")
#set FDR
fdrThresh2 <- 0.1 # fdr threshold

sparccOkP2 <- SparccP%>% filter(fdr < fdrThresh2)

SparccP_plot2c <- SparccP %>% ggplot(aes(x = Var2, y = Var1, fill = cor)) + geom_tile() + scale_fill_gradient2(low = "#800000FF", mid = "white", high = "#4D004B", midpoint = 0) + scale_x_discrete(limits=rev, position = "top") + scale_y_discrete(limits=rev, position = "right") + theme(
  axis.text.x = element_text(angle = 45, hjust = 0, vjust = 1),
  axis.title = element_blank()) + geom_point(data = sparccOkP2, shape = 8, size = 0.5) + labs(fill = "Correlation")

#We want a triangular layout of our tiles. Asked ChatGPT:
#1. Convert dataframe to triangular correlation matrix with dplyr, then put that into ggplot2
cor_matrix <- SparccP %>%
  as.matrix() %>%
  as.dist() %>%
  hclust() %>%
  cutree(k = nrow(SparccP)) %>%
  dist() %>%
  as.matrix()
#did not work; trying another way by converting dataframe to long format:
sparcc_test <- pivot_longer(SparccP, cols = everything(), names_to = "var1", values_to = "correlation")
#tried multiple options that did NOT work; think I'll have to redo my sparcc and bootstrapped p values AGAIN and re-order the matrix before i combine it with the bootstrapped p values into a dataframe.
```

**Trying SparCC again**
```{r}
chi.filter <- filter_taxa(chi_prune, function(Abundance) mean(Abundance)>=50, TRUE)
ntaxa(chi.filter) #43
otu <- as.data.frame(otu_table(chi.filter, taxa_are_rows=TRUE))
#need to remove the unidentified species from here: row 11, 25, 37, 38, 40
  #source: https://stackoverflow.com/questions/12328056/how-do-i-delete-rows-in-a-data-frame
#corresponding respectively to: Corynebacterium sp. KPL1859, Streptococcus sp. SK643, Alkalibacterium sp. 20, uncultured Clostridium sp., Nocardioides sp. Root122
otu <- otu[-c(11, 25, 37, 38, 40),]

#clean up otu table
row.names(otu)[3] <- "Haemophilus haemolyticus"
row.names(otu)[10] <- "Corynebacterium pseudodiphtheriticum"
row.names(otu)[11] <- "Corynebacterium accolens"

totu <- t(otu)
sparcc.chi1 <- sparcc(totu)

sparcc.chi1$Cor[1:5, 1:5]
#threw away names; will add back from otu dataframe
rownames(sparcc.chi1$Cor) <- colnames(totu)
colnames(sparcc.chi1$Cor) <- colnames(totu)
rownames(sparcc.chi1$Cov) <- colnames(totu)
colnames(sparcc.chi1$Cov) <- colnames(totu)
#look to confirm
sparcc.chi1$Cor[1:5, 1:5]

#Plotting SparCC correlation
#first generate helper function to get upper triangle of the correlation matrix
get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
}

# get_upper_tri <- function(mat){
#   lt <- mat
#   lt[lower.tri(mat)] <- ""
#   mat <- as.data.frame(upper)
#   mat
# }

#helper function to reorder matrix (?)
reorder_cormat <- function(cormat){
# Use correlation between variables as distance
dd <- as.dist((1-cormat)/2)
hc <- hclust(dd)
cormat <-cormat[hc$order, hc$order]
}

plotableSparcc <- sparcc.chi1$Cor %>% reorder_cormat %>% get_upper_tri() %>% reshape2::melt() %>% na.omit()
Sparcc_plot <- plotableSparcc %>% ggplot(aes(x = Var2, y = Var1, fill = value)) + geom_tile() + scale_fill_gradient2() + theme(axis.text.x = element_text(angle = 90, hjust = 1))
Sparcc_plot

# plotableSparcc <- sparcc.chi1$Cor %>% reorder_cormat() %>% get_upper_tri() %>% reshape2::melt() %>% na.omit()
# Sparcc_plot <- plotableSparcc %>% ggplot(aes(reorder(Var2, -desc(as.character(Var2))), reorder(Var1, -desc(as.character(Var1))), fill = value)) + geom_tile() + scale_fill_gradient2() + theme(axis.text.x = element_text(angle = 90, hjust = 1))
# Sparcc_plot

#save sparcc plot
png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/sparcc_tile_v3.png",
    width = 12, height = 9, units = 'in', res = 600)
Sparcc_plot
dev.off()

#calculate sparcc p values
#first have to boostrap the sparccc value
tp0 <- proc.time()
out2 <- sparccboot(totu, R = 1000)
tp1 <- proc.time()
tp1 - tp0

#calculate p values from the bootstrapped values
outP <- pval.sparccboot(out2)
data.frame(outP$cors, outP$pvals) %>% head

#these values are hard to use because they don’t tell you which p-values correspond to which species; source to fix: https://github.com/zdk123/SpiecEasi/issues/17

cors <- outP$cors
pvals <- outP$pvals
sparCCpcors <- diag(0.5, nrow = dim(sparcc.chi1$Cor)[1], ncol = dim(sparcc.chi1$Cor)[1])
sparCCpcors[upper.tri(sparCCpcors, diag=FALSE)] <- cors
sparCCpcors <- sparCCpcors + t(sparCCpcors)

sparCCpval <- diag(0.5, nrow = dim(sparcc.chi1$Cor)[1], ncol = dim(sparcc.chi1$Cor)[1])
sparCCpval[upper.tri(sparCCpval, diag=FALSE)] <- pvals
sparCCpval <- sparCCpval + t(sparCCpval)

rownames(sparCCpcors) <- colnames(totu)
colnames(sparCCpcors) <- colnames(totu)
rownames(sparCCpval) <- colnames(totu)
colnames(sparCCpval) <- colnames(totu)

sparCCpcors[1:5, 1:5]
sparCCpval[1:5, 1:5]

#now need to process and account for false discovery rates 
#add helper function first
reorder_cor_and_p <- function(cormat, pmat){
  dd <- as.dist((1-cormat)/2)
  hc <- hclust(dd)
  cormat <-cormat[hc$order, hc$order]
  pmat <- pmat[hc$order, hc$order]
  list(r = cormat, p = pmat)
}

reordered_all_sparcc <- reorder_cor_and_p(sparCCpcors, sparCCpval)
reordered_sparccCor <- reordered_all_sparcc$r
reordered_sparccP<- reordered_all_sparcc$p

sparccCor_processed <- reordered_sparccCor  %>% get_upper_tri() %>% reshape2::melt() %>% na.omit() %>% rename(cor = value)
sparccP_processed <- reordered_sparccP  %>% get_upper_tri() %>% reshape2::melt() %>% na.omit() %>% rename(p = value)

# join the two data frames

SparccP <- left_join(sparccCor_processed, sparccP_processed, by = c("Var1", "Var2")) %>%
  # # remove self correlations
  # filter(Var1 != Var2) %>% 
  # calculate the false discovery rate to adjust for multiple p values
  mutate(fdr = p.adjust(p, method = "BH"))

#what is our false discovery rate threshold? for Maaslin2, we used a q value of 0.2; here, will use 0.1.
fdrThresh2 <- 0.1 # fdr threshold

sparccOkP2 <- SparccP%>% filter(fdr < fdrThresh2) 

SparccP_plot2c <- SparccP %>% ggplot(aes(x = Var2, y = Var1, fill = cor)) + geom_tile() + scale_fill_gradient2(low = "#800000FF", mid = "white", high = "#4D004B", midpoint = 0) + scale_x_discrete(position = "bottom", limits=rev) + theme(
  axis.text.x = element_markdown(angle = 45, hjust = 1, face = "italic"),
  axis.text.y = element_markdown(face = "italic"),
  axis.title = element_blank()) + geom_point(data = sparccOkP2, shape = 8, size = 0.5) + labs(fill = "Correlation")


png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/sparccP2c.png",
    width = 12, height = 9, units = 'in', res = 600)
SparccP_plot2c
dev.off()

#save as pdf using ggsave
ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig4SparCC.pdf", plot = SparccP_plot2c, dpi = 300, width = 10, height = 7)

#will re-recreate our sparCC plot that is alphabetical and share with MK. We can either have an alphabetical plot or a triangle plot, but I don't know how to create a plot that is both
SparccPcsv <- read.csv("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/SparCCredo_cortax2_test.csv")

SparccP_plot2d <- SparccPcsv %>% ggplot(aes(x = Var2, y = Var1, fill = cor)) + geom_tile() + scale_fill_gradient2(low = "#800000FF", mid = "white", high = "#4D004B", midpoint = 0) + scale_x_discrete(limits=rev, position = "top") + scale_y_discrete(limits=rev, position = "right") + theme(
   axis.text.x = element_markdown(angle = 45, hjust = 0, face = "italic"),
  axis.text.y = element_markdown(face = "italic"),
  axis.title = element_blank()) + geom_point(data = sparccOkP2, shape = 8, size = 0.5) + labs(fill = "Correlation")

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig4SparCC_V2.pdf", plot = SparccP_plot2d, dpi = 300, width = 10, height = 7)

#UPDATE 9-22-23 FROM MK: make legend symmetrical and reduce size of "correlation"; could also use rho symbol but will keep text for clarity
SparccP_plot2e <- SparccP %>% ggplot(aes(x = Var2, y = Var1, fill = cor)) + geom_tile() + scale_fill_gradientn(
  colors = c("#800000FF", "white", "#4D004B"), limits = c(-1, 1)) + scale_x_discrete(position = "bottom", limits=rev) + theme(
  axis.text.x = element_markdown(angle = 45, hjust = 1, face = "italic"),
  axis.text.y = element_markdown(face = "italic"),
  legend.title = element_text(size = 9),
  legend.text.align = 1,
  axis.title = element_blank()) + geom_point(data = sparccOkP2, shape = 8, size = 0.5) + labs(fill = "Correlation")

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig4SparCC_V3.pdf", plot = SparccP_plot2e, dpi = 300, width = 10, height = 7)

```

### Differential abundance plots 

```{r}
###See k-medioid clustering section to create chi.prune, chi.rel, and melted_df!

# Create dataframes with overall relative abundances of phyla and genera
melted_df$Phylum <- as.character(melted_df$Phylum)
phyla_abundances <- aggregate(melted_df$Abundance, by=list(Phylum=melted_df$Phylum), FUN=sum)
phyla_abundances$x <- (phyla_abundances$x)/(nsamples(chi.rel))
names(phyla_abundances)[names(phyla_abundances) == 'x'] <- 'phyla_Ab'
sum(phyla_abundances$phyla_Ab)    # Should sum to 1 (sum of relative abundances of phyla)
nrow(phyla_abundances)            # Corresponds to # of unique phyla: 116

melted_df$Genus <- as.character(melted_df$Genus)
genus_abundances <- aggregate(melted_df$Abundance, by=list(Phylum=melted_df$Phylum, Genus=melted_df$Genus,
                                                           OTU=melted_df$OTU), FUN=mean)
genus_abundances <- aggregate(genus_abundances$x, by=list(Phylum=genus_abundances$Phylum,
                                                          Genus=genus_abundances$Genus), FUN=sum)
names(genus_abundances)[names(genus_abundances) == 'x'] <- 'genus_Ab'
sum(genus_abundances$genus_Ab)    # Should sum to 1 (sum of relative abundances of genera)
nrow(genus_abundances)            # Corresponds to # of unique genera: 2146
abundances <- merge(genus_abundances, phyla_abundances, by="Phylum")

melted_df$Species <- as.character(melted_df$Species)
species_abundances <- aggregate(melted_df$Abundance, by=list(Phylum=melted_df$Phylum, Genus=melted_df$Genus,Species=melted_df$Species,
                                                           OTU=melted_df$OTU), FUN=mean)
species_abundances <- aggregate(species_abundances$x, by=list(Phylum=species_abundances$Phylum,
                                                              Genus=species_abundances$Genus, Species=species_abundances$Species), FUN=sum)
names(species_abundances)[names(species_abundances) == 'x'] <- 'species_Ab'
sum(species_abundances$species_Ab)    # Should sum to 1 (sum of relative abundances of genera)
nrow(species_abundances)            # Corresponds to # of unique species: 6708 ***WHY is this different from 6709?###
abundances <- merge(abundances, species_abundances, by=c("Phylum", "Genus"))

phyla_abundances <- arrange(phyla_abundances, desc(phyla_Ab)) 
# Create a df with the top 5 phyla
TOPPhyla <- unique(phyla_abundances$Phylum[1:5])
phylum_df <- phyla_abundances[phyla_abundances$Phylum %in% TOPPhyla,]
phylum_df$Phylum <- factor(phylum_df$Phylum, levels = phylum_df$Phylum[order(-phylum_df$phyla_Ab)])

#create df with top 10 genera
genus_abundances <- arrange(genus_abundances, desc(genus_Ab))  
TOPGenera <- unique(genus_abundances$Genus[1:10])
genus_df <- genus_abundances[genus_abundances$Genus %in% TOPGenera,]
genus_df$Genus <- factor(genus_df$Genus, levels = genus_df$Genus[order(-genus_df$genus_Ab)])

#create df with top 10 species
species_abundances <- arrange(species_abundances, desc(species_Ab))  
TOPSpecies <- unique(species_abundances$Species[1:10])
species_df <- species_abundances[species_abundances$Species %in% TOPSpecies,]
species_df$Species <- factor(species_df$Species, levels = species_df$Species[order(-species_df$species_Ab)])

# Rename species other than top10 as "Other" in creating dataframe relative_df
# Rename genera other than top10 as "Other" in creating dataframe relative_df 
relative_df <- merge(melted_df, abundances, by=c("Genus", "Species"))

####9-26
#can recode Species variable in melted_df to match with the TOPSpecies list, so any value not in that list is recoded as other as in line 2541###
### don't need relative_df

#keep getting vector memory exhausted error; need to reboot R so will save melted_df and abundances as csv files
  #source for fix: https://stackoverflow.com/questions/51295402/r-on-macos-error-vector-memory-exhausted-limit-reached
# write.csv(melted_df, 'melted_df.csv')
# write.csv(abundances, 'abundances.csv')
# abundances <- read.csv("abundances.csv")
# melted_df <- read.csv("melted_df.csv")
#note: using the csv files led to very weird errors (sample sums 0.99 instead of 1, 1 phyla missing), 
  #so fixed memory error and reran the code instead of using the abundances and melted_df csv files

relative_df$Species[!(relative_df$Species %in% TOPSpecies)] <- "Other"
sum(relative_df$Abundance)#sums to 143.0008; n=143 so tracks?
table(relative_df$Species) #We have 11 levels for the Species variable, which are consistent with our top 10 + other

###Prepping for barplot

#To generate color palette of 11 colors used previously: 
library(RColorBrewer)
library(scales)
chisp2 <- c("#293352", "#800000FF", "#767676FF", "#FFA319FF", "#8A9045FF", "#155F83FF", "#C16622FF", "#4D004B", "#8F3931FF", "#58593FFF", "#350E20FF")
show_col(chisp2)

#source for rotating x labels: https://stackoverflow.com/questions/1330989/rotating-and-spacing-axis-labels-in-ggplot2
relative_df$subject_abundances <- (relative_df$Abundance)/(length(unique(relative_df$Sample))) 
#abbreviate HIV status
relative_df$hiv2 <- NA
relative_df$hiv2 [relative_df$hiv == "Infected"] <- "CLWH"
relative_df$hiv2 [relative_df$hiv == "Exposed Uninfected"] <- "HEU children"
relative_df$hiv2 [relative_df$hiv == "Unexposed"] <- "HUU children"
table(relative_df$hiv2)
#matches up with table(relative_df$hiv)
relative_df$hiv2 <- as.factor(relative_df$hiv2)
relative_df$hiv2 <- reorder(relative_df$hiv2, new.order=c("CLWH", "HEU children", "HUU children"))
relative_df$Species <- factor(relative_df$Species)
relative_df$Species <- reorder(relative_df$Species, new.order=c("Corynebacterium accolens", "Corynebacterium propinquum", "Corynebacterium pseudodiphtheriticum", "Dolosigranulum pigrum", "Haemophilus influenzae", "Moraxella catarrhalis", "Moraxella lincolnii", "Moraxella nonliquefaciens", "Staphylococcus aureus", "Streptococcus pneumoniae", "Other"))
                                    
```

- I kept getting a weird striped output in my barplots with the code in RMD (which did not happen in a standard R file), as described here: https://stackoverflow.com/questions/43997029/unwanted-stripes-in-ggplot2-bar-chart-when-knitted-to-pdf


```{r, echo=FALSE, dev='png'}

#Manuscript revisions: Do not italicize "Other"
#Source for using ggtext: https://stackoverflow.com/questions/48688404/how-to-italicize-one-category-in-a-legend-in-ggplot2
library(ggtext)
subject_plot2 <- ggplot(relative_df[order(relative_df$Species, decreasing = TRUE),], aes(x=hiv2, y=subject_abundances, fill=Species)) + 
  geom_bar(stat="identity", position="fill") + theme_classic() + 
  theme(legend.position="right", 
  legend.text= element_markdown(),
  axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), 
  axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + 
  scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  xlab("HIV Status") + ylab("Mean relative abundance") 

#unwanted stripes; do they persist when saved as a .png?
png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig2b_TEST.png", width = 9, height = 5, units = 'in', res = 600)
plot(subject_plot2)
dev.off()
#saves as a png with no problems!

#asked ChatGPT for help 9/13/23!
#1. will try to save as a pdf using ggsave with increased resolution
ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig2b_TEST.pdf", plot = subject_plot2, dpi = 300)
#Still looks striped in Preview, so then I opened the same file in Acrobat Reader and THE STRIPES ARE GONE!!

p2 <- plot(subject_plot2)
p2

#may require some size troubleshooting when we reassemble our final 4 part figure, but we will save as pdf using ggsave

# ggsave(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig2b_TEST.eps", width = 12, height = 4, units = "in")
# p2
# dev.off()

```

### DETOUR: Assessing correlation between D. pigrum and S. pyogenes for Matt (will not be included in paper, delete from final RMD file that goes on GitHub)

- will use our "relative_df" dataframe that has abundances for all 6708 species identified and pull out D. pigrum and S. pyogenes

```{r}
###NOT WORKING; come back to 
gas <- filter(relative_df, Species == "Streptococcus pyogenes")
dpig <- filter(relative_df, Species == "Dolosigranulum pigrum")

#drop all but species name and species abundance, rename the abundance variables for each, and then merge into 1 dataframe
gas <- rename(gas, ab.gas = Abundance)
dpig <- rename(dpig, ab.dpig = Abundance)


#Create simple dataframe with abundance variables and sample_id, hiv, age, and merge on sample_id
corr_mk <- dpig[c("sample_id", "age", "hiv", "ab.dpig")]
test <- gas[c("sample_id", "ab.gas")]
corr_mk <- merge(corr_mk, test, by ="sample_id", sort = TRUE)

write.csv(corr_mk, "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/corr_gas_dpig_forMK.csv", row.names = F)
corr_mk <- read.csv("corr_gas_dpig_forMK.csv")

cor.test(corr_mk$ab.dpig, corr_mk$ab.gas, method = "pearson")  #P=0.13, cor = -0.127
cor.test(corr_mk$ab.dpig, corr_mk$ab.gas, method = "spearman", exact = FALSE)  #P=0.35, estimate = -0.079
#no correlation BUT large number of zeros for GAS so may be down to sample size

library(ggtext)

mk <- ggplot(data = corr_mk, aes(x = ab.dpig, y = ab.gas)) + 
  geom_point() + 
  geom_smooth(method = "lm", color="red", size=0.5, se = FALSE) +
  labs(title="*D. pigrum* vs *S. pyogenes*",x="*D. pigrum*", y = "*S. pyogenes*")+
  geom_text(size=6, x=0.4, y=0.003, label="r = - 0.08, p = 0.35") +
  scale_x_continuous(breaks = seq(-0.1, 0.7, 0.2), limits = c(0, 0.7)) +
  theme_bw() +
  theme(
    plot.title = ggtext::element_markdown(size=14, hjust=0.5),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12),
    legend.position = "right",
    axis.text = element_text(size = 14),
    axis.title = element_text(size = 14),
    axis.title.x = ggtext::element_markdown(),
    axis.title.y = ggtext::element_markdown())

png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/corr_forMK.png",
     width = 5, height = 5, units = 'in', res = 600)
mk
dev.off()

```

### Subanalysis of factors associated with microbiome composition among CLWH

#### Creating new variables

```{r}
hiv = subset_samples(chi_prune, hiv == "Infected")
chi_hiv <- data.frame(sample_data(hiv))

##############
#NEW VARIABLES
##############
#How many kids receiving TMP-SMX? NOTE: we did not include these kids in the abx category. Need to re-do analyses with this included?
table(chi_hiv$child_tmpsmx) #15 kids on TMP-SMX
#manual inspection: 10 kids on TMP-SMX without other abx exposure

#Need to make a binary variable for tmpsmx use
chi_hiv$tmpsmx <- NA
chi_hiv$tmpsmx [chi_hiv$child_tmpsmx == 1] <- "Yes"
chi_hiv$tmpsmx[chi_hiv$child_tmpsmx == 2] <- "No"
table(chi_hiv$tmpsmx)

#Don't need new variable, but how many kids had received abx for acute infxn?
table(chi_hiv$abx)

#I want the most recent viral load values for my analysis. Will try creating a new variable to ID which value is more recent first:
#missing 3 vl1 and 14vl2's
chi_hiv$newest_vL <- NA
chi_hiv$newest_vL [chi_hiv$child_vl1_weeks < chi_hiv$child_vl2_weeks] <- 1
chi_hiv$newest_vL [chi_hiv$child_vl2_weeks < chi_hiv$child_vl1_weeks] <- 2
chi_hiv$newest_vL [chi_hiv$child_vl2_weeks < 0] <- 1  #we want to include the viral loads collected BEFORE enrollment where possible
chi_hiv$newest_vL [is.na(chi_hiv$child_vl_2)] <- 1    #if we are missing vl_2, we will use vl_1
chi_hiv$newest_vL [is.na(chi_hiv$child_vl_1) & is.na(chi_hiv$child_vl_2)] <- NA
table(chi_hiv$newest_vL)

#Next, can we pull the viral load that was most recent using our newest_vL variable?
#source: https://stackoverflow.com/questions/67944727/creating-new-variable-by-selecting-column-based-on-value-of-another-column
#Need to do this with rows that have values (aka need to exclude the NA viral load children for this to work)
#For future phyloseq analyses: will need to remove B-04, B-07, B-52
chi_viral <- subset(chi_hiv, !is.na(newest_vL))
chi_viral$Var1 <- chi_viral$child_vl_1
chi_viral$Var2 <- chi_viral$child_vl_2

chi_viral <- chi_viral %>% rowwise() %>%
  mutate(newest_vlnum = get(paste0('Var', newest_vL)))
#now we have a dataframe of the 41 children with viral load measurements and 1 column containing the most recent viral loads for each
summary(chi_viral$newest_vlnum)
table(chi_viral$newest_vlnum) #27 with viral suppression, 14 without viral suppression

#make new variable for viral suppression or not, with vL < 400 = suppressed
chi_viral$vl_suppr <- NA
chi_viral$vl_suppr [chi_viral$newest_vlnum < 400] <- "1"
chi_viral$vl_suppr [chi_viral$newest_vlnum >= 400] <- "0"
table(chi_viral$vl_suppr)

#REPEAT the above steps to get date associated with most recent viral load, most recent CD4, and date a/w most recent CD4
#Viral load collection date
chi_viral$Var1 <- chi_viral$child_vl1_weeks
chi_viral$Var2 <- chi_viral$child_vl2_weeks

chi_viral <- chi_viral %>% rowwise() %>%
  mutate(newest_vldate = get(paste0('Var', newest_vL)))   #now have info for when vL were checked in relation to enrollment

summary(chi_viral$newest_vldate)

chi_viral <- within(chi_viral, rm(Var1, Var2))

#CD4
#missing CD4 data for B-45 and B-52
#will focus on percentages since this is what is presented for kids
chi_hiv$newest_cd4 <- NA
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks1 < chi_hiv$chi_cd4_weeks2] <- 1
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks2 < chi_hiv$chi_cd4_weeks1] <- 2
chi_hiv$newest_cd4 [chi_hiv$chi_cd4_weeks2 < 0] <- 1  #we want to include the viral loads collected BEFORE enrollment where possible
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_2)] <- 1    #if we are missing cd4_2, we will use cd4_1
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_1)] <- 2    #one child is missing cd4_1 but has cd4_2
chi_hiv$newest_cd4 [is.na(chi_hiv$chi_cd4_perc_1) & is.na(chi_hiv$chi_cd4_perc_2)] <- NA
table(chi_hiv$newest_cd4)

#create new dataframe without B-45 and B-52 (aka new df with children missing CD4 data removed)
chi_cd4 <- subset(chi_hiv, !is.na(newest_cd4))
chi_cd4$Var1 <- chi_cd4$chi_cd4_perc_1
chi_cd4$Var2 <- chi_cd4$chi_cd4_perc_2

chi_cd4 <- chi_cd4 %>% rowwise() %>%
  mutate(newest_cd4perc = get(paste0('Var', newest_cd4)))

summary(chi_cd4$newest_cd4perc)
table(chi_cd4$newest_cd4perc)

#make new variable for immune competent CD4 or not, with CD4 percentage >=25% = immune competent
  #d/w MK re: potentially adding a cutoff at 15% for immunosuppression, but we only have 2 kids with CD4% < 15 so will stay with 25% binary for now
chi_cd4$cd4nL <- NA
chi_cd4$cd4nL [chi_cd4$newest_cd4perc >=25] <- "1"
chi_cd4$cd4nL [chi_cd4$newest_cd4perc < 25] <- "0"
table(chi_cd4$cd4nL)  #11 kids with CD4% < 25

#REPEAT the above steps to get date associated with most recent CD4
#CD4 collection date
chi_cd4$Var1 <- chi_cd4$chi_cd4_weeks1
chi_cd4$Var2 <- chi_cd4$chi_cd4_weeks2

chi_cd4 <- chi_cd4 %>% rowwise() %>%
  mutate(newest_cd4date = get(paste0('Var', newest_cd4)))   #now have info for when cd4 were checked in relation to enrollment

summary(chi_cd4$newest_cd4date)

chi_cd4 <- within(chi_cd4, rm(Var1, Var2))

#In case you need to go back and redo everything with only those children who have BOTH viral loads and CD4s:
test <- chi_viral[c("sample_id", "newest_vlnum", "vl_suppr", "newest_vldate")]
hiv_labs <- merge(chi_cd4, test, by="sample_id")
remove(test)

```
#### Demographics by viral suppression and CD4 classification

```{r}

##################################
#DEMOGRAPHICS BY VIRAL SUPPRESSION
##################################
#will look at HIV specific variables (particularly viral suppression, CD4 perc) and general vars
#ex if you are not virally suppressed, is that a/w SES? 

#Age
hist(chi_viral$age)
summary(chi_viral$age)
tapply(chi_viral$age, chi_viral$vl_suppr, summary)
wilcox.test(age ~ vl_suppr, data = chi_viral) #P=0.19 with warning: cannot compute exact p-value with ties
t.test(age ~ vl_suppr, data = chi_viral)  #P=0.17

#Sex
table(chi_viral$sex)  #22 girls
table(chi_viral$sex, chi_viral$vl_suppr)
prop.table(table(chi_viral$sex, chi_viral$vl_suppr), 2)
summary(table(chi_viral$sex, chi_viral$vl_suppr)) #P=0.74

#Maternal age
hist(chi_viral$mat_age2)  #looks quite normal, but will still present median and IQR for consistency
summary(chi_viral$mat_age2)
tapply(chi_viral$mat_age2, chi_viral$vl_suppr, summary)
wilcox.test(mat_age2 ~ vl_suppr, data = chi_viral) #P=0.15
t.test(mat_age2 ~ vl_suppr, data = chi_viral) #P=0.11

#Season
table(chi_viral$season)  
table(chi_viral$season, chi_viral$vl_suppr)
prop.table(table(chi_viral$season, chi_viral$vl_suppr), 2)
summary(table(chi_viral$season, chi_viral$vl_suppr))  #P=0.44

#Maternal education
table(chi_viral$mat_educ2)  
table(chi_viral$mat_educ2, chi_viral$vl_suppr)
prop.table(table(chi_viral$mat_educ2, chi_viral$vl_suppr), 2)
summary(table(chi_viral$mat_educ2, chi_viral$vl_suppr))  
fisher.test(chi_viral$mat_educ2, chi_viral$vl_suppr) #P=0.34

#Electricity
table(chi_viral$elec)  #33 kids have electricity
table(chi_viral$elec, chi_viral$vl_suppr)
prop.table(table(chi_viral$elec, chi_viral$vl_suppr), 2)
summary(table(chi_viral$elec, chi_viral$vl_suppr))
fisher.test(chi_viral$elec, chi_viral$vl_suppr) #P=1.0

#Wood
table(chi_viral$wood)  #25 kids live in wood-burning households
table(chi_viral$wood, chi_viral$vl_suppr)
prop.table(table(chi_viral$wood, chi_viral$vl_suppr), 2)
summary(table(chi_viral$wood, chi_viral$vl_suppr))
fisher.test(chi_viral$wood, chi_viral$vl_suppr) #P=0.18

#Num household members
hist(chi_viral$hhsize)
summary(chi_viral$hhsize)
tapply(chi_viral$hhsize, chi_viral$vl_suppr, summary)
wilcox.test(hhsize ~ vl_suppr, data = chi_viral) #P=0.97 with warning: cannot compute exact p-value with ties
t.test(hhsize ~ vl_suppr, data = chi_viral) #P=0.68

#antibiotics
table(chi_viral$abx)  #6 kids received abx
table(chi_viral$abx, chi_viral$vl_suppr)
prop.table(table(chi_viral$abx, chi_viral$vl_suppr), 2)
fisher.test(chi_viral$abx, chi_viral$vl_suppr) #P=0.39

#Clinic in past 3 months
table(chi_viral$clinic)  #11 kids with clinic visit
table(chi_viral$clinic, chi_viral$vl_suppr)
prop.table(table(chi_viral$clinic, chi_viral$vl_suppr), 2)
fisher.test(chi_viral$clinic, chi_viral$vl_suppr) #P=0.46

#URI in past 1 month
table(chi_viral$uri_rec2)  #18 kids with recent URI
table(chi_viral$uri_rec2, chi_viral$vl_suppr)
prop.table(table(chi_viral$uri_rec2, chi_viral$vl_suppr), 2)
summary(table(chi_viral$uri_rec2, chi_viral$vl_suppr)) #P=0.92

#URI currently (don't need both this and uri recent in table 1; need to choose)
table(chi_viral$uri_cur2)  #16 kids with current URI
table(chi_viral$uri_cur2, chi_viral$vl_suppr)
prop.table(table(chi_viral$uri_cur2, chi_viral$vl_suppr), 2)
summary(table(chi_viral$uri_cur2, chi_viral$vl_suppr)) #P=0.75

#Hospital in past 3 months -->> 4 children hospitalized, all 4 HIV+
#*MENTION IN MANUSCRIPT
table(chi_viral$hosp)  #3 kids hospitalized who had viral load data
table(chi_viral$hosp, chi_viral$vl_suppr) #2/3 kids not virally suppressed
prop.table(table(chi_viral$hosp, chi_viral$vl_suppr), 2)
fisher.test(chi_viral$hosp, chi_viral$vl_suppr) #P=0.26 (but numbers are SO SMALL)

#PCV doses
table(chi_viral$pcv)  
table(chi_viral$pcv, chi_viral$vl_suppr)
prop.table(table(chi_viral$pcv, chi_viral$vl_suppr), 2)
fisher.test(chi_viral$pcv, chi_viral$vl_suppr) #P=1.0

#HiB
table(chi_viral$dpt)  
table(chi_viral$dpt, chi_viral$vl_suppr)
prop.table(table(chi_viral$dpt, chi_viral$vl_suppr), 2)
fisher.test(chi_viral$dpt, chi_viral$vl_suppr) #P=1.0

#TMP-SMX
table(chi_viral$tmpsmx)  
table(chi_viral$tmpsmx, chi_viral$vl_suppr)
prop.table(table(chi_viral$tmpsmx, chi_viral$vl_suppr), 2)
summary(table(chi_viral$tmpsmx, chi_viral$vl_suppr))  
fisher.test(chi_viral$tmpsmx, chi_viral$vl_suppr) #P=0.17

##########################################
#DEMOGRAPHICS BY CD4 STATUS (< vs >/= 25%)  
##########################################
#Age
hist(chi_cd4$age)
summary(chi_cd4$age)
tapply(chi_cd4$age, chi_cd4$cd4nL, summary)
wilcox.test(age ~ cd4nL, data = chi_cd4) #P=0.10 with warning: cannot compute exact p-value with ties
t.test(age ~ cd4nL, data = chi_cd4)  #P=0.10

#Sex
table(chi_cd4$sex)  #23 girls
table(chi_cd4$sex, chi_cd4$cd4nL)
prop.table(table(chi_cd4$sex, chi_cd4$cd4nL), 2)
summary(table(chi_cd4$sex, chi_cd4$cd4nL)) #P=0.99

#Maternal age
hist(chi_cd4$mat_age2)
summary(chi_cd4$mat_age2)
tapply(chi_cd4$mat_age2, chi_cd4$cd4nL, summary)
wilcox.test(mat_age2 ~ cd4nL, data = chi_cd4) #P=0.31
t.test(mat_age2 ~ cd4nL, data = chi_cd4)  #P=0.60

#Season
table(chi_cd4$season) 
table(chi_cd4$season, chi_cd4$cd4nL)
prop.table(table(chi_cd4$season, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$season, chi_cd4$cd4nL)  #P=1.0

#Maternal education
table(chi_cd4$mat_educ2) 
table(chi_cd4$mat_educ2, chi_cd4$cd4nL)
prop.table(table(chi_cd4$mat_educ2, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$mat_educ2, chi_cd4$cd4nL)  #P=1.0

#Electricity
table(chi_cd4$elec) 
table(chi_cd4$elec, chi_cd4$cd4nL)
prop.table(table(chi_cd4$elec, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$elec, chi_cd4$cd4nL)  #P=0.68

#Wood
table(chi_cd4$wood) #26 kids from wood-burning households
table(chi_cd4$wood, chi_cd4$cd4nL)
prop.table(table(chi_cd4$wood, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$wood, chi_cd4$cd4nL)  #P=0.16

#Num household members
hist(chi_viral$hhsize)
summary(chi_viral$hhsize)
tapply(chi_viral$hhsize, chi_viral$vl_suppr, summary)
wilcox.test(hhsize ~ vl_suppr, data = chi_viral) #P=0.97 with warning: cannot compute exact p-value with ties
t.test(hhsize ~ vl_suppr, data = chi_viral) #P=0.68

#antibiotics
table(chi_cd4$abx) #7 kids received abx
table(chi_cd4$abx, chi_cd4$cd4nL)
prop.table(table(chi_cd4$abx, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$abx, chi_cd4$cd4nL)  #P=0.009

#Clinic in past 3 months
table(chi_cd4$clinic) #11 kids with clinic visit
table(chi_cd4$clinic, chi_cd4$cd4nL)
prop.table(table(chi_cd4$clinic, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$clinic, chi_cd4$cd4nL)  #P=0.12

#URI in past 1 month
table(chi_cd4$uri_rec2) #19 kids with recent URI
table(chi_cd4$uri_rec2, chi_cd4$cd4nL)
prop.table(table(chi_cd4$uri_rec2, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$uri_rec2, chi_cd4$cd4nL)  #P=0.0008

#URI currently (don't need both this and uri recent in table 1; need to choose)
table(chi_cd4$uri_cur2) #16 kids with current URI
table(chi_cd4$uri_cur2, chi_cd4$cd4nL)
prop.table(table(chi_cd4$uri_cur2, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$uri_cur2, chi_cd4$cd4nL)  #P=0.07

#Hospital in past 3 months -->> 4 children hospitalized, all 4 HIV+
#*MENTION IN MANUSCRIPT
table(chi_cd4$hosp) 
table(chi_cd4$hosp, chi_cd4$cd4nL)  #3 of 4 kids had CD4 < 25%
prop.table(table(chi_cd4$hosp, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$hosp, chi_cd4$cd4nL)  #P=0.05 (numbers are SO SMALL though)

#PCV doses
table(chi_cd4$pcv) #35 kids fully vaxed
table(chi_cd4$pcv, chi_cd4$cd4nL)
prop.table(table(chi_cd4$pcv, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$pcv, chi_cd4$cd4nL)  #P=0.48

#HiB
table(chi_cd4$dpt) #34 kids fully vaxed 
table(chi_cd4$dpt, chi_cd4$cd4nL)
prop.table(table(chi_cd4$dpt, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$dpt, chi_cd4$cd4nL)  #P=1.0

#TMPSMX use
table(chi_cd4$tmpsmx) #15 kids on TMP-SMX 
table(chi_cd4$tmpsmx, chi_cd4$cd4nL)
prop.table(table(chi_cd4$tmpsmx, chi_cd4$cd4nL), 2)
fisher.test(chi_cd4$tmpsmx, chi_cd4$cd4nL)  #P=0.03; more kids with low CD4 on TMP-SMX

###COME BACK TO: TMP-SMX GUIDELINES IN KIDS
```

**Summary:** no significant association identified b/w viral suppression and SES, clinical factors, BUT kids with low CD4: higher percent received abx and TMP-SMX and reported recent URI
- will incorporate abx and rec_uri into maaslin2 model below

#### Alpha diversity
```{r}
#Create dataframe with diversity indices for each specimen using HIV pseq object
diversity1 <- estimate_richness(hiv, measures = c("Shannon", "Simpson", "Chao1", "Observed"))

setDT(diversity1, keep.rownames = TRUE)[]
colnames(diversity1) <- c("sample_id", "Observed", "Chao1", "se.chao1", "Shannon", "Simpson")
rownames(diversity1) <- diversity1$sample_id
diversity <- merge(diversity1, chi_viral, by="sample_id")
div_cd4 <- merge(diversity1, chi_cd4, by="sample_id")
div_all <- merge(diversity1, chi_hiv, by="sample_id")

#################
#OBSERVED SPECIES
#################

#How many species on average per sample?
hist(diversity$Observed)
summary(diversity$Observed) 
sd(diversity$Observed)
#Median (IQR) # spec: 113 (67, 201)
#Mean (SD): 163 (173)

#what does it look like by TMP-SMX use? NOTE: N=44
tapply(div_all$Observed, div_all$tmpsmx, summary)   
wilcox.test(Observed ~ tmpsmx, data = div_all)  #P=0.42 with warning (ties)

#what does it look like by viral supp? NOTE: N=41
tapply(diversity$Observed, diversity$vl_suppr, summary)   
wilcox.test(Observed ~ vl_suppr, data = diversity)  #P=0.87 with warning (ties)

########
#SHANNON
########

#Is our SDI data normally distributed? 
shapiro.test(diversity$Shannon)  #yes! P=0.42
histogram(diversity$Shannon)  #looks good
qqnorm(diversity$Shannon, pch = 1, frame = FALSE)
qqline(diversity$Shannon, col = "steelblue", lwd = 2)

#Shannon diversity median and IQR for all samples
summary(diversity$Shannon)
sd(diversity$Shannon)
#Median (IQR) SDI: 1.73 (1.44, 2.26)
#Mean (SD) 1.78 (0.65)  

#Viral suppression
tapply(diversity$Shannon, diversity$vl_suppr, summary)
t.test(Shannon ~ vl_suppr, data = diversity)        #P=0.62

#TMP-SMX use
tapply(div_all$Shannon, div_all$tmpsmx, summary)
t.test(Shannon ~ tmpsmx, data = div_all)  #P=0.28

#CD4 (continuous and binary)
summary(div <- lm(Shannon ~ newest_cd4perc, data = div_cd4))  #P=0.09
tapply(div_cd4$Shannon, div_cd4$cd4nL, summary)
t.test(Shannon ~ cd4nL, data = div_cd4)  #P=0.89

#Abx use
tapply(div_all$Shannon, div_all$abx, summary)
t.test(Shannon ~ abx, data = div_all)  #P=0.99

#MULTIVARIABLE MODEL
summary(div_mv <- lm(Shannon ~ vl_suppr + tmpsmx, data = diversity))
  #with abx
summary(div_mv <- lm(Shannon ~ vl_suppr + tmpsmx + abx, data = diversity))

######
#CHAO1
######

#Is our Chao1 data normally distributed? 
shapiro.test(diversity$Chao1)  #no, because p<0.05; also looks skewed on histogram and QQ is wonky
histogram(diversity$Chao1)
qqnorm(diversity$Chao1, pch = 1, frame = FALSE)
qqline(diversity$Chao1, col = "steelblue", lwd = 2)

#What does it look like if we log transform it? Will this allow for parametric testing?
diversity1$Chao1_log <- log(diversity1$Chao1)
#NOW RE-RUN ALL DIVERSITY SUB-DATAFRAMES ABOVE (CD4, VL, ALL)
shapiro.test(diversity$Chao1_log) 
histogram(diversity$Chao1_log)
qqnorm(diversity$Chao1_log, pch = 1, frame = FALSE)
qqline(diversity$Chao1_log, col = "steelblue", lwd = 2)
#Looks better! Can use t.test and linear regression on transformed data

#Log transform for all diversity df (n=44 for all HIV+ kids, n=42 for CD4 kids)
div_all$Chao1_log <- log(div_all$Chao1)
div_cd4$Chao1_log <- log(div_cd4$Chao1)

#Chao1 median and IQR for all samples
summary(diversity$Chao1)
sd(diversity$Chao1)
#Median (IQR) Chao1: 109 (63, 205)
#Mean (SD) 216 (300.4)

#Viral suppression
tapply(diversity$Chao1_log, diversity$vl_suppr, summary)
t.test(Chao1_log ~ vl_suppr, data = diversity)        #P=0.96

#TMP-SMX use (N=44)
tapply(div_all$Chao1_log, div_all$tmpsmx, summary)
t.test(Chao1_log ~ tmpsmx, data = div_all)  #P=0.52

#CD4 (continuous and binary)
summary(div <- lm(Chao1_log ~ newest_cd4perc, data = div_cd4))  #P=0.65
tapply(div_cd4$Chao1_log, div_cd4$cd4nL, summary)
t.test(Chao1_log ~ cd4nL, data = div_cd4)  #P=0.25

#abx (N=44)
tapply(div_all$Chao1_log, div_all$abx, summary)
t.test(Chao1_log ~ abx, data = div_all)  #P=0.92

#MULTIVARIABLE MODEL
summary(div_mv <- lm(Chao1_log ~ vl_suppr + tmpsmx, data = diversity))
  #including abx
summary(div_mv <- lm(Chao1_log ~ vl_suppr + tmpsmx + abx, data = diversity))

#Skipping boxplots
remove(div, div_all, div_cd4, diversity, diversity1)

```


#### Beta diversity: PCoA plots

```{r}
#remove kids missing viral load data: B-04, B-07, B-52 and add chi_viral as sample data
vl <- hiv
chi_viral <- as.data.frame(chi_viral)
row.names(chi_viral) <- chi_viral$sample_id
chi_viral <- sample_data(chi_viral)
head(chi_viral)
vl <- merge_phyloseq(vl, chi_viral)
#now see if it worked:
bloop <- data.frame(sample_data(vl))
#not working for some reason on 6/2/22; merged new sample data with pseq FIRST, then removed samples and it worked
#success!
remove(bloop)
vl = subset_samples(vl, sample_id != "B.04.CHI" & sample_id != "B.07.CHI" & sample_id != "B.52.CHI")
chi_viral <- data.frame(sample_data(vl))

#do the same to create a CD4 pseq object to plot
cd4 <- hiv
chi_cd4 <- as.data.frame(chi_cd4)
row.names(chi_cd4) <- chi_cd4$sample_id
chi_cd4 <- sample_data(chi_cd4)
head(chi_cd4)
cd4 <- merge_phyloseq(cd4, chi_cd4)
#now see if it worked:
bloop <- data.frame(sample_data(cd4))
table(bloop$cd4nL)
#success!
remove(bloop)
#missing: B-45 and B-52
cd4 = subset_samples(cd4, sample_id != "B.45.CHI" & sample_id != "B.52.CHI")
chi_cd4 <- data.frame(sample_data(cd4))

#COME BACK TO: use same steps to make hiv_labs pseq that has n=40 and both vl and CD4 variables
labs <- hiv
row.names(hiv_labs) <- hiv_labs$sample_id
hiv_labs <- sample_data(hiv_labs)
labs <- merge_phyloseq(labs, hiv_labs)
#now see if it worked:
bloop <- data.frame(sample_data(labs))
#success!
remove(bloop)
#missing: B-45 and B-52
labs = subset_samples(labs, sample_id != "B.04.CHI" & sample_id != "B.07.CHI" & sample_id != "B.45.CHI" & sample_id != "B.52.CHI")
hiv_labs <- data.frame(sample_data(labs))

#Now transform our pseq objects
vl_clr <- transform(vl, "clr")
cd4_clr <- transform(cd4, "clr")
hiv_clr <- transform(hiv, "clr")

#PCoA plots#
#VIRAL SUPPRESSION
clr_pcoa <- ordinate(vl_clr, method = "PCoA", distance = "euclidean")
lines <- c("dashed", "dashed")
clr_vl <- plot_ordination(vl_clr, clr_pcoa, color = "vl_suppr", title = "Euclidean PCoA Plot by Viral Suppression") +
  geom_point(size = 2) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values=c('#800000FF','#155F83FF'))+
  stat_ellipse(aes(linetype=vl_suppr), geom="polygon", alpha=0, type="t", level=0.8, size=0.7) +
  xlab("PC1 (13.4%)") + ylab("PC2 (12%)") +
  scale_linetype_manual(values=lines)
png(file="clr_vlsuppr.png",
    width = 5, height = 4, units = 'in', res=600)
clr_vl
dev.off()

#CD4 (binary) using cd4_clr
###AUG 3 UPDATE: change labels: so will need to create new variable and import back into cd4_clr
chi_cd4$cd4cat <- NA
chi_cd4$cd4cat [chi_cd4$cd4nL=="1"] <- "Normal CD4"
chi_cd4$cd4cat [chi_cd4$cd4nL=="0"] <- "Low CD4"
table(chi_cd4$cd4cat)
class(chi_cd4$cd4cat)
chi_cd4$cd4cat <- as.factor(chi_cd4$cd4cat)
chi_cd4$cd4cat <- reorder(chi_cd4$cd4cat, new.order=c("Low CD4", "Normal CD4"))

#Import back into pseq (source: https://jacobrprice.github.io/2017/08/26/phyloseq-to-vegan-and-back.html)
sample_data(cd4_clr) <- as.data.frame(chi_cd4)
#now see if it worked:
bloop <- data.frame(sample_data(cd4_clr))
table(bloop$cd4cat)
#success!
remove(bloop)

clr_pcoa <- ordinate(cd4_clr, method = "PCoA", distance = "euclidean")
lines <- c("dashed", "dashed")
clr_cd4 <- plot_ordination(cd4_clr, clr_pcoa, color = "cd4cat") +
  geom_point(size = 2) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values=c('#800000FF','#155F83FF'))+
  stat_ellipse(aes(linetype=cd4cat), geom="polygon", alpha=0, type="t", level=0.8, size=0.7) +
  xlab("PC1 (13.4%)") + ylab("PC2 (11.5%)") +
  scale_linetype_manual(values=lines)
# png(file="clr_cd4nl.png",
#     width = 6, height = 4, units = 'in', res=600)
clr_cd4
p3 <- clr_cd4
p3
# dev.off()

#TMP-SMX use
clr_tmpsmx <- plot_ordination(cd4_clr, clr_pcoa, color = "tmpsmx", title = "Euclidean PCoA Plot by TMP-SMX use") +
  geom_point(size = 2) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values=c('#800000FF','#155F83FF'))+
  stat_ellipse(aes(linetype=tmpsmx), geom="polygon", alpha=0, type="t", level=0.8, size=0.7) +
  xlab("PC1 (13.4%)") + ylab("PC2 (12%)") +
  scale_linetype_manual(values=lines)
png(file="clr_tmpsmx.png",
    width = 5, height = 4, units = 'in', res=600)
clr_tmpsmx
dev.off()
```

#### Beta diversity: Permanova

```{r}
#PERMANOVA: viral load & TMP-SMX
otu <- as.data.frame(otu_table(vl_clr, taxa_are_rows=TRUE))
#Need species in columns and observations in rows: transpose using t() function and save as a dataframe
#Source: https://github.com/edamame-course/2015-tutorials/blob/master/final/Ashley_Intro_to_R.md
otu_trans <- data.frame(t(otu))
chi_viral <- data.frame(sample_data(vl_clr))
BC.dist=vegdist(otu_trans, method="euclidean")

#viral suppression
adonis(BC.dist ~ vl_suppr, data = chi_viral, permutations = 1000)
#No significant difference by HIV status among children (P = 0.83)
  #viral suppression + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis2(BC.dist ~ vl_suppr + age, data = chi_viral, permutations = 1000)
#No significant difference by HIV status among children (P = 0.82) or age (P=0.15)

#TMPSMX use (n=41)
adonis(BC.dist ~ tmpsmx, data = chi_viral, permutations = 1000)
#Trend towards difference in overall composition by tmp-smx use but did not reach signif (P = 0.07, R2 = 0.034)
  #TMPSMX + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis(BC.dist ~ tmpsmx + age, data = chi_viral, permutations = 1000)
#No significant difference by TMPSMX among children (P = 0.07) or age (P=0.22)

#PERMANOVA: CD4 (both continuous and binary), TMPSMX, abx
otu <- as.data.frame(otu_table(cd4_clr, taxa_are_rows=TRUE))
#Need species in columns and observations in rows: transpose using t() function and save as a dataframe
#Source: https://github.com/edamame-course/2015-tutorials/blob/master/final/Ashley_Intro_to_R.md
otu_trans <- data.frame(t(otu))
chi_cd4 <- data.frame(sample_data(cd4_clr))
BC.dist=vegdist(otu_trans, method="euclidean")

#CD4 (continuous)
adonis(BC.dist ~ newest_cd4perc, data = chi_cd4, permutations = 1000)
#Trend towards difference in overall composition by CD4 percentage but did not reach signif (P = 0.052, R2 = 0.035)
    #CD4 + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis(BC.dist ~ newest_cd4perc + age, data = chi_cd4, permutations = 1000)
#Trend towards difference in overall composition by CD4 percentage but did not reach signif (P = 0.06, R2 = 0.035); age P=0.4

#CD4 (binary)
adonis(BC.dist ~ cd4nL, data = chi_cd4, permutations = 1000)  #P=0.01, R = 0.042
  #CD4 + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis2(BC.dist ~ cd4nL + age, data = chi_cd4, permutations = 1000) #P=0.01, R = 0.042 for CD4; age P = 0.20

#TMPSMX use (n=42)  #NOTE: at some point adonis got deprecated so had to switch to adonis2 for updates
adonis(BC.dist ~ tmpsmx, data = chi_cd4, permutations = 1000) #P=0.12
    #TMPSMX + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis2(BC.dist ~ tmpsmx + age, data = chi_cd4, permutations = 1000) #P=0.13, R2=0.031

#abx (n=42) with age
adonis2(BC.dist ~ abx + age, data = chi_cd4, permutations = 1000) #P=0.19, R2=0.029

#PERMANOVA: TMP-SMX and abx for all children with HIV (n=44)
otu <- as.data.frame(otu_table(hiv_clr, taxa_are_rows=TRUE))
otu_trans <- data.frame(t(otu))
BC.dist=vegdist(otu_trans, method="euclidean")
  #TMP-SMX
adonis2(BC.dist ~ tmpsmx + age, data = chi_hiv, permutations = 1000) #P=0.07, R2=0.032
  #ABX
adonis2(BC.dist ~ abx + age, data = chi_hiv, permutations = 1000) #P=0.20, R2=0.028

#clean up
remove(hiv_clr, cd4_clr, vl_clr, otu, otu_trans, clr_cd4, clr_pcoa, clr_tmpsmx, clr_vl)
remove(BC.dist, lines)
```

#### Relative abundance plots

```{r}
##############################
#RELATIVE ABUNDANCE PLOTS: CD4
##############################
#Because we see a compositional difference by binary CD4, we will generate relative abundance plots for this pseq and for viral load

###UPDATE 9-22-23; MK recommends using the overall top 10 species for all relative abundance plots in our manuscript. So don't need to re-generate a new relative_df dataframe; we need to merge our cd4 classification with our relative_df
#Create relative abundances table and screenshot without manual labels so you have 2 ways to make sure your labeled figure is correct

#df with only CLWH
test <- subset(relative_df, relative_df$hiv == "Infected")
table(test$Species)
sum(test$Abundance) #sums to 44.00003 and we have 44 CLWH 

#Now will attempt to merge with chi_cd4
test2 <- chi_cd4[c("study_id", "cd4cat")]
rel_cd4 <- merge(test, test2, by=c("study_id"))
sum(rel_cd4$Abundance)   # sums to 42.00003 (and we have n=42 kids...?)
table(rel_cd4$Species)

#the subsequent plots look wrong and I can't figure out where the error is; so will make a plot with just 'test' to start
test$subject_abundances <- (test$Abundance)/(length(unique(test$Sample))) 
test$Species <- factor(test$Species)
test$Species <- reorder(test$Species, new.order=c("Corynebacterium accolens","Corynebacterium propinquum", "Corynebacterium pseudodiphtheriticum", "Dolosigranulum pigrum", "Haemophilus influenzae", "Moraxella catarrhalis", "Moraxella lincolnii",  "Moraxella nonliquefaciens", "Staphylococcus aureus", "Streptococcus pneumoniae", "Other"))

ggplot(test[order(test$Species, decreasing = TRUE),], aes(x=1, y=subject_abundances, fill=Species)) + geom_bar(stat="identity", position="fill") + theme_classic() + theme(legend.position="right", legend.text=element_markdown(), axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  ylab("Mean relative abundance") 

#what happens if we take out decreasing = true and the labels?
ggplot(test[order(test$Species),], aes(x=1, y=subject_abundances, fill=Species)) + geom_bar(stat="identity", position="fill") + theme_classic() + theme(legend.position="right", legend.text=element_markdown(), axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + scale_fill_manual(values = chisp2) +
  ylab("Mean relative abundance") 

#this lines up with our CLWH column in figure 1, so the problem has to be with merging the relative abundance data with our cd4 data
#create a dataframe with ONLY study.id, Species, and subject_abundances and merge this with test2
test3 <- test[c("study_id", "Species", "subject_abundances")]
rel_cd4 <- merge(test2, test3, by=c("study_id"))
#this also doesn't work

#instead of merging, re-run your CD4 code to create the variables in your test dataframe and see if that works 
test$newest_cd4 <- NA
test$newest_cd4 [test$chi_cd4_weeks1 < test$chi_cd4_weeks2] <- 1
test$newest_cd4 [test$chi_cd4_weeks2 < test$chi_cd4_weeks1] <- 2
test$newest_cd4 [test$chi_cd4_weeks2 < 0] <- 1  #we want to include the cd4 collected BEFORE enrollment where possible
test$newest_cd4 [is.na(test$chi_cd4_perc_2)] <- 1    #if we are missing cd4_2, we will use cd4_1
test$newest_cd4 [is.na(test$chi_cd4_perc_1)] <- 2    #one child is missing cd4_1 but has cd4_2
test$newest_cd4 [is.na(test$chi_cd4_perc_1) & is.na(test$chi_cd4_perc_2)] <- NA
table(test$newest_cd4)

#create new dataframe without B-45 and B-52 (aka new df with children missing CD4 data removed)
test2 <- subset(test, !is.na(newest_cd4))
test2$Var1 <- test2$chi_cd4_perc_1
test2$Var2 <- test2$chi_cd4_perc_2

test2 <- test2 %>% rowwise() %>%
  mutate(newest_cd4perc = get(paste0('Var', newest_cd4)))

summary(test2$newest_cd4perc)
table(test2$newest_cd4perc)

#make new variable for immune competent CD4 or not, with CD4 percentage >=25% = immune competent
  #d/w MK re: potentially adding a cutoff at 15% for immunosuppression, but we only have 2 kids with CD4% < 15 so will stay with 25% binary for now
test2$cd4nL <- NA
test2$cd4nL [test2$newest_cd4perc >=25] <- "1"
test2$cd4nL [test2$newest_cd4perc < 25] <- "0"
table(test2$cd4nL)  #11 kids with CD4% < 25

#CREATING RELATIVE ABUNDANCE PLOT
#Clarify CD4 cutoff
test2$cd4cat <- NA
test2$cd4cat [test2$cd4nL==1] <- "Normal CD4"
test2$cd4cat [test2$cd4nL==0] <- "Low CD4"
table(test2$cd4cat)
class(test2$cd4cat)
test2$cd4cat <- as.factor(test2$cd4cat)
test2$cd4cat <- reorder(test2$cd4cat, new.order=c("Low CD4", "Normal CD4"))

# test2$Species <- factor(test2$Species)
# test2$Species <- reorder(test2$Species, new.order=c("Corynebacterium accolens","Corynebacterium propinquum", "Corynebacterium pseudodiphtheriticum", "Dolosigranulum pigrum", "Haemophilus influenzae", "Moraxella catarrhalis", "Moraxella lincolnii",  "Moraxella nonliquefaciens", "Staphylococcus aureus", "Streptococcus pneumoniae", "Other"))

subject_plot_cd4_2 <- ggplot(test2[order(test2$Species, decreasing = TRUE),], aes(x=cd4cat, y=subject_abundances, fill=Species)) + geom_bar(stat="identity", position="fill") + theme_classic() + theme(legend.position="right", legend.text=element_markdown(), axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + scale_fill_manual(values = chisp2) 

# labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  # ylab("Mean relative abundance"

p4 <- plot(subject_plot_cd4_2)



#############
#DELETE BELOW
#############
chi.rel <- transform_sample_counts(cd4, function(Abundance) Abundance/sum(Abundance))
sample_sums(chi.rel)  # This is a sanity check to make sure that relative abundance was calculated for each sample prior to pooling data
#All samples add to 1
melted_df <- psmelt(chi.rel)
#clean up melted_df
melted_df <- within(melted_df, rm(sib_weight, sib_sex, sib_height, sib_muac, sib_race, sib_bfeed_1,
                                  sib_bfeed_2, sib_uri_recent, sib_uri_current, sib_pcr_1, sib_pcr_2,
                                  sib_bcg, sib_hepb, sib_clinic, sib_clinic_dx___5, sib_clinic_dx_oth,
                                  sib_hosp, sib_meds, sib_meds_name, sibmo, sibyr, sib_vl1_weeks, sib_vl_weeks2,
                                  sib_med_days, sib_dpt, sib_pcv, sib_rota, sib_polio, sib_measles, sib_age))
melted_df <- within(melted_df, rm(samp_storage, enr_clinic, enr_clinic_other, mat_np))

# Create dataframes with overall relative abundances of phyla and genera
melted_df$Phylum <- as.character(melted_df$Phylum)
phyla_abundances <- aggregate(melted_df$Abundance, by=list(Phylum=melted_df$Phylum), FUN=sum)
phyla_abundances$x <- (phyla_abundances$x)/(nsamples(chi.rel))
names(phyla_abundances)[names(phyla_abundances) == 'x'] <- 'phyla_Ab'
sum(phyla_abundances$phyla_Ab)    # Should sum to 1 (sum of relative abundances of phyla)
nrow(phyla_abundances)            # Corresponds to # of unique phyla: 116

melted_df$Genus <- as.character(melted_df$Genus)
genus_abundances <- aggregate(melted_df$Abundance, by=list(Phylum=melted_df$Phylum, Genus=melted_df$Genus,
                                                           OTU=melted_df$OTU), FUN=mean)
genus_abundances <- aggregate(genus_abundances$x, by=list(Phylum=genus_abundances$Phylum,
                                                          Genus=genus_abundances$Genus), FUN=sum)
names(genus_abundances)[names(genus_abundances) == 'x'] <- 'genus_Ab'
sum(genus_abundances$genus_Ab)    # Should sum to 1 (sum of relative abundances of genera)
nrow(genus_abundances)            # Corresponds to # of unique genera: 2146
abundances <- merge(genus_abundances, phyla_abundances, by="Phylum")

melted_df$Species <- as.character(melted_df$Species)
species_abundances <- aggregate(melted_df$Abundance, by=list(Phylum=melted_df$Phylum, Genus=melted_df$Genus,Species=melted_df$Species,
                                                             OTU=melted_df$OTU), FUN=mean)
species_abundances <- aggregate(species_abundances$x, by=list(Phylum=species_abundances$Phylum,
                                                              Genus=species_abundances$Genus, Species=species_abundances$Species), FUN=sum)
names(species_abundances)[names(species_abundances) == 'x'] <- 'species_Ab'
sum(species_abundances$species_Ab)    # Should sum to 1 (sum of relative abundances of genera)
nrow(species_abundances)            # Corresponds to # of unique species: 6708 ***WHY is this different from 6709?###
abundances <- merge(abundances, species_abundances, by=c("Phylum", "Genus"))
  #merging on Phylum AND Genus gets rid of the phylum.x and phylum.y problem we were having earlier

phyla_abundances <- arrange(phyla_abundances, desc(phyla_Ab)) 
# Create a df with the top 5 phyla
TOPPhyla <- unique(phyla_abundances$Phylum[1:5])
phylum_df <- phyla_abundances[phyla_abundances$Phylum %in% TOPPhyla,]
phylum_df$Phylum <- factor(phylum_df$Phylum, levels = phylum_df$Phylum[order(-phylum_df$phyla_Ab)])

genus_abundances <- arrange(genus_abundances, desc(genus_Ab))  
TOPGenera <- unique(genus_abundances$Genus[1:10])
genus_df <- genus_abundances[genus_abundances$Genus %in% TOPGenera,]
genus_df$Genus <- factor(genus_df$Genus, levels = genus_df$Genus[order(-genus_df$genus_Ab)])

#create df with top 10 species
species_abundances <- arrange(species_abundances, desc(species_Ab))  
TOPSpecies <- unique(species_abundances$Species[1:10])
species_df <- species_abundances[species_abundances$Species %in% TOPSpecies,]
species_df$Species <- factor(species_df$Species, levels = species_df$Species[order(-species_df$species_Ab)])

# Rename species other than top10 as "Other" in creating dataframe relative_df
relative_df <- merge(melted_df, abundances, by=c("Genus", "Species"))
#keep getting vector memory exhausted error; need to reboot R so will save melted_df and abundances as csv files
#source for fix: https://stackoverflow.com/questions/51295402/r-on-macos-error-vector-memory-exhausted-limit-reached

relative_df$Species[!(relative_df$Species %in% TOPSpecies)] <- "Other"
sum(relative_df$Abundance) #sums to 42.00003 (and we have n=42 kids...?)
table(relative_df$Species)

#############
#DELETE ABOVE
#############

#To generate color palette of 11 colors used previously: 
library(RColorBrewer)
library(scales)
chisp2 <- c("#293352", "#800000FF", "#767676FF", "#FFA319FF", "#8A9045FF", "#155F83FF", "#C16622FF", "#4D004B", "#8F3931FF", "#58593FFF", "#350E20FF")
show_col(chisp2)

#CREATING RELATIVE ABUNDANCE PLOT
rel_cd4$subject_abundances <- (rel_cd4$Abundance)/(length(unique(rel_cd4$Sample))) 
# #Clarify CD4 cutoff 
# rel_cd4$cd4cat <- NA
# rel_cd4$cd4cat [rel_cd4$cd4nL==1] <- "Normal CD4"
# rel_cd4$cd4cat [rel_cd4$cd4nL==0] <- "Low CD4"
# table(rel_cd4$cd4cat)
class(rel_cd4$cd4cat)
# rel_cd4$cd4cat <- as.factor(rel_cd4$cd4cat)
rel_cd4$cd4cat <- reorder(rel_cd4$cd4cat, new.order=c("Low CD4", "Normal CD4"))

rel_cd4$Species <- factor(rel_cd4$Species)
rel_cd4$Species <- reorder(rel_cd4$Species, new.order=c("Corynebacterium accolens","Corynebacterium propinquum", "Corynebacterium pseudodiphtheriticum", "Dolosigranulum pigrum", "Haemophilus influenzae", "Moraxella catarrhalis", "Moraxella lincolnii",  "Moraxella nonliquefaciens", "Staphylococcus aureus", "Streptococcus pneumoniae", "Other"))

subject_plot_cd4_2 <- ggplot(rel_cd4[order(rel_cd4$Species, decreasing = TRUE),], aes(x=cd4cat, y=subject_abundances, fill=Species)) + geom_bar(stat="identity", position="fill") + theme_classic() + theme(legend.position="right", legend.text=element_markdown(), axis.title.y = element_text(size=12, margin=margin(0,20,0,0)), axis.text.y = element_text(size=10), axis.title.x = element_blank(), axis.text.x = element_text(size=10, angle = 45, vjust = 1, hjust = 1, colour = "black")) + scale_fill_manual(values = chisp2, labels = c("*Corynebacterium accolens*", "*Corynebacterium propinquum*", "*Corynebacterium pseudodiphtheriticum*", "*Dolosigranulum pigrum*", "*Haemophilus influenzae*", "*Moraxella catarrhalis*", "*Moraxella lincolnii*", "*Moraxella nonliquefaciens*",  "*Staphylococcus aureus*", "*Streptococcus pneumoniae*", "Other")) +
  ylab("Mean relative abundance") 

p4 <- plot(subject_plot_cd4_2)


#NOW: combine all using patchwork vs cowplot (great ref: https://wilkelab.org/cowplot/articles/aligning_plots.html)
#For future reference: p1 and p2 code is in Fog Kaiju 5-26-22.R file, p3 and p4 are here
#After MK feedback, removed x axis titles for B and D, only capitalized "Mean" in y axis titles for B&D, and increased width of A&C (so ratio from 1, 1.65 to 1, 1.55)
fig2 <- plot_grid(p1, p2, p3, p4, labels = "AUTO", nrow = 2, ncol = 2, align = "h", axis = "b", rel_widths = c(1,1.3))

fig2

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/fig1NEW_V2.pdf", plot = fig2, dpi = 300, height = 9, width = 11)

```

####Comparing immunocompetent CLWH to HEU and HUU children

- will create a figure of the PCoA plots and rel abundance plots to include in LM appendix at request of reviewer

```{r}
#need to remove the specific kids with HIV that have low CD4 (N=11)
test <- chi_cd4[c("sample_id", "cd4nL")]
#remove B.01, B.03, B.04, B.05, B.09, B.11, B.18, B.25, B.27, B.32, B.46
  #also need to remove the kids missing CD4: B.45 and B.52
#so 50 HUU kids + 44 HEI kids - 2 kids without CD4 - 11 kids with low CD4 = 81
hucd4 = subset_samples(chi_prune, sample_id != "B.01.CHI" & sample_id != "B.03.CHI" & sample_id != "B.04.CHI" &
                         sample_id != "B.05.CHI" & sample_id != "B.09.CHI" & sample_id != "B.11.CHI" &
                         sample_id != "B.18.CHI" & sample_id != "B.25.CHI" & sample_id != "B.27.CHI" &
                         sample_id != "B.32.CHI" & sample_id != "B.45.CHI" & sample_id != "B.46.CHI" &
                         sample_id != "B.52.CHI")
hu <- data.frame(sample_data(hucd4))
table(hu$hiv) #adds up: 50 HUU kids + 49 HEU kids + 31 kids with HIV (44 orig - 13 removed above)

#SECOND: PCOA plots and PERMANOVA
#Transform our pseq object
hucd4_clr <- transform(hucd4, "clr")

#Create new immunocompetent-CLWH variable
table(hu$hiv)
hu$hiv2 <- NA
hu$hiv2 [hu$hiv == "Infected"] <- "Immunocompetent CLWH"
hu$hiv2 [hu$hiv == "Exposed Uninfected"] <- "HEU children"
hu$hiv2 [hu$hiv == "Unexposed"] <- "HUU children"
table(hu$hiv2)
class(hu$hiv2)
hu$hiv2 <- as.factor(hu$hiv2)
hu$hiv2 <- reorder(hu$hiv2, new.order=c("Immunocompetent CLWH", "HEU children", "HUU children"))

#Import back into pseq (source: https://jacobrprice.github.io/2017/08/26/phyloseq-to-vegan-and-back.html)
sample_data(hucd4_clr) <- as.data.frame(hu)
#now see if it worked:
bloop <- data.frame(sample_data(hucd4_clr))
table(bloop$hiv2)
#success!
remove(bloop)

#PCoA plot#
clr_pcoa <- ordinate(hucd4_clr, method = "PCoA", distance = "euclidean")
lines <- c("dashed", "dashed", "dashed")
clr_eiuu <- plot_ordination(hucd4_clr, clr_pcoa, color = "hiv2") +
  geom_point(size = 2) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values=c('#800000FF','#155F83FF', '#FFA319FF'))+
  stat_ellipse(aes(linetype=hiv2), geom="polygon", alpha=0, type="t", level=0.8, size=0.7) +
  xlab("PC1 (14.9%)") + ylab("PC2 (9.1%)") +
  scale_linetype_manual(values=lines)

clr_eiuu

#PERMANOVA
otu <- as.data.frame(otu_table(hucd4_clr, taxa_are_rows=TRUE))
#Need species in columns and observations in rows: transpose using t() function and save as a dataframe
#Source: https://github.com/edamame-course/2015-tutorials/blob/master/final/Ashley_Intro_to_R.md
otu_trans <- data.frame(t(otu))
BC.dist=vegdist(otu_trans, method="euclidean")

#hiv + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis(BC.dist ~ hiv + age, data = hu, permutations = 1000)
#No significant difference by HIV status among children (P = 0.18), but age remains significant (P=0.02, R2 = 0.014)
#Now getting message that adonis will be deprecated and I need to use adonis2
adonis2(BC.dist ~ hiv + age, data = hu, permutations = 1000)
#No significant difference by HIV status among children (P = 0.15, R2=0.018), but age remains significant (P=0.012, R2 = 0.014)
#check with age first given issues with order of variables in adonis; not signif different
adonis2(BC.dist ~ age + hiv, data = hu, permutations = 1000)

#clean up
remove(hucd4_clr, test, otu, otu_trans, clr_eiuu, clr_pcoa, bloop, BC.dist, lines)

#FOR RELATIVE ABUNDANCE PLOTS - SEE 9-28-23 FILE (WILL MERGE)

#MAASLIN 2: CLWH (immunocompetent only), HEU, HUU
hu.filter <- filter_taxa(hucd4, function(Abundance) mean(Abundance)>=50, TRUE)
ntaxa(hu.filter) #51

otu <- as.data.frame(otu_table(hu.filter, taxa_are_rows=TRUE))
#need to remove the unidentified species from here: row 10, 27, 44, 45, 47
#source: https://stackoverflow.com/questions/12328056/how-do-i-delete-rows-in-a-data-frame
#corresponding respectively to: Corynebacterium sp. KPL1859, Streptococcus sp. SK643, Alkalibacterium sp. 20, uncultured Clostridium sp., Nocardioides sp. Root122
otu <- otu[-c(10, 27, 44, 45, 47),]

#Univariable
fit_data2 = Maaslin2(
  input_data = otu, 
  input_metadata = hu, 
  min_prevalence = 0.2,
  max_significance = 0.2,
  output = "maaslin_immunocompetent_UV", 
  fixed_effects = c("hiv"),
  reference = c("hiv,Unexposed"))
#No significant assoc

fit_data2 = Maaslin2(
  input_data = otu, 
  input_metadata = hu, 
  min_prevalence = 0.2,
  max_significance = 0.2,
  output = "maaslin_immunocompetent_mv", 
  fixed_effects = c("hiv", "season", "age", "abx", "wood", "uri_rec2"),
  reference = c("hiv,Unexposed"))
#Signif assoc with age and season, but not HIV

#Comparing differentially abundant species: used lefse
```
**For final file: pull the lefse code in from the 'HIV MB LEfSe RMD file'**

####Comparing immunocompromised CLWH to HEU and HUU children

- will create a figure of the PCoA plots and rel abundance plots to include in LM appendix at request of reviewer (4 part figure combined with the immunocompetent CLWH sub-analysis above)

```{r}
hlowcd4 = subset_samples(chi_prune, sample_id != "B.06.CHI" & sample_id != "B.07.CHI" & sample_id != "B.12.CHI" &
              sample_id != "B.13.CHI" & sample_id != "B.14.CHI" & sample_id != "B.15.CHI" &
              sample_id != "B.16.CHI" & sample_id != "B.17.CHI" & sample_id != "B.19.CHI" &
              sample_id != "B.20.CHI" & sample_id != "B.22.CHI" & sample_id != "B.23.CHI" &
              sample_id != "B.26.CHI" & sample_id != "B.28.CHI" & sample_id != "B.29.CHI" &
              sample_id != "B.30.CHI" & sample_id != "B.31.CHI" & sample_id != "B.33.CHI" &
              sample_id != "B.34.CHI" & sample_id != "B.35.CHI" & sample_id != "B.36.CHI" &
              sample_id != "B.37.CHI" & sample_id != "B.38.CHI" & sample_id != "B.39.CHI" &
              sample_id != "B.40.CHI" & sample_id != "B.41.CHI" & sample_id != "B.44.CHI" &
              sample_id != "B.45.CHI" & sample_id != "B.48.CHI" & sample_id != "B.49.CHI" &
              sample_id != "B.50.CHI" & sample_id != "B.51.CHI" & sample_id != "B.52.CHI")
hlow <- data.frame(sample_data(hlowcd4))
table(hlow$hiv) #adds up: 50 HUU kids + 49 HEU kids + 11 kids with HIV (44 orig - 33 removed above)

#SECOND: PERMANOVA +/- PCoA plot
#Transform our pseq object
hlowcd4_clr <- transform(hlowcd4, "clr")

#PERMANOVA
otu <- as.data.frame(otu_table(hlowcd4_clr, taxa_are_rows=TRUE))
#Need species in columns and observations in rows: transpose using t() function and save as a dataframe
#Source: https://github.com/edamame-course/2015-tutorials/blob/master/final/Ashley_Intro_to_R.md
otu_trans <- data.frame(t(otu))
BC.dist=vegdist(otu_trans, method="euclidean")

#hiv + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis2(BC.dist ~ hiv + age, data = hlow, permutations = 1000)
#Significant difference by HIV status among children (P = 0.003, R2 = 0.032) and age (P=0.03, R2 = 0.015)
#check with age first given issues with order of variables in adonis; not signif different
adonis2(BC.dist ~ age + hiv, data = hlow, permutations = 1000)

#PCoA plot#
table(hlow$hiv)
hlow$hiv2 <- NA
hlow$hiv2 [hlow$hiv == "Infected"] <- "Immunocompromised CLWH"
hlow$hiv2 [hlow$hiv == "Exposed Uninfected"] <- "HEU children"
hlow$hiv2 [hlow$hiv == "Unexposed"] <- "HUU children"
table(hlow$hiv2)
class(hlow$hiv2)
hlow$hiv2 <- as.factor(hlow$hiv2)
hlow$hiv2 <- reorder(hlow$hiv2, new.order=c("Immunocompromised CLWH", "HEU children", "HUU children"))

#Import back into pseq (source: https://jacobrprice.github.io/2017/08/26/phyloseq-to-vegan-and-back.html)
sample_data(hlowcd4_clr) <- as.data.frame(hlow)
#now see if it worked:
bloop <- data.frame(sample_data(hlowcd4_clr))
table(bloop$hiv2)
#success!
remove(bloop)

clr_pcoa <- ordinate(hlowcd4_clr, method = "PCoA", distance = "euclidean")
lines <- c("dashed", "dashed", "dashed")

clr_hlow <- plot_ordination(hlowcd4_clr, clr_pcoa, color = "hiv2") +
  geom_point(size = 0.005) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values=c('#800000FF','#155F83FF', '#FFA319FF'))+
  stat_ellipse(aes(linetype=hiv2), geom="polygon", alpha=0, type="t", level=0.8, size=0.5) +
  xlab("PC1 (15.4%)") + ylab("PC2 (9.2%)") +
  scale_linetype_manual(values= c("dashed", "dashed", "dashed"))

clr_hlow

#clean up
# remove(clr_pcoa, hlowcd4_clr, otu, otu_trans, BC.dist, lines)

#CREATE MELTED_DF FOR RELATIVE ABUNDANCE PLOTS --> SEE 9-28-23 FILE

#NOW: make a 4 part figure combining our PCoA plots and relative abundance plots for our subanalyses of HEU and HUU children vs 1) CLWH and low CD4 and 2) CLWH and normal CD4

fig_HEU_HUU_CD4hilow <- plot_grid(clr_hlow, spec_hlow, clr_eiuu, spec_hu,labels = "AUTO", nrow = 2, ncol = 2, align = "h", axis = "b", rel_widths = c(1,1.3))

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig S_CD4hilow_HEU_HUU_V2.pdf", plot = fig_HEU_HUU_CD4hilow, dpi = 300, width = 11, height = 9)

png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig S_CD4hilow_HEU_HUU_V2.png",
    width = 11, height = 9, units = 'in', res = 600)
fig_HEU_HUU_CD4hilow
dev.off()

###PULL IN MAASLIN2 AND LEFSE CODE FOR FINAL GITHUB FILE###
```

####HEU-HUU Comparisons

A reviewer asked "And you also might wonder whether there is a difference HEU vs. HUU and what that might mean for the other two comparisons if there were?"

So we will compare beta diversity and maaslin2 for HEU vs HUU kids only. Can generate PCoA plot, but will not create a supplemental figure unless we see differences of note.

```{r}
no_hiv = subset_samples(chi_prune, hiv != "Infected")

nohiv <- data.frame(sample_data(no_hiv))
table(nohiv$hiv) #adds up: 50 HUU kids + 49 HEU kids only

################
#ALPHA DIVERSITY
################
#Create dataframe with diversity indices for each specimen using PRUNED pseq object
diversity <- estimate_richness(no_hiv, measures = c("Shannon", "Simpson", "Chao1", "Observed"))

setDT(diversity, keep.rownames = TRUE)[]
colnames(diversity) <- c("sample_id", "Observed", "Chao1", "se.chao1", "Shannon", "Simpson")
rownames(diversity) <- diversity$sample_id
diversity <- merge(diversity, nohiv, by="sample_id")

class(diversity$hiv) #character; need to make factor to relevel
diversity$hiv <- as.factor(diversity$hiv)
diversity$hiv <- relevel(diversity$hiv, ref = "Unexposed")
summary(div_mv <- lm(Shannon ~ age + hiv + wood + season + abx + uri_rec2, data = diversity))

#No association between HEU status and alpha diversity measures

###################
#PERMANOVA and PCoA
###################

#Transform our pseq object
nohiv_clr <- transform(no_hiv, "clr")

otu <- as.data.frame(otu_table(nohiv_clr, taxa_are_rows=TRUE))
#Need species in columns and observations in rows: transpose using t() function and save as a dataframe
#Source: https://github.com/edamame-course/2015-tutorials/blob/master/final/Ashley_Intro_to_R.md
otu_trans <- data.frame(t(otu))
BC.dist=vegdist(otu_trans, method="euclidean")

#hiv + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis2(BC.dist ~ hiv + age, data = nohiv, permutations = 1000, by = "margin")
#Significant difference by age among children (P = 0.02, R2 = 0.018) but not HEU status (P=0.09, R2 = 0.014)

#PCoA plot#
table(nohiv$hiv)
nohiv$hiv2 <- NA
nohiv$hiv2 [nohiv$hiv == "Exposed Uninfected"] <- "HEU children"
nohiv$hiv2 [nohiv$hiv == "Unexposed"] <- "HUU children"
table(nohiv$hiv2)
class(nohiv$hiv2)
nohiv$hiv2 <- as.factor(nohiv$hiv2)
nohiv$hiv2 <- reorder(nohiv$hiv2, new.order=c("HEU children", "HUU children"))

#Import back into pseq (source: https://jacobrprice.github.io/2017/08/26/phyloseq-to-vegan-and-back.html)
sample_data(nohiv_clr) <- as.data.frame(nohiv)
#now see if it worked:
bloop <- data.frame(sample_data(nohiv_clr))
table(bloop$hiv2)
#success!
remove(bloop)

clr_pcoa <- ordinate(nohiv_clr, method = "PCoA", distance = "euclidean")
lines <- c("dashed", "dashed")

p_no <- plot_ordination(nohiv_clr, clr_pcoa, color = "hiv2") +
  geom_point(size = 0.005) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values=c('#800000FF','#155F83FF'))+
  stat_ellipse(aes(linetype=hiv2), geom="polygon", alpha=0, type="t", level=0.8, size=0.5) +
  xlab("PC1 (15.9%)") + ylab("PC2 (9.6%)") +
  scale_linetype_manual(values= c("dashed", "dashed"))

p_no

#########
#MAASLIN2
#########
#Maaslin2 tutorial: https://github.com/biobakery/biobakery/wiki/maaslin2

#UPDATE per d/w MK 5/21: filter based on round abundance of 50 and re-run the model
nohiv.filter <- filter_taxa(no_hiv, function(Abundance) mean(Abundance)>=50, TRUE)
ntaxa(nohiv.filter) #54

otu <- as.data.frame(otu_table(nohiv.filter, taxa_are_rows=TRUE))
###PAUSED HERE 8-24-23
#need to remove the unidentified species from here (n=6): row 9, 29, 46, 47, 49, 51
  #source: https://stackoverflow.com/questions/12328056/how-do-i-delete-rows-in-a-data-frame
#corresponding respectively to: Corynebacterium sp. KPL1859, Streptococcus sp. SK643, Alkalibacterium sp. 20, uncultured Clostridium sp., Nocardioides sp. Root122, Microbacterium sp. 70-16
otu <- otu[-c(9, 29, 46, 47, 49, 51),]

#Corynebacterium sp. KPL1995 -> Corynebacterium pseudodiphtheriticum
#Corynebacterium sp. KPL1986 -> Corynebacterium accolens
#Haemophilus sp. C1 -> Haemophilus haemolyticus

row.names(otu)[3] <- "Haemophilus haemolyticus"
row.names(otu)[8] <- "Corynebacterium pseudodiphtheriticum"
row.names(otu)[9] <- "Corynebacterium accolens"

fit_data2 = Maaslin2(
  input_data = otu, 
  input_metadata = nohiv, 
  min_prevalence = 0.2,
  max_significance = 0.2,
  output = "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/maaslin_mv_filt50_nohiv", 
  fixed_effects = c("hiv", "season", "age", "abx", "wood", "uri_rec2"),
  reference = c("hiv,Unexposed"))
```

**Summary of findings: no difference in beta diversity or differentially abundant species among HEU children when compared to HUU children.**

### HEU-CLWH Sibling Comparisons

Create our limited pseq object
```{r}
sib_pair = subset_samples(pruned, sample_id == "B.09.CHI" | sample_id == "B.09.SIB" | sample_id == "B.16.CHI" | 
                            sample_id == "B.16.SIB" | sample_id == "B.17.CHI" | sample_id == "B.17.SIB" | sample_id == "B.20.CHI" |
                            sample_id == "B.20.SIB" | sample_id == "B.23.CHI" | sample_id == "B.23.SIB" | sample_id == "B.25.CHI" |
                            sample_id == "B.25.SIB" | sample_id == "B.32.CHI" | sample_id == "B.32.SIB" | sample_id == "B.35.CHI" |
                            sample_id == "B.35.SIB" | sample_id == "B.48.CHI" | sample_id == "B.48.SIB")

sib <- data.frame(sample_data(sib_pair))

#cleaned + simplified metadata with standardized variables already created and saved, but will re-create to add breastfeeding given reviewer request:
sib2 <- sib[c("sample_id", "study_id", "age", "sex", "season", "subject", "abx", "uri_rec2", "pcv", "dpt", "bfeed_1", "muac", "sib_age", "sib_sex", "sib_uri_recent", "sib_meds", "sib_meds_name", "sib_dpt", "sib_pcv", "sibmo", "sib_bfeed_1", "sib_muac")]

#FIRST: create new variables for sibling abx, season, and uri_rec2
#abx: cleaned up sib med names in separate metadata file --> "sib_abx 8-10-22"csv file
#GO BACK AND CLEAN IN REDCAP ONCE ACCESS RE-ESTABLISHED
abx <- read.csv("~/Library/CloudStorage/OneDrive-DukeUniversity/Fogarty coding/sib_abx 08-10-22.csv")
#need to add NAs? 
#drop current med_names variable from metadata and then merge with abx df by study_id
#standardize study_id
toString(abx$study_id)
abx$study_id <- gsub("-", ".", abx$study_id)
sib2$sib_meds_name <- NULL
sib2$sib_meds <- NULL
sib2 <- merge(sib2, abx, by ="study_id", sort = TRUE)
table(sib2$sib_meds_name)
sib2$sib_abx <- "No"
sib2$sib_abx [sib2$sib_meds_name == "amoxicillin"] <- "Yes"
table(sib2$sib_abx)

#season
#For consistency with MK: Rainy season = Nov to March. Dry season = April to Oct
sib2$sib_seas <- "99"
sib2$sib_seas [sib2$sibmo == 1 | sib2$sibmo == 2 | sib2$sibmo == 3 | sib2$sibmo == 11 | sib2$sibmo == 12] <- "Rainy"
sib2$sib_seas [sib2$sibmo == 4 | sib2$sibmo == 5 | sib2$sibmo == 6 | sib2$sibmo == 7 | sib2$sibmo == 8 | sib2$sibmo == 9 | sib2$sibmo == 10] <- "Dry"
table(sib2$sib_seas)

#recent uri
table(sib2$sib_uri_recent)
sib2$sib_uri_rec2 <- NA
sib2$sib_uri_rec2 [sib2$sib_uri_recent == 1] <- "Yes"
sib2$sib_uri_rec2 [sib2$sib_uri_recent == 2] <- "No"
table(sib2$sib_uri_rec2)

#breastfeeding CLWH
table(sib2$bfeed_1)
sib2$bfeed_any <- NA
sib2$bfeed_any [sib$bfeed_1 == 1 | sib$bfeed_1 == 3] <- "Yes"
sib2$bfeed_any [sib$bfeed_1 == 2] <- "No"
table(sib2$bfeed_any)

#breastfeeding sib
table(sib2$sib_bfeed_1) #no sibling received both breast milk and formula
sib2$sib_bfeed_any <- NA
sib2$sib_bfeed_any [sib2$sib_bfeed_1 == 1] <- "Yes"
sib2$sib_bfeed_any [sib2$sib_bfeed_1 == 2] <- "No"
table(sib2$sib_bfeed_any)

###UPDATE again 9-19-23; add in height and weight for the sib pairs
sib2$height <- sib$height
sib2$sib_height <- sib$sib_height
sib2$weight <- sib$weight
sib2$sib_weight <- sib$sib_weight

#NEXT: pull the variables out separately, rename the sibling variables so standardized with chi vars, then merge back together
sib2 <- within(sib2, rm(sib_uri_recent, sibmo, sib_meds, sib_meds_name))
sib_chi <- subset(sib2, sib2$subject == "child")
sib_chi <- sib_chi[c("sample_id", "study_id", "age", "sex", "season", "subject", "abx", "uri_rec2", "pcv", "dpt", "bfeed_any", "muac", "height", "weight")]
sib_sib <- subset(sib2, sib2$subject == "sib")
sib_sib <- sib_sib[c("sample_id", "study_id", "subject", "sib_age", "sib_sex",
              "sib_uri_rec2", "sib_abx", "sib_dpt", "sib_pcv", "sib_seas", "sib_bfeed_any", "sib_muac", "sib_height", "sib_weight")]
#rename sib variables
sib_sib <- rename(sib_sib, age = "sib_age")
sib_sib <- rename(sib_sib, sex = "sib_sex")
sib_sib <- rename(sib_sib, uri_rec2 = "sib_uri_rec2")
sib_sib <- rename(sib_sib, abx = "sib_abx")
sib_sib <- rename(sib_sib, dpt = "sib_dpt")
sib_sib <- rename(sib_sib, pcv = "sib_pcv")
sib_sib <- rename(sib_sib, season = "sib_seas")
sib_sib <- rename(sib_sib, bfeed_any = "sib_bfeed_any")
sib_sib <- rename(sib_sib, muac = "sib_muac")
sib_sib <- rename(sib_sib, height = "sib_height")
sib_sib <- rename(sib_sib, weight = "sib_weight")

#combine: use rbind
sib3 <- rbind(sib_chi, sib_sib)
#save as csv for easier reference
write.csv(sib3, 'sibpairs.csv')
sib3 <- read.csv("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/sibpairs.csv")

#now compare demographic characteristics by group AND pairwise
  #paired comparisons for numeric data: wilcoxon paired
  #paired comparisons for 2x2 tables: McNemar test: https://rpubs.com/kaz_yos/mcnemar
library(exact2x2)
#Age
hist(sib3$age)
summary(sib3$age)
tapply(sib3$age, sib3$subject, summary)
wilcox.test(age ~ subject, data = sib3, exact=FALSE) #P=0.57
  #paired
wilcox.test(age ~ subject, data=sib3, paired=TRUE, exact=FALSE) #P=0.59

#Sex
table(sib3$sex)  #11 girls
table(sib3$sex, sib3$subject)
prop.table(table(sib3$sex, sib3$subject), 2)
fisher.test(sib3$sex, sib3$subject) #P=1.0
  #paired
mcnemar.test(sib3$sex, sib3$subject)  #P=0.72

#Season
table(sib3$season)  
table(sib3$season, sib3$subject)
prop.table(table(sib3$season, sib3$subject), 2)
fisher.test(sib3$season, sib3$subject)  #P=1.0
  #paired
mcnemar.test(sib3$season, sib3$subject) #P=1.0

#antibiotics
table(sib3$abx)  #3 kids received abx
table(sib3$abx, sib3$subject)
prop.table(table(sib3$abx, sib3$subject), 2)
fisher.test(sib3$abx, sib3$subject) #P=1.0
  #paired
mcnemar.test(sib3$abx, sib3$subject) #P=0.08

#URI in past 1 month
table(sib3$uri_rec2)  #11 kids with recent URI
table(sib3$uri_rec2, sib3$subject)
prop.table(table(sib3$uri_rec2, sib3$subject), 2)
fisher.test(sib3$uri_rec2, sib3$subject)
  #paired
mcnemar.test(sib3$uri_rec2, sib3$subject) #P=0.75

#PCV doses
table(sib3$pcv)  
table(sib3$pcv, sib3$subject)
prop.table(table(sib3$pcv, sib3$subject), 2)
fisher.test(sib3$pcv, sib3$subject) #P=1.0
  #paired
wilcox.test(pcv ~ subject, data=sib3, paired=TRUE, exact=FALSE) #P=0.85

#HiB
table(sib3$dpt)  
table(sib3$dpt, sib3$subject)
prop.table(table(sib3$dpt, sib3$subject), 2)
fisher.test(sib3$dpt, sib3$subject) #P=1.0
  #paired
wilcox.test(dpt ~ subject, data=sib3, paired=TRUE, exact=FALSE) #P=0.85

#Any breastfeeding
table(sib3$bfeed_any)  #8 kids with breastfeeding
table(sib3$bfeed_any, sib3$subject)
prop.table(table(sib3$bfeed_any, sib3$subject), 2)
  #paired
mcnemar.test(sib3$bfeed_any, sib3$subject) #P=1.0

#MUAC
hist(sib3$muac)
summary(sib3$muac)
tapply(sib3$muac, sib3$subject, summary)
wilcox.test(muac ~ subject, data = sib3, exact=FALSE) #P=0.05
  #paired
wilcox.test(muac ~ subject, data=sib3, paired=TRUE, exact=FALSE)
  #can't run paired tests on all 9 pairs; 3 pairs missing muac in either child or sib, so will drop them and run paired wilcoxon.
sib4 <- subset(sib4, study_id != "B.23" & study_id != "B.35" & study_id != "B.48")
wilcox.test(muac ~ subject, data=sib4, paired=TRUE, exact=FALSE) #only with 6 pairs in analysis; p = 0.04

#calculate height for age and weight for age z scores and then do paired comparison
#sex needs to be coded as male (1) and female (2)
sib3$sex2 <- NA
sib3$sex2 [sib3$sex == 0] <- 1
sib3$sex2 [sib3$sex == 1] <- 2
table(sib3$sex)
table(sib3$sex2) #matches up
#age needs to be in days, not months
sib3$age_d <- sib3$age*(30.4375) #confirmed that B-01 age and age_d match up

#weight for age
sib3 <- addWGSR(data=sib3, sex="sex2", firstPart = "weight", secondPart = "age_d", index="wfa")
summary(sib3$wfaz) 
hist(sib3$wfaz) #roughly normal, slight left skew
tapply(sib3$wfaz, sib3$subject, summary)
wilcox.test(wfaz ~ subject, data=sib3, paired=TRUE, exact=FALSE)

#height for age
sib3 <- addWGSR(data=sib3, sex="sex2", firstPart = "height", secondPart = "age_d", index="hfa")
summary(sib3$hfaz)
hist(sib3$hfaz) #roughly normal, slight right skew
tapply(sib3$hfaz, sib3$subject, summary)
wilcox.test(hfaz ~ subject, data=sib3, paired=TRUE, exact=FALSE)

sib_clr <- transform(sib_pair, "clr")

#################
#Paired PERMANOVA
#################
otu <- as.data.frame(otu_table(sib_clr, taxa_are_rows=TRUE))
#Need species in columns and observations in rows: transpose using t() function and save as a dataframe
#Source: https://github.com/edamame-course/2015-tutorials/blob/master/final/Ashley_Intro_to_R.md
otu_trans <- data.frame(t(otu))
BC.dist=vegdist(otu_trans, method="euclidean")

pairwise.adonis2(BC.dist ~ subject, data = sib3, strata = 'study_id', p.adjust.m = "BH")
pairwise.adonis2(BC.dist ~ age, data = sib3, strata = 'study_id', p.adjust.m = "BH") #something about this code is not right; generates comparisons for all age options
pairwise.adonis2(BC.dist ~ sex, data = sib3, strata = 'study_id', p.adjust.m = "BH") #not signif
pairwise.adonis2(BC.dist ~ bfeed_any, data = sib3, strata = 'study_id', p.adjust.m = "BH") #not signif
pairwise.adonis2(BC.dist ~ season, data = sib3, strata = 'study_id', p.adjust.m = "BH") #not signif
pairwise.adonis2(BC.dist ~ uri_rec2, data = sib3, strata = 'study_id', p.adjust.m = "BH") #not signif

#not sure if we can run pairwise multivariable permanova
pairwise.adonis2(BC.dist ~ subject + season + uri_rec2 + age, data = sib3, strata = 'study_id', p.adjust.m = "BH")
  #here, uri sx are a/w permanova stratified by study id, but why significant here and not in univariable analysis?

#regular non-pairwise permanova: subject significantly associated with composition but not surprising since subject is a surrogate for HIV status (child = CLWH, sib = HEU child)
adonis2(BC.dist ~ subject + season + uri_rec2 + age + bfeed_any, data = sib3, permutations = 1000, by = "margin")

```

### Sensitivity analysis: breastfeeding

- We see a strong association between HIV status and any history of breastfeeding in our cohort. Given the established association between breastfeeding and NP microbiome composition, what do our results look like if we compare NP microbiome characteristics among children who were breastfed? Initially we analyzed only children with exclusive bf, but after d/w MK will conduct analysis with children who received any breastmilk (n=84)

```{r}
#Compare numbers of any breastfeeding to exclusive breastfeeding 
#Create variable that combines exclusive and partial breastfeeding
table(child$bfeed_1)
child$bfeed_any <- NA
child$bfeed_any [child$bfeed_1 == 1 | child$bfeed_1 == 3] <- "Yes"
child$bfeed_any [child$bfeed_1 == 2] <- "No"
table(child$bfeed_any)
table(child$bfeed_any, child$hiv)
prop.table(table(child$bfeed_any, child$hiv), 2)
summary(table(child$bfeed_any, child$hiv))

table(child$bfeed_1) #49 kids exclusively breastfed, 59 formula fed, and 35 received both breastmilk and formula
table(child$bfeed_1, child$hiv)
prop.table(table(child$bfeed_1, child$hiv), 2)
# summary(table(child$bfeed_1, child$hiv))
fisher.test(child$bfeed_1, child$hiv)

summary(child$bfeed_2)

#Create pseq object of breastfed children
chi_bf = subset_samples(chi_prune, bfeed_1 != 2) #should have 84 kids and it does
bf <- data.frame(sample_data(chi_bf))
table(bf$hiv)
```

#### Alpha diversity analyses 
```{r}
################
#ALPHA DIVERSITY
################
#Create dataframe with diversity indices for each specimen using PRUNED pseq object
diversity <- estimate_richness(chi_bf, measures = c("Shannon", "Simpson", "Chao1", "Observed"))

setDT(diversity, keep.rownames = TRUE)[]
colnames(diversity) <- c("sample_id", "Observed", "Chao1", "se.chao1", "Shannon", "Simpson")
rownames(diversity) <- diversity$sample_id
diversity <- merge(diversity, bf, by="sample_id")

class(diversity$hiv) #character; need to make factor to relevel
diversity$hiv <- as.factor(diversity$hiv)
diversity$hiv <- relevel(diversity$hiv, ref = "Unexposed")
summary(div_bf <- lm(Shannon ~ hiv, data = diversity)) #no difference in univariable analysis
summary(div_mv <- lm(Shannon ~ age + hiv + wood + season + abx + uri_rec2, data = diversity))

#No association between hIV status and alpha diversity measures among breastfed children
```

#### Beta diversity

- will run permanova and maaslin2

```{r}
###################
#PERMANOVA and PCoA
###################

#Transform our pseq object
chibf_clr <- transform(chi_bf, "clr")

otu <- as.data.frame(otu_table(chibf_clr, taxa_are_rows=TRUE))
otu_trans <- data.frame(t(otu))
BC.dist=vegdist(otu_trans, method="euclidean")

#hiv + age (per 6/9 MK discussion to account for age in all PERMANOVAs)
adonis2(BC.dist ~ hiv + age, data = bf, permutations = 1000, by = "margin")
#Significant difference by HIV status among breastfed children (P = 0.01, R2 = 0.039) and age (P=0.048, R2 = 0.018)

#full multivariable permanova
adonis2(BC.dist ~ hiv + season + uri_rec2 + age, data = bf, permutations = 1000, by = "margin")

#In a multivariable model, HIV status (P=0.015, R2=0.036), season (P=0.02, R2=0.020), recent URI (P=0.04, R2=0.018), and age (P=0.03, R2=0.020) were a/w differences in overall microbiome composition among breastfed children in our cohort (R2 and p values are from by = "margin")

#what was the average age among exclusively breasfed children?
summary(bf$age) #median (IQR) 35 (19, 52); not that different from overall cohort

#PCoA plot#
table(bf$hiv)
bf$hiv2 <- NA
bf$hiv2 [bf$hiv == "Infected"] <- "CLWH"
bf$hiv2 [bf$hiv == "Exposed Uninfected"] <- "HEU children"
bf$hiv2 [bf$hiv == "Unexposed"] <- "HUU children"
table(bf$hiv2)
class(bf$hiv2)
bf$hiv2 <- as.factor(bf$hiv2)
bf$hiv2 <- reorder(bf$hiv2, new.order=c("CLWH", "HEU children", "HUU children"))

#Import back into pseq (source: https://jacobrprice.github.io/2017/08/26/phyloseq-to-vegan-and-back.html)
sample_data(chibf_clr) <- as.data.frame(bf)
#now see if it worked:
bloop <- data.frame(sample_data(chibf_clr))
table(bloop$hiv2)
#success!
remove(bloop)

clr_pcoa <- ordinate(chibf_clr, method = "PCoA", distance = "euclidean")
p_bf <- plot_ordination(chibf_clr, clr_pcoa, color = "hiv2") +
  geom_point(size = 0.005) + theme_classic() + theme(
    plot.title = element_text(size = 14, hjust = 0.5),
    legend.title = element_blank(),
    legend.text = element_text(size = 10),
    legend.position = "right",
    legend.background = element_blank(),
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 12)) +
  scale_color_manual(values=c('#800000FF','#155F83FF', '#FFA319FF'))+
  stat_ellipse(aes(linetype=hiv2), geom="polygon", alpha=0, type="t", level=0.8, size=0.5) +
  xlab("PC1 (14.9%)") + ylab("PC2 (9.7%)") +
  scale_linetype_manual(values= c("dashed", "dashed", "dashed"))

p_bf

#########
#MAASLIN2
#########
#Maaslin2 tutorial: https://github.com/biobakery/biobakery/wiki/maaslin2

chibf.filter <- filter_taxa(chi_bf, function(Abundance) mean(Abundance)>=50, TRUE)
ntaxa(chibf.filter) #36

otu <- as.data.frame(otu_table(chibf.filter, taxa_are_rows=TRUE))

#need to remove the unidentified species from here (n=4): row 8, 20, 32, 33
  #source: https://stackoverflow.com/questions/12328056/how-do-i-delete-rows-in-a-data-frame
#corresponding respectively to: Corynebacterium sp. KPL1859, Streptococcus sp. SK643, Alkalibacterium sp. 20, uncultured Clostridium sp.
otu <- otu[-c(8, 20, 32, 33),]

#Corynebacterium sp. KPL1995 -> Corynebacterium pseudodiphtheriticum
#Corynebacterium sp. KPL1986 -> Corynebacterium accolens
#Haemophilus sp. C1 -> Haemophilus haemolyticus

row.names(otu)[3] <- "Haemophilus haemolyticus"
row.names(otu)[7] <- "Corynebacterium pseudodiphtheriticum"
row.names(otu)[8] <- "Corynebacterium accolens"

fit_data2 = Maaslin2(
  input_data = otu, 
  input_metadata = bf, 
  min_prevalence = 0.2,
  max_significance = 0.2,
  output = "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/maaslin_mv_filt50_bf_redo", 
  fixed_effects = c("hiv2", "season", "age", "abx", "wood", "uri_rec2"),
  reference = c("hiv2,HUU children"))

#Maaslin with HEU as reference
fit_data2 = Maaslin2(
  input_data = otu, 
  input_metadata = bf, 
  min_prevalence = 0.2,
  max_significance = 0.2,
  output = "/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Sequencing/maaslin_mv_filt50_bf_HEUref", 
  fixed_effects = c("hiv2", "season", "age", "abx", "wood", "uri_rec2"),
  reference = c("hiv2,HEU children"))

#Among children with any breastfeeding history, no differentially abundant species by HIV status between CLWH and HUU children, BUT species remained differentially abundant between HEU and CLWH
```

#### Relative abundance plots: REDONE (see 9-28-23 file)

```{r}
#Create supplemental sensitivity analysis figure
fig_bf <- plot_grid(p_bf, spec_bf, labels = c('A', 'B'), align = "h", axis = "b", label_size = 12, rel_widths = c(1, 1.3))
fig_bf

ggsave("/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig_bf_V2.pdf", plot = fig_bf, dpi = 300, width = 11, height = 4)

png(file="/Users/swetapatel/OneDrive - Duke University/Fogarty coding/Fig S_bf_V2.png",
    width = 11, height = 4, units = 'in', res = 600)
fig_bf
dev.off()

```

